---
title: "IBD_risk_v2"
author: "Iwan"
date: "2024-09-10"
output: html_document
---

```{r setup, include=FALSE}
library(ggplot2)
library(vegan)
library(dplyr)
library(tidyverse)
library(ggsignif)
```


# loading in functions

```{r }
#### Ranko's geweldige filter functie -----
#================================================================================
  # function for filtering microbiome (metaphlan) results
  # - takes dataframe (any)
  # - replaces NAs with 0.0
  # - filters OUT taxa present in less then presPerc samples
  # - filters OUT taxa with mean releative abundance < minMRelAb
  # - filters OUT taxa with median relative abundance < minMedRelAb
  # - filters OUT taxonomic levels not in keepLevels ()
  #   -> keepLevels should be vector of following: T = strain, S = species, G = Genera, F = families
  #                                                O = Orders, C = classes, P = phyla, K = Kingdoms
  #   -> example: keepLevels=c('S','G') keeps only species and genera
  # - filters OUT domains not in keepDomains: 
#   -> input is vector of following: Eukaryota, Bacteria, Viruses, Archaea
#   -> example: keepDomains=c('B') keeps only bacteria, removes viruses, archaea and eukarya
# - if rescaleTaxa = True, rescales relative abundancies after filtering
# returns modified dataframe
# NOTES:
# - assumes metaphlan encoding (k__<kingdom>.o__<order> ... ); it will mess stuff
# up if non-metagenome rows are encoded like this!
# DEFAULTS: 
# - removes non-bacteria
# - keeps all except kingdoms
filterMetaGenomeDF <- function(inDF,presPerc = 0.1,minMRelAb = 0.01,minMedRelAb=0.0,
                               rescaleTaxa=T,verbose=T,
                               keepDomains=c('Bacteria','Archaea'),
                               keepLevels=c('S','G','F','O','C','P'),
                               keepUnknown=F) {
  
  # -- drop unknown & rescale? --
  if (!keepUnknown) {
    inDF$UNKNOWN <- NULL
  }
  tCols = grep('k__',colnames(inDF)) # colums with microbiome
  tColsNMG = grep('k__',colnames(inDF),invert = T) # colums with microbiome
  # replaces NAs in microbiome with 0s
  for (c in tCols) {
    inDF[,c][is.na(inDF[,c])] <- 0.0
  }
  
  # filter for presence
  # -----------------
  nrRemoved = 0
  toRemove = c()
  for (c in tCols) {
    nrnZ = as.numeric(sum(inDF[,c]!=0.0))
    if ( (nrnZ/as.numeric(nrow(inDF))) < presPerc) {
      # if (verbose) {
      #   print (paste('col',c,': ',colnames(inDF)[c],'; nr non Zero:',nrnZ,'=',nrnZ/as.numeric(nrow(inDF)),'>> Column removed!'))
      # }
      nrRemoved = nrRemoved + 1
      toRemove <- c(toRemove,c)
    }
  }
  if (length(toRemove) > 0) {
    inDF <- inDF[,-toRemove]
  }
  tCols = grep('^[dgtspcfko]__',colnames(inDF)) # colums with microbiome
  if (verbose) {print (paste(' > presence filter: Removed',nrRemoved,'taxa!, ',length(tCols),'taxa left!')); }
  
  # filter for abundance (mean)
  # ---------------------------
  nrRemoved = 0
  toRemove = c()
  for (c in tCols) {
    mn = mean(inDF[,c])
    if ( mn < minMRelAb) {
      #if (verbose) {
      #  print (paste('col',c,': ',colnames(inDF)[c],'; mean rel abundance:',mn,' >> Column removed!'))
      #}
      nrRemoved = nrRemoved + 1
      toRemove <- c(toRemove,c)
    }
  }
  if (length(toRemove) > 0) {
    inDF <- inDF[,-toRemove]
  }
  tCols = grep('k__',colnames(inDF)) # colums with microbiome
  if (verbose) {print (paste(' > mean abundance filter: Removed',nrRemoved,'taxa!, ',length(tCols),'taxa left!')); }
  
  # filter for abundance (median)
  # -----------------------------
  nrRemoved = 0
  toRemove = c()
  for (c in tCols) {
    mn = median(inDF[,c])
    if ( mn < minMedRelAb) {
      #if (verbose) {
      #  print (paste('col',c,': ',colnames(inDF)[c],'; median rel abundance:',mn,' >> Column removed!'))
      #}
      nrRemoved = nrRemoved + 1
      toRemove <- c(toRemove,c)
    }
  }
  if (length(toRemove) > 0) {
    inDF <- inDF[,-toRemove]
  }
  if (verbose) {print (paste(' > median abundance filter: Removed',nrRemoved,'taxa!, ',length(tCols),'taxa left!')); }  
  
  # keep domains
  # -----------------------------
  toKeep <- NULL
  if (length(keepDomains) == 1) {
    if (keepDomains == 'All' | keepDomains == "") {
      toKeep = grep('k__',colnames(inDF),invert = T)
    } else {
      toKeep = grep(paste('k__',d,sep=''),colnames(inDF))
    }
  } else {
    for (d in keepDomains) {
      #d print(d)
      toKeep = c(toKeep,grep(paste('k__',d,sep=''),colnames(inDF)))
    }
  }
  inDF <- inDF[,toKeep]
  taxaP <- c()
  taxaP <- as.data.frame(taxaP)
  taxaC <- c()
  taxaC <- as.data.frame(taxaC)
  taxaO <- c()
  taxaO <- as.data.frame(taxaO)
  taxaK <- c()
  taxaK <- as.data.frame(taxaK)
  taxaF <- c()
  taxaF <- as.data.frame(taxaF)
  taxaG <- c()
  taxaG <- as.data.frame(taxaG)
  taxaS <- c()
  taxaS <- as.data.frame(taxaS)
  taxaT <- c()
  taxaT <- as.data.frame(taxaT)
    # remove taxonomic levels
  # -----------------------------
  inDFnonTaxa <- as.data.frame(inDF[,grep('k__',colnames(inDF),invert=T)])
  colnames(inDFnonTaxa) <- colnames(inDF)[grep('k__',colnames(inDF),invert=T)]
  
  inDF2 <- as.data.frame(inDF[,grep('k__',colnames(inDF),invert=F)])
  # pick strains (T)
  taxaTCols <- grep('t__',colnames(inDF2))
  taxaT <- as.data.frame(inDF2[,taxaTCols])
  colnames(taxaT) <- colnames(inDF2)[grep('t__',colnames(inDF2))]
  if (length(taxaTCols) > 0) {inDF2 <- inDF2[,-taxaTCols]}
  # pick species (S)
  taxaSCols <- grep('s__',colnames(inDF2))
  if (length(taxaSCols) > 0) {
    taxaS <- as.data.frame(inDF2[,taxaSCols])
    colnames(taxaS) <- colnames(inDF2)[grep('s__',colnames(inDF2))]
  }
  if (length(taxaSCols) > 0) {inDF2 <- inDF2[,-taxaSCols]}
  # pick genera (G)
  taxaGCols <- grep('g__',colnames(inDF2))
  if (length(taxaGCols) > 0) {
    taxaG <- as.data.frame(inDF2[,taxaGCols])
    colnames(taxaG) <- colnames(inDF2)[grep('g__',colnames(inDF2))]
  }
  if (length(taxaGCols) > 0) {inDF2 <- inDF2[,-taxaGCols]}
  # pick families (F)
  taxaFCols <- grep('f__',colnames(inDF2))
  if (length(taxaFCols) > 0) {    
    taxaF <- as.data.frame(inDF2[,taxaFCols])
    colnames(taxaF) <- colnames(inDF2)[grep('f__',colnames(inDF2))]
  }
  if (length(taxaFCols) > 0) {inDF2 <- inDF2[,-taxaFCols]}
  # pick orders (O)
  taxaOCols <- grep('o__',colnames(inDF2))
  if (length(taxaOCols) > 0) {
    taxaO <- as.data.frame(inDF2[,taxaOCols])
    colnames(taxaO) <- colnames(inDF2)[grep('o__',colnames(inDF2))]
  }
  if (length(taxaOCols) > 0) {inDF2 <- inDF2[,-taxaOCols]}
  # pick classes (C)
  taxaCCols <- grep('c__',colnames(inDF2))
  if (length(taxaCCols) > 0) {
    taxaC <- as.data.frame(inDF2[,taxaCCols])
    colnames(taxaC) <- colnames(inDF2)[grep('c__',colnames(inDF2))]
  }
  if (length(taxaCCols) > 0) {
    if( length(colnames(inDF2)) - length(taxaCCols) == 1) {
      ccn <- colnames(inDF2)[grep('c__',colnames(inDF2))]
      tempN <- colnames(inDF2)[!(colnames(inDF2) %in% ccn)]
      inDF2 <- as.data.frame(inDF2[,-taxaCCols])
      colnames(inDF2) <- tempN
    } else {
      inDF2 <- inDF2[,-taxaCCols]
    }
  }
  # pick phyla (P)
  taxaPColsKeepN <- NULL
  taxaPCols <- grep('p__',colnames(inDF2))
  if (length(taxaPCols) > 0) {
    taxaPColsN <- colnames(inDF2)[grep('p__',colnames(inDF2))]
    taxaPColsKeep <- grep('p__',colnames(inDF2),invert = T)
    taxaPColsKeepN <- colnames(inDF2)[grep('p__',colnames(inDF2),invert = T)]
    taxaP <- as.data.frame(inDF2[,taxaPCols])
    colnames(taxaP) <- taxaPColsN
  }
  # pick kingdoms (K)
  if (length(taxaPCols) > 0) {inDF2 <- as.data.frame(inDF2[,taxaPColsKeep])}
  if (!is.null(taxaPColsKeepN)){
    colnames(inDF2) <- taxaPColsKeepN
  }
  taxaK <- inDF2
  # pick 
  oDF <- inDFnonTaxa
  if (verbose) {print ('Keeping following taxonomic levels:'); print(keepLevels)}
  if (ncol(taxaK) > 0) {  
    if ('K' %in% keepLevels) {
      if (verbose){print(paste(' -> kept',ncol(taxaK),'Kingdoms'))}
      if (rescaleTaxa) {taxaK <- taxaK/rowSums(taxaK)}
      oDF <- cbind(oDF,taxaK)}
  }
  if (ncol(taxaP) > 0) {
    if ('P' %in% keepLevels) {
      if (verbose){print(paste(' -> kept',ncol(taxaP),'Phyla'))} 
      if (rescaleTaxa) {taxaP <- taxaP/rowSums(taxaP)}
      oDF <- cbind(oDF,taxaP)}
  }
  if (ncol(taxaC) > 0) {
    if ('C' %in% keepLevels) {
      if (verbose) {print(paste(' -> kept',ncol(taxaC),'Classes'))} 
      if (rescaleTaxa) {taxaC <- taxaC/rowSums(taxaC)}
      oDF <- cbind(oDF,taxaC)}
  }
  if (ncol(taxaO) > 0) {
    if ('O' %in% keepLevels) {
      if (verbose){print(paste(' -> kept',ncol(taxaO),'Orders'))}
      if (rescaleTaxa) {taxaO <- taxaO/rowSums(taxaO)}
      oDF <- cbind(oDF,taxaO)}
  }
  if (ncol(taxaF) > 0) {
    if ('F' %in% keepLevels) {
      if (verbose){print(paste(' -> kept',ncol(taxaF),'Families'))}
      if (rescaleTaxa) {taxaF <- taxaF/rowSums(taxaF)}
      oDF <- cbind(oDF,taxaF)}
  }
  if (ncol(taxaG) > 0) { 
    if ('G' %in% keepLevels) {if (verbose){ print(paste(' -> kept',ncol(taxaG),'Genera'))}
      if (rescaleTaxa) {taxaG <- taxaG/rowSums(taxaG)}
      oDF <- cbind(oDF,taxaG)}
  }
  if (ncol(taxaS) > 0) {
    if ('S' %in% keepLevels) {if (verbose){print(paste(' -> kept',ncol(taxaS),'Species'))}
      if (rescaleTaxa) {taxaS <- taxaS/rowSums(taxaS)}
      oDF <- cbind(oDF,taxaS)}
  }
  if (ncol(taxaT) > 0) {
    if ('T' %in% keepLevels) {if (verbose){print(paste(' -> kept',ncol(taxaT),'Strains'))}
      if (rescaleTaxa) {taxaT <- taxaT/rowSums(taxaT)}
      oDF <- cbind(oDF,taxaT)}
  }
  if (verbose) {print ('data processing done, returning Dataframe')}
  oDF
}
```




# prepping microbiome data

```{r }
df_metaphlan_3_all <- read.csv('/Users/iwan/Research/IBD_risk/local_metaphlan/metaphlan3_lld_base_fup_500fg_300ob_ibd_2713samples_merged_taxa_abundance.tsv', sep ='\t')
df_metadata_lld_IBD_2 <- read.csv('/Users/iwan/Research/IBD_risk/local_metaphlan/IBD_LLD_metadata_filtered_wFEC_v4.csv')
df_metadata_lld_IBD <- readxl::read_xlsx('/Users/iwan/Research/IBD_risk/local_metaphlan/IBD_LLD_metadata_filtered.xlsx')
df_deep_linkage <- read.csv('/Users/iwan/Research/IBD_risk/local_metaphlan/linkage_file_MGS_gpath_2.txt', sep ='\t', header = FALSE)
df_metaphlan_3 <- read.csv('/Users/iwan/Research/IBD_risk/local_metaphlan/Merged_MPAv3.txt', sep='\t')
df_species_with_markers <- read.csv('/Users/iwan/Research/IBD_risk/local_metaphlan/microbiome_risk_species.tsv', sep='\t')
df_deeplinkage_2 <- read.csv('/Users/iwan/Research/IBD_risk/local_metaphlan/linkage_file_MGS.txt', sep=' ', header = F)
df_metadata_IBD_3 <- readxl::read_xlsx('/Users/iwan/Research/IBD_risk/local_metaphlan/Final_Metadata_IBD_V46_clean.xlsx')
## adds an X to colnames



overlap_names <- c()
for (name in colnames(df_metaphlan_3_all)){
  overlap_names <- c(overlap_names, str_split_1(name, '_')[1])
}

fixed_names <- c()
for (name in df_metadata_lld_IBD$IndividualWGSsampleID){
  fixed_names <- c(fixed_names, paste('X', name, sep=''))
}
df_metadata_lld_IBD$updated_sample <- fixed_names


fixed_names_2 <- c()
for (name in df_metadata_lld_IBD_2$IndividualWGSsampleID){
  fixed_names_2 <- c(fixed_names_2, paste('X', name, sep=''))
}
df_metadata_lld_IBD_2$updated_sample <- fixed_names_2

intersect(overlap_names, df_metadata_lld_IBD$IndividualWGSsampleID)
intersect(overlap_names, fixed_names)

intersect(overlap_names, df_metadata_lld_IBD_2$IndividualWGSsampleID)
intersect(overlap_names, fixed_names_2)

for (name in overlap_names){
  print(name)
}

isSingleString <- function(input) {
    is.character(input) & length(input) == 1
}

newnames_strings <- c()
diagnosis <- c()
for (name in overlap_names){
  #print(intersect(name, df_metadata_lld_IBD_2$IndividualWGSsampleID))
  if (isSingleString(intersect(name, df_metadata_lld_IBD_2$IndividualWGSsampleID)) ){
    newnames_strings <- c(newnames_strings, sort(df_metadata_lld_IBD_2$UMCGIBDResearchIDorLLDeepID[df_metadata_lld_IBD_2$IndividualWGSsampleID == name]))
    if (isSingleString(sort(df_metadata_lld_IBD_2$Diagnosis[df_metadata_lld_IBD_2$IndividualWGSsampleID == name])) ){
      diagnosis <- c( diagnosis, sort(df_metadata_lld_IBD_2$Diagnosis[df_metadata_lld_IBD_2$IndividualWGSsampleID == name]))
    } else {
      diagnosis <- c( diagnosis, 'Missing')
    }
    
  } else if (isSingleString(intersect(name, df_metadata_lld_IBD$IndividualWGSsampleID)) ){
    newnames_strings <- c(newnames_strings, sort(df_metadata_lld_IBD$UMCGIBDResearchIDorLLDeepID[df_metadata_lld_IBD$IndividualWGSsampleID == name]))
    diagnosis <- c( diagnosis, sort(df_metadata_lld_IBD$Diagnosis[df_metadata_lld_IBD$IndividualWGSsampleID == name]) )
  } else if (isSingleString(intersect(name, fixed_names)) ){
    newnames_strings <- c(newnames_strings, sort(df_metadata_lld_IBD$UMCGIBDResearchIDorLLDeepID[df_metadata_lld_IBD$updated_sample == name]))
    diagnosis <- c( diagnosis, sort(df_metadata_lld_IBD$Diagnosis[df_metadata_lld_IBD$updated_sample == name]))
  } else if (isSingleString(intersect(name, fixed_names_2)) ){
    newnames_strings <- c(newnames_strings, sort(df_metadata_lld_IBD_2$UMCGIBDResearchIDorLLDeepID[df_metadata_lld_IBD_2$updated_sample == name]))
    diagnosis <- c( diagnosis, sort(df_metadata_lld_IBD_2$Diagnosis[df_metadata_lld_IBD_2$updated_sample == name]))
  } else if (grepl( "IBD", name, fixed = TRUE)) {
    newnames_strings <- c( newnames_strings, name)
    diagnosis <- c( diagnosis, 'IBD') 
  } else {
    newnames_strings <- c( newnames_strings, name)
    diagnosis <- c( diagnosis, 'Missing') 
  }
}


updated_newnames <- c()
for (updated_string in newnames_strings){
  if (isSingleString(df_deep_linkage$V1[df_deep_linkage$V3 == updated_string])){
    updated_newnames <- c(updated_newnames, df_deep_linkage$V1[df_deep_linkage$V3 == updated_string])
  } else {
    updated_newnames <- c(updated_newnames, updated_string)
  }

}

small_list <- c()
for (i in df_deeplinkage_2$V2){
  small_list<- c(small_list, paste('X', i, sep=''))
}


df_deeplinkage_2$V3 <- small_list

another_name_update <- c()
for (newestnames in updated_newnames){
  if (isSingleString(df_deeplinkage_2$V1[df_deeplinkage_2$V3 == newestnames])){
    another_name_update <- c(another_name_update, df_deeplinkage_2$V1[df_deeplinkage_2$V3 == newestnames])
  } else {
    another_name_update <- c(another_name_update, newestnames)
  }
}




#colnames(df_metaphlan_3_all) <- updated_newnames
colnames(df_metaphlan_3_all) <- another_name_update
rownames(df_metaphlan_3_all) <- df_metaphlan_3_all$clade
df_metaphlan_3_all$clade <- NULL
df_metaphlan_3_all$NCBI <- NULL
t_df_metaphlan_3_all <- as.data.frame(t(df_metaphlan_3_all))

df_metaphlan_3_species_all <- filterMetaGenomeDF(t_df_metaphlan_3_all, presPerc = 0.00,minMRelAb = 0.0 ,minMedRelAb=0.0,
                                      rescaleTaxa=T,verbose=T,
                                      keepDomains=c('Bacteria','Archaea'),
                                      keepLevels=c('S'),
                                      keepUnknown=F) 

df_metaphlan_3_species_all$Diagnosis <- diagnosis[3:length(diagnosis)]


colnames(df_metaphlan_3_all)[grepl('LLDeep', colnames(df_metaphlan_3_all))]


shortnames <- c()
for (i in colnames(df_metaphlan_3_species_all)){
  shortnames <- c(shortnames, paste('s__' ,str_split_i(i, '__',8), sep=''))
}
colnames(df_metaphlan_3_species_all) <- shortnames

metaphlan3_species <- df_metaphlan_3_species_all[, df_species_with_markers$Species]

for (col in colnames(metaphlan3_species)){
  metaphlan3_species[,col] <- metaphlan3_species[,col] * as.numeric(df_species_with_markers[df_species_with_markers$Species == col,]$all_gFC)
}
metaphlan3_species$IBD_score <- rowSums(metaphlan3_species)



metaphlan3_species$Diagnosis <- diagnosis[3:length(diagnosis)]
metaphlan3_species$Diagnosis[metaphlan3_species$Diagnosis == 'CD'] <- 'IBD'
metaphlan3_species$Diagnosis[metaphlan3_species$Diagnosis == 'UC'] <- 'IBD'
metaphlan3_species$Diagnosis[metaphlan3_species$Diagnosis == 'IBDU'] <- 'IBD'

metaphlan3_species$Diagnosis[metaphlan3_species$Diagnosis == 'generalpopulation'] <- 'Non-IBD'
metaphlan3_species$Diagnosis[metaphlan3_species$Diagnosis == 'HC'] <- 'Non-IBD'
metaphlan3_species_plot <- metaphlan3_species[metaphlan3_species$Diagnosis != 'Missing',]

p <- ggplot(metaphlan3_species_plot , aes(x=Diagnosis, y=IBD_score, fill=Diagnosis)) +
  geom_jitter(aes(color=Diagnosis), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=Diagnosis), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = Diagnosis), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("combined fold change")+
  ggtitle("FC microbiome features")+ 
  scale_x_discrete(labels=c('IBD, n=399', 'Non-IBD, n=897'  ))

p
#rownames(metaphlan3_species)[metaphlan3_species$Diagnosis == 'Missing'] 
wilcox.test(metaphlan3_species_plot[metaphlan3_species_plot$Diagnosis == 'IBD',]$IBD_score, metaphlan3_species_plot[metaphlan3_species_plot$Diagnosis == 'Non-IBD',]$IBD_score)
#colnames(t_df_metaphlan_3_all) <- t_df_metaphlan_3_all$




rownames(df_metaphlan_3) <- df_metaphlan_3$ID
matching_rows <- intersect(rownames(df_metaphlan_3_all), rownames(df_metaphlan_3))
matching_cols <- intersect(colnames(df_metaphlan_3_all), colnames(df_metaphlan_3))

df_metaphlan_3_all[matching_rows, matching_cols]
df_metaphlan_3[matching_rows, matching_cols]

all.equal(df_metaphlan_3_all[matching_rows, matching_cols],df_metaphlan_3[matching_rows, matching_cols])

i <- 4
for (i in 1:50){
  print(c(df_metaphlan_3[matching_rows[1], matching_cols[i]],
  df_metaphlan_3_all[matching_rows[1], matching_cols[i]]))
}


df_metaphlan_3 <- df_metaphlan_3[, !colnames(df_metaphlan_3) %in% matching_cols]


df_metaphlan_3$rownames <- rownames(df_metaphlan_3)
df_metaphlan_3_all$rownames <- rownames(df_metaphlan_3_all)

all_metaphlan_3_merged <- merge(df_metaphlan_3_all[matching_rows, ], df_metaphlan_3[matching_rows,], by = "rownames", all = TRUE)


rownames(all_metaphlan_3_merged) <- all_metaphlan_3_merged$rownames
all_metaphlan_3_merged$rownames <- NULL



bad_ids <- c('UMCGIBD00169','UMCGIBD00189','UMCGIBD00233','UMCGIBD00491','IBDFEC0096' ,'IBDFEC0308.1','IBDFEC0471.1','IBDFEC0704.1','IBDFEC0950.1')
#all_metaphlan_3_merged$

all_metaphlan_3_merged <- all_metaphlan_3_merged[,!names(all_metaphlan_3_merged) %in% bad_ids]  



list_of_mising_gs_1 <- colnames(df_metaphlan_3_all)[grepl('G8', colnames(df_metaphlan_3_all))]
list_of_mising_gs_2 <- colnames(df_metaphlan_3_all)[grepl('G1', colnames(df_metaphlan_3_all))]
new_ibd_ids <-df_metadata_IBD_3$Feces.II.Project.BROAD

#intersect(new_ibd_ids, list_of_mising_gs_2)
#intersect(new_ibd_ids, list_of_mising_gs_1)

``` 

```{r }

#overlap_names <- c()
#for (name in colnames(all_metaphlan_3_merged)){
#  overlap_names <- c(overlap_names, str_split_1(name, '_')[1])
#}

#fixed_names <- c()
#for (name in df_metadata_lld_IBD$IndividualWGSsampleID){
#  fixed_names <- c(fixed_names, paste('X', name, sep=''))
#}
#df_metadata_lld_IBD$updated_sample <- fixed_names


#fixed_names_2 <- c()
#for (name in df_metadata_lld_IBD_2$IndividualWGSsampleID){
#  fixed_names_2 <- c(fixed_names_2, paste('X', name, sep=''))
#}
#df_metadata_lld_IBD_2$updated_sample <- fixed_names_2

#intersect(overlap_names, df_metadata_lld_IBD$IndividualWGSsampleID)
#intersect(overlap_names, fixed_names)#

#intersect(overlap_names, df_metadata_lld_IBD_2$IndividualWGSsampleID)
#intersect(overlap_names, fixed_names_2)


#isSingleString <- function(input) {
#    is.character(input) & length(input) == 1
#}


newnames_strings <- c()
diagnosis <- c()
for (name in colnames(all_metaphlan_3_merged)){
  #print(intersect(name, df_metadata_lld_IBD_2$IndividualWGSsampleID))
  if (isSingleString(intersect(name, df_metadata_lld_IBD_2$IndividualWGSsampleID)) ){
    newnames_strings <- c(newnames_strings, sort(df_metadata_lld_IBD_2$UMCGIBDResearchIDorLLDeepID[df_metadata_lld_IBD_2$IndividualWGSsampleID == name]))
    if (isSingleString(sort(df_metadata_lld_IBD_2$Diagnosis[df_metadata_lld_IBD_2$IndividualWGSsampleID == name])) ){
      diagnosis <- c( diagnosis, sort(df_metadata_lld_IBD_2$Diagnosis[df_metadata_lld_IBD_2$IndividualWGSsampleID == name]))
    } else {
      diagnosis <- c( diagnosis, 'Missing')
    }

  } else if (isSingleString(intersect(name, df_metadata_lld_IBD$IndividualWGSsampleID)) ){
    newnames_strings <- c(newnames_strings, sort(df_metadata_lld_IBD$UMCGIBDResearchIDorLLDeepID[df_metadata_lld_IBD$IndividualWGSsampleID == name]))
    diagnosis <- c( diagnosis, sort(df_metadata_lld_IBD$Diagnosis[df_metadata_lld_IBD$IndividualWGSsampleID == name]) )
  } else if (isSingleString(intersect(name, fixed_names)) ){
    newnames_strings <- c(newnames_strings, sort(df_metadata_lld_IBD$UMCGIBDResearchIDorLLDeepID[df_metadata_lld_IBD$updated_sample == name]))
    diagnosis <- c( diagnosis, sort(df_metadata_lld_IBD$Diagnosis[df_metadata_lld_IBD$updated_sample == name]))
  } else if (isSingleString(intersect(name, fixed_names_2)) ){
    newnames_strings <- c(newnames_strings, sort(df_metadata_lld_IBD_2$UMCGIBDResearchIDorLLDeepID[df_metadata_lld_IBD_2$updated_sample == name]))
    diagnosis <- c( diagnosis, sort(df_metadata_lld_IBD_2$Diagnosis[df_metadata_lld_IBD_2$updated_sample == name]))
  } else if (isSingleString(intersect(name, df_metadata_lld_IBD$UMCGIBDResearchIDorLLDeepID)) ) {
    newnames_strings <- c( newnames_strings, name)
    diagnosis <- c( diagnosis, sort(df_metadata_lld_IBD$Diagnosis[df_metadata_lld_IBD$UMCGIBDResearchIDorLLDeepID == name]))

  } else if (isSingleString(intersect(name, df_metadata_lld_IBD_2$UMCGIBDResearchIDorLLDeepID)) ) {
    newnames_strings <- c( newnames_strings, name)
    diagnosis <- c( diagnosis, sort(df_metadata_lld_IBD_2$Diagnosis[df_metadata_lld_IBD_2$UMCGIBDResearchIDorLLDeepID == name])) 
    
  } else if (grepl( "IBD", name, fixed = TRUE)) {
    newnames_strings <- c( newnames_strings, name)
    diagnosis <- c( diagnosis, 'IBD') 
  } else {
    newnames_strings <- c( newnames_strings, name)
    diagnosis <- c( diagnosis, 'Missing') 
  }
}


updated_newnames <- c()
for (updated_string in newnames_strings){
  if (isSingleString(df_deep_linkage$V1[df_deep_linkage$V3 == updated_string])){
    updated_newnames <- c(updated_newnames, df_deep_linkage$V1[df_deep_linkage$V3 == updated_string])
  } else {
    updated_newnames <- c(updated_newnames, updated_string)
  }

}

colnames(all_metaphlan_3_merged) <- updated_newnames
rownames(all_metaphlan_3_merged) <- matching_rows



all_metaphlan_3_merged$clade <- NULL
all_metaphlan_3_merged$NCBI <- NULL
t_df_metaphlan_3_all <- as.data.frame(t(all_metaphlan_3_merged))

t_df_metaphlan_3_all = as.data.frame(sapply(t_df_metaphlan_3_all, as.numeric))
df_metaphlan_3_species_all <- filterMetaGenomeDF(t_df_metaphlan_3_all, presPerc = 0.00,minMRelAb = 0.0 ,minMedRelAb=0.0,
                                      rescaleTaxa=T,verbose=T,
                                      keepDomains=c('Bacteria','Archaea'),
                                      keepLevels=c('S'),
                                      keepUnknown=F) 




df_metaphlan_3_species_all$Diagnosis <- diagnosis
#[3:length(diagnosis)]


temp_convert <- as.data.frame(t(df_metaphlan_3_species_all))
colnames(temp_convert) <- updated_newnames
# Identify duplicated column names
duplicate_cols <- duplicated(names(temp_convert))

# Remove columns with duplicated names
temp_convert <- temp_convert[, !duplicate_cols]

df_metaphlan_3_species_all <- as.data.frame(t(temp_convert))

df_metaphlan_3_species_all[,1:709] = as.data.frame(sapply(df_metaphlan_3_species_all[,1:709], as.numeric))


#rownames(df_metaphlan_3_species_all) <- updated_newnames


# Identify columns to remove
#cols_to_remove <- grep("^G", rownames(df_metaphlan_3_species_all), value = TRUE)

# Remove identified columns
#df_metaphlan_3_species_all <- df_metaphlan_3_species_all[!names(df_metaphlan_3_species_all) %in% cols_to_remove, ]


shortnames <- c()
for (i in colnames(df_metaphlan_3_species_all)){
  shortnames <- c(shortnames, paste('s__' ,str_split_i(i, '__',8), sep=''))
}
colnames(df_metaphlan_3_species_all) <- shortnames

df_species_with_markers

fixed_df_species_with_markers <- df_species_with_markers[df_species_with_markers$all_P.value < 0.000000000000001,]
# here we update the possible species list with different options 

fixed_df_species_with_markers <- df_species_with_markers[df_species_with_markers$all_P.value < 0.0001,]
#df_species_with_markers$fdr <- p.adjust(df_species_with_markers$all_P.value )
# 78 FDR significant species
#fixed_df_species_with_markers <- df_species_with_markers[df_species_with_markers$fdr < 0.05,]
# option two, the 31 ish that they mark
# 
# df_metaphlan_3_species_all$s__Asaccharobacter_celatus
# df_metaphlan_3_species_all$s__Eubacterium_hallii
# df_metaphlan_3_species_all$s__Eubacterium_ventriosum
# df_metaphlan_3_species_all$s__Ruminococcus_torques
# df_metaphlan_3_species_all$s__Coprococcus_comes
#   
# df_metaphlan_3_species_all$s__Dorea_longicatena
# df_metaphlan_3_species_all$s__Fusicatenibacter_saccharivorans
# df_metaphlan_3_species_all$s__Lachnospira_pectinoschiza  
# df_metaphlan_3_species_all$s__Oscillibacter_sp_57_20  
# df_metaphlan_3_species_all$s__Faecalibacterium_prausnitzii
# 
# df_metaphlan_3_species_all$s__Gemmiger_formicilis
# df_metaphlan_3_species_all$s__Clostridium_leptum
# df_metaphlan_3_species_all$s__Eubacterium_siraeum
# df_metaphlan_3_species_all$s__Ruminococcus_bromii
# df_metaphlan_3_species_all$s__Firmicutes_bacterium_CAG_110
# 
# df_metaphlan_3_species_all$s__Firmicutes_bacterium_CAG_83
# df_metaphlan_3_species_all$s__Roseburia_inulinivorans
# df_metaphlan_3_species_all$s__Akkermansia_muciniphila
# df_metaphlan_3_species_all$s__Bifidobacterium_adolescentis
# df_metaphlan_3_species_all$s__Eubacterium_sp_CAG_274
# 
# df_metaphlan_3_species_all$s__Holdemania_filiformis
# df_metaphlan_3_species_all$s__Bacteroides_plebeius
# df_metaphlan_3_species_all$s__Bacteroides_xylanisolvens
# df_metaphlan_3_species_all$s__Bilophila_wadsworthia
# df_metaphlan_3_species_all$s__Gordonibacter_pamelaeae
# 
# df_metaphlan_3_species_all$s__Lawsonibacter_asaccharolyticus
# df_metaphlan_3_species_all$s__Eubacterium_eligens
# df_metaphlan_3_species_all$s__Barnesiella_intestinihominis
# df_metaphlan_3_species_all$s__Phascolarctobacterium_faecium
# df_metaphlan_3_species_all$s__Intestinimonas_butyriciproducens
# 
# df_metaphlan_3_species_all$s__Clostridium_sp_CAG_58

species_list_from_paper <- c('s__Asaccharobacter_celatus', 's__Eubacterium_hallii', 's__Eubacterium_ventriosum', 's__Ruminococcus_torques','s__Coprococcus_comes',
                             's__Dorea_longicatena','s__Fusicatenibacter_saccharivorans','s__Lachnospira_pectinoschiza','s__Oscillibacter_sp_57_20','s__Faecalibacterium_prausnitzii',
                             's__Gemmiger_formicilis','s__Clostridium_leptum','s__Eubacterium_siraeum','s__Ruminococcus_bromii','s__Firmicutes_bacterium_CAG_110',
                             's__Firmicutes_bacterium_CAG_83','s__Roseburia_inulinivorans','s__Akkermansia_muciniphila','s__Bifidobacterium_adolescentis','s__Eubacterium_sp_CAG_274',
                             's__Holdemania_filiformis','s__Bacteroides_plebeius','s__Bacteroides_xylanisolvens','s__Bilophila_wadsworthia','s__Gordonibacter_pamelaeae',
                             's__Lawsonibacter_asaccharolyticus','s__Eubacterium_eligens','s__Barnesiella_intestinihominis','s__Phascolarctobacterium_faecium','s__Intestinimonas_butyriciproducens',
                             's__Clostridium_sp_CAG_58')  
  
  
  
metaphlan3_species <- df_metaphlan_3_species_all[, species_list_from_paper]

# now with the significant species
#metaphlan3_species <- df_metaphlan_3_species_all[, fixed_df_species_with_markers$Species]

metaphlan3_species[metaphlan3_species == 0] <- NA

metaphlan3_species[sapply(metaphlan3_species, is.numeric)] <- lapply(metaphlan3_species[sapply(metaphlan3_species, is.numeric)], function(a) ifelse(is.na(a), min(a, na.rm = TRUE)/2, a))

metaphlan3_species_plot <- metaphlan3_species

for (col in colnames(metaphlan3_species)){
  metaphlan3_species[,col] <- log10(metaphlan3_species[,col]) * as.numeric(fixed_df_species_with_markers[fixed_df_species_with_markers$Species == col,]$all_gFC)
}
metaphlan3_species$IBD_score <- rowSums(metaphlan3_species)



metaphlan3_species$Diagnosis <- df_metaphlan_3_species_all$s__NA #[3:length(diagnosis)]
metaphlan3_species$Diagnosis[metaphlan3_species$Diagnosis == 'CD'] <- 'IBD'
metaphlan3_species$Diagnosis[metaphlan3_species$Diagnosis == 'UC'] <- 'IBD'
metaphlan3_species$Diagnosis[metaphlan3_species$Diagnosis == 'IBDU'] <- 'IBD'

for (rowsibd in rownames(metaphlan3_species)){
  if ( grepl('IBD', rowsibd) ){
    
    if (metaphlan3_species[rowsibd, 'Diagnosis'] == "NA"){
      #print(metaphlan3_species[rowsibd, 'Diagnosis'] )
      print(metaphlan3_species[rowsibd, 'Diagnosis'])
      metaphlan3_species[rowsibd, 'Diagnosis'] <- 'IBD'
      
    }
    
  }
} 


metaphlan3_species$Diagnosis[metaphlan3_species$Diagnosis == 'generalpopulation'] <- 'Non-IBD'
metaphlan3_species$Diagnosis[metaphlan3_species$Diagnosis == 'HC'] <- 'Non-IBD'
metaphlan3_species_plot_1 <- metaphlan3_species[metaphlan3_species$Diagnosis != 'Missing',]
metaphlan3_species_plot_2 <- metaphlan3_species_plot_1[metaphlan3_species_plot_1$Diagnosis != 'NA',]
p <- ggplot(metaphlan3_species_plot_2 , aes(x=Diagnosis, y=IBD_score, fill=Diagnosis)) +
  geom_jitter(aes(color=Diagnosis), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=Diagnosis), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = Diagnosis), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("combined fold change")+
  ggtitle("FC microbiome features")+ 
  scale_x_discrete(labels=c('IBD, n=434', 'Non-IBD, n=1134'  ))

pval <- wilcox.test(metaphlan3_species_plot_2[metaphlan3_species_plot_2$Diagnosis == 'IBD',]$IBD_score,  metaphlan3_species_plot_2[metaphlan3_species_plot_2$Diagnosis == 'Non-IBD',]$IBD_score)

p <- p +
  geom_signif(comparisons = list(c("IBD", 'Non-IBD')), annotations = '2.82e-92',
              map_signif_level = TRUE, step_increase=0.1) 
p

pdf('/Users/iwan/Research/IBD_risk/project_output/images/microbiome_score_plot.pdf')

p


dev.off()

```

```{r checking out the weird samples}
# get thedots with the same constant value
weird_df <- metaphlan3_species[metaphlan3_species$IBD_score > 1.0585e-05 ,]
samples_with_weird_ids <- rownames(weird_df[weird_df$IBD_score < 2e-05 ,])
present_weird_samples <- intersect(samples_with_weird_ids, colnames(df_metaphlan_3_all))

weird_samples_df_2 <- df_metaphlan_3_all[,present_weird_samples]
t_weird_samples_df_2 <- as.data.frame(t(weird_samples_df_2))
numeric_training_df <- mutate_all(t_weird_samples_df_2, function(x) as.numeric(as.character(x)))
shannon_training <- diversity(numeric_training_df)
shannon_training
boxplot(shannon_training)


weird_samples_df_3 <- df_metaphlan_3_all[,!names(df_metaphlan_3_all) %in% present_weird_samples]
t_weird_samples_df_3 <- as.data.frame(t(weird_samples_df_3))
numeric_training_df_3 <- mutate_all(t_weird_samples_df_3, function(x) as.numeric(as.character(x)))
shannon_training_3 <- diversity(numeric_training_df_3)
shannon_training_3
boxplot(shannon_training, shannon_training_3)

# alpha diversity is lower but not weirdly low.
pdf('/Users/iwan/Research/IBD_risk/project_output/images/alpha_diversity_weird_samples.pdf')

boxplot(shannon_training, shannon_training_3)


dev.off()
```

```{r Redo microbiome score without weird samples}

metaphlan3_species_plot_3 <- metaphlan3_species_plot_2[!rownames(metaphlan3_species_plot_2) %in% present_weird_samples,]
p <- ggplot(metaphlan3_species_plot_3 , aes(x=Diagnosis, y=IBD_score, fill=Diagnosis)) +
  geom_jitter(aes(color=Diagnosis), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=Diagnosis), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = Diagnosis), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("combined fold change")+
  ggtitle("FC microbiome features without weird samples")+ 
  scale_x_discrete(labels=c('IBD, n=408', 'Non-IBD, n=917'  ))

pval <- wilcox.test(metaphlan3_species_plot_3[metaphlan3_species_plot_3$Diagnosis == 'IBD',]$IBD_score,  metaphlan3_species_plot_3[metaphlan3_species_plot_3$Diagnosis == 'Non-IBD',]$IBD_score)

p <- p +
  geom_signif(comparisons = list(c("IBD", 'Non-IBD')), annotations = '2.12e-25',
              map_signif_level = TRUE, step_increase=0.1) 
p

pdf('/Users/iwan/Research/IBD_risk/project_output/images/microbiome_score_no_weird_samples.pdf')

p

dev.off()
```


```{r get and plot alpha diversity}


df_metaphlan_3_species_all$s__NA <- NULL 


numeric_training_df <- mutate_all(df_metaphlan_3_species_all, function(x) as.numeric(as.character(x)))
shannon_training <- diversity(numeric_training_df)

shannon_df <- as.data.frame(shannon_training)
shannon_df$type <-metaphlan3_species$Diagnosis
shannon_df_plot<- shannon_df[shannon_df$type != 'Missing',]
shannon_df_plot<- shannon_df_plot[shannon_df_plot$shannon_training > 1,]
#shannon_df <- shannon_df[shannon_df$shannon_training < 6 & shannon_df$shannon_training > 1 ,]
shannon_df_plot<-na.omit(shannon_df_plot)
p <- ggplot(shannon_df_plot , aes(x=type, y=shannon_training, fill=type)) +
  geom_jitter(aes(color=type), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=type), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = type), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("Shannon Diversity Index")+
  ggtitle("Shannon diversity in IBD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('IBD, n=432', 'Non-IBD, n=917'  ))


pval <- wilcox.test(shannon_df_plot[shannon_df_plot$type == 'IBD',]$shannon_training,  shannon_df_plot[shannon_df_plot$type == 'Non-IBD',]$shannon_training)


p <- p +
  geom_signif(comparisons = list(c("IBD", 'Non-IBD')), annotations = '2.44e-09',
              map_signif_level = TRUE, step_increase=0.1) 
p



pdf('/Users/iwan/Research/IBD_risk/project_output/images/alpha_diversity_plot.pdf')

p


dev.off()

```

```{r combine into output dataframe}

#metaphlan3_species$IBD_score
#shannon_df$shannon_training
output_df <- shannon_df
output_df$type <- NULL
output_df$IBD_score <- metaphlan3_species$IBD_score
output_df$Microbiome_diagnosis <- metaphlan3_species$Diagnosis
output_df$umcg_id <- rownames(output_df)
output_df<-output_df[output_df$Microbiome_diagnosis != 'Missing',]
write_csv(output_df, '/Users/iwan/Research/IBD_risk/project_output/dataframes/Microbiome_scores_v2.csv', )
#merge(metaphlan3_species, shannon_df)
```





#### now we switch to the PRS setup
```{r loading PRS output}
# 2015 batch

ibd_PRS <- read_table('/Users/iwan/Research/IBD_risk/local_gsa/prs_cs/IBD_2_prs_results.sscore')
LLD_PRS <- read_table('/Users/iwan/Research/IBD_risk/local_gsa/prs_cs/LLD_2_prs_results.sscore')
linkage <- read_table('/Users/iwan/Research/IBD_risk/local_gsa/prs_cs/1000IBD_gsa_linkage.tsv')
gsa_linkage <- read_csv('/Users/iwan/Research/IBD_risk/local_gsa/prs_cs/cytosnp_linkage_file_v2_OV23_00813.csv')
gsa_linkage_2 <- read_csv('/Users/iwan/Research/IBD_risk/local_gsa/prs_cs/deep_linkage_file_v2_OV23_00813.csv')

# match to ids
ibd_gens_ids <- str_split_i(ibd_PRS$IID,'_', 2)  
ibd_PRS$ibd_gens <- ibd_gens_ids

IBD_genetics_prs <- merge(ibd_PRS, linkage, by.x ='ibd_gens', by.y = '#IID')

lld_merged_1 <- merge(LLD_PRS, gsa_linkage, by.x='IID', by.y='cytosnp_ID')

lld_merged_2 <- merge(lld_merged_1, gsa_linkage_2, by = 'project_pseudo_id')



LLD_merged_3 <- merge(lld_merged_2, df_metadata_lld_IBD, by.x='LLDEEP_ID', by.y='UMCGIBDResearchIDorLLDeepID')

IBD_genetics_prs$Diagnosis <- 'IBD'

lld_prs_df <- LLD_merged_3[,c('SCORE1_SUM', 'Diagnosis', 'LLDEEP_ID')]
ibd_prs_df <- IBD_genetics_prs[,c('SCORE1_SUM', 'Diagnosis', 'Research')]
colnames(ibd_prs_df) <- c('PRS_score', 'Diagnosis', 'research_id')
colnames(lld_prs_df) <- c('PRS_score', 'Diagnosis', 'research_id')


combined_prs_dataframe <- rbind(ibd_prs_df, lld_prs_df)



```

```{r plotting prs}

combined_prs_dataframe$Diagnosis <- factor(combined_prs_dataframe$Diagnosis, levels = c("IBD", "generalpopulation"))

p <- ggplot(combined_prs_dataframe , aes(x=Diagnosis, y=PRS_score, fill=Diagnosis)) +
  geom_jitter(aes(color=Diagnosis), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=Diagnosis), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = Diagnosis), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("PRS sum score")+
  ggtitle("PRS score in IBD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('IBD, n=1071', 'general population, n=1304'  ))


pval <- wilcox.test(combined_prs_dataframe[combined_prs_dataframe$Diagnosis == 'IBD',]$PRS_score,  combined_prs_dataframe[combined_prs_dataframe$Diagnosis == 'generalpopulation',]$PRS_score)


p <- p +
  geom_signif(comparisons = list(c("IBD", 'generalpopulation')), annotations = '3.35e-101',
              map_signif_level = TRUE, step_increase=0.1) 

p


pdf('/Users/iwan/Research/IBD_risk/project_output/images/prs_2015_plot.pdf')

p


dev.off()

```

```{r }
write_csv(combined_prs_dataframe, '/Users/iwan/Research/IBD_risk/project_output/dataframes/prs_scores.csv', )

```





## here we try to create the prediction model on the training set
```{r # loading the full combined metadatafile}

# selecting samples with full info
updated_full_train_data <-  read_csv('/Users/iwan/Research/IBD_risk/project_output/dataframes/train_metafull.csv')
updated_full_train_data$...1 <- NULL


```







```{r redo plots with updated_metadata}


table(updated_full_train_data$base_IBDconf)

prs_plot_df <- na.omit(updated_full_train_data[,c('PRS_score', 'base_IBDconf')])
microbiome_score_plot_df<- na.omit(updated_full_train_data[,c('microbiome_score', 'base_IBDconf', 'A_ID_LLDUMCGIBD')])
shannon_plot_df <- na.omit(updated_full_train_data[,c('microbiome_shannon', 'base_IBDconf')]) 

prs_plot_df_test <- updated_full_train_data[,c('PRS_score', 'base_IBDconf','A_ID_LLDUMCGIBD')]

```




```{r  prs score}

library(ggsignif)
prs_plot_df$base_IBDconf[prs_plot_df$base_IBDconf == '0'] <- 'GeneralPopulation'
prs_plot_df$base_IBDconf[prs_plot_df$base_IBDconf == '1'] <- 'IBD'

table(prs_plot_df$base_IBDconf)

p <- ggplot(prs_plot_df , aes(x=base_IBDconf, y=PRS_score, fill=base_IBDconf)) +
  geom_jitter(aes(color=base_IBDconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_IBDconf), alpha=0, width=0.3, linewidth=0.8)+
  geom_violin(aes(color = base_IBDconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("PRS sum score")+
  ggtitle("PRS score in IBD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('general population, n=1252', 'IBD, n=494'  ))


pval <- wilcox.test(prs_plot_df$PRS_score[prs_plot_df$base_IBDconf == 'IBD'],  prs_plot_df$PRS_score[prs_plot_df$base_IBDconf == 'GeneralPopulation'])


p <- p +
  geom_signif(comparisons = list(c("IBD", 'GeneralPopulation')), annotations = '1.27e-66',
              map_signif_level = TRUE, step_increase=0.1) 

p
```


```{r  microbiome score}

microbiome_score_plot_df$base_IBDconf[microbiome_score_plot_df$base_IBDconf == '0'] <- 'GeneralPopulation'
microbiome_score_plot_df$base_IBDconf[microbiome_score_plot_df$base_IBDconf == '1'] <- 'IBD'

table(microbiome_score_plot_df$base_IBDconf)

p <- ggplot(microbiome_score_plot_df , aes(x=base_IBDconf, y=microbiome_score, fill=base_IBDconf)) +
  geom_jitter(aes(color=base_IBDconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_IBDconf), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = base_IBDconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("Microbiome score")+
  ggtitle("Microbiome score in IBD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('general population, n=874', 'IBD, n=439'  ))


pval <- wilcox.test(microbiome_score_plot_df[microbiome_score_plot_df$base_IBDconf == 'IBD',]$microbiome_score,  microbiome_score_plot_df[microbiome_score_plot_df$base_IBDconf == 'GeneralPopulation',]$microbiome_score)


p <- p +
  geom_signif(comparisons = list(c("IBD", 'GeneralPopulation')), annotations = '6.94e-31',
              map_signif_level = TRUE, step_increase=0.1) 

p
```


```{r  shannon score}

shannon_plot_df$base_IBDconf[shannon_plot_df$base_IBDconf == '0'] <- 'GeneralPopulation'
shannon_plot_df$base_IBDconf[shannon_plot_df$base_IBDconf == '1'] <- 'IBD'
shannon_plot_df_plot <- shannon_plot_df[shannon_plot_df$microbiome_shannon > 1,]
table(shannon_plot_df_plot$base_IBDconf)

p <- ggplot(shannon_plot_df_plot , aes(x=base_IBDconf, y=microbiome_shannon, fill=base_IBDconf)) +
  geom_jitter(aes(color=base_IBDconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_IBDconf), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = base_IBDconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("Shannon diversity")+
  ggtitle("alpha diversity in IBD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('general population, n=873', 'IBD, n=437'  ))


pval <- wilcox.test(shannon_plot_df[shannon_plot_df$base_IBDconf == 'IBD',]$microbiome_shannon,  shannon_plot_df[shannon_plot_df$base_IBDconf == 'GeneralPopulation',]$microbiome_shannon)


p <- p +
  geom_signif(comparisons = list(c("IBD", 'GeneralPopulation')), annotations = '1.92e-09',
              map_signif_level = TRUE, step_increase=0.1) 

p

```

```{r redo the weird sample? probbly not }


rownames(microbiome_score_plot_df) <-  microbiome_score_plot_df$A_ID_LLDUMCGIBD
microbiome_score_plot_df_v2 <- microbiome_score_plot_df[!microbiome_score_plot_df$A_ID_LLDUMCGIBD %in% present_weird_samples,]
table(microbiome_score_plot_df_v2$base_IBDconf)

p <- ggplot(microbiome_score_plot_df_v2 , aes(x=base_IBDconf, y=microbiome_score, fill=base_IBDconf)) +
  geom_jitter(aes(color=base_IBDconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_IBDconf), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = base_IBDconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("combined fold change")+
  ggtitle("FC microbiome features without weird samples")+ 
  scale_x_discrete(labels=c('Non-IBD, n=873', 'IBD, n=413'  ))

pval <- wilcox.test(microbiome_score_plot_df_v2[microbiome_score_plot_df_v2$base_IBDconf == 'IBD',]$microbiome_score,  microbiome_score_plot_df_v2[microbiome_score_plot_df_v2$base_IBDconf == 'GeneralPopulation',]$microbiome_score)

p <- p +
  geom_signif(comparisons = list(c("IBD", 'GeneralPopulation')), annotations = '1.13e-24',
              map_signif_level = TRUE, step_increase=0.1) 
p
```






```{r model training and testing}
# split test and train
model_dataframe <- na.omit(updated_full_train_data[, c('microbiome_score', 'PRS_score', 'diet_LLDS', 'envscore_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBDconf')])

trainIndex <- sample(1:nrow(model_dataframe), 0.75 * nrow(model_dataframe))

# setup and train
formula <- 'base_IBDconf ~microbiome_score + PRS_score + diet_LLDS + envscore_u15_sel_weigh + base_AgeFecalDiet + base_sex'
clinical_model <- glm(formula, data=model_dataframe[trainIndex,], family='gaussian')

x_test <- model_dataframe[-trainIndex,c('microbiome_score', 'PRS_score', 'diet_LLDS', 'envscore_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex')]
AUC_clincical <- pROC::auc(pROC::roc(as.data.frame(model_dataframe[-trainIndex, 'base_IBDconf'])$base_IBDconf, predict(clinical_model, x_test, quiet=T)))
    #cat('Clinical features auc = ', micro_ratio_model_AUC_clincical)
plot(pROC::roc(as.data.frame(model_dataframe[-trainIndex, 'base_IBDconf'])$base_IBDconf, predict(clinical_model, x_test, quiet=T)))

# Load required libraries
library(pROC)
library(ggplot2)

# Calculate the ROC
roc_obj <- pROC::roc(as.data.frame(model_dataframe[-trainIndex, 'base_IBDconf'])$base_IBDconf, 
                     predict(clinical_model, x_test, quiet = TRUE))

# Extract ROC data for ggplot
roc_data <- data.frame(
  tpr = roc_obj$sensitivities,  # True Positive Rate (Sensitivity)
  fpr = 1 - roc_obj$specificities,  # False Positive Rate (1 - Specificity)
  thresholds = roc_obj$thresholds
)

# Calculate AUC
auc_value <- round(pROC::auc(roc_obj), 3)

# Plot ROC curve using ggplot2
ggplot(roc_data, aes(x = fpr, y = tpr)) +
  geom_line(color = "#2c7bb6", size = 1.2) +  # ROC curve
  geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "gray") +  # Diagonal line
  theme_minimal() +  # Clean theme
  labs(title = "ROC Curve for IBD risk training Model",
       subtitle = paste("AUC =", auc_value),
       x = "False Positive Rate (1 - Specificity)",
       y = "True Positive Rate (Sensitivity)") +
  theme(plot.title = element_text(size = 16, face = "bold"),
        plot.subtitle = element_text(size = 14),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.text = element_text(size = 10)) +
  coord_equal()  # Equal scaling for axes

# test

# plot


```



## dag3 data prep
```{r dag3 microbiome}

df_dag3_microbiome <- read.csv('/Users/iwan/Research/IBD_risk/local_metaphlan/DMP_metaphlan3_raw.csv', sep=',')

rownames(df_dag3_microbiome) <- df_dag3_microbiome$ID

df_dag3_microbiome$ID  <- NULL

sample_names <- rownames(df_dag3_microbiome)

#t_df_dag3_microbiome = as.data.frame(sapply(as.data.frame(t(df_dag3_microbiome)), as.numeric))
df_dag3_microbiome_species_all <- filterMetaGenomeDF(df_dag3_microbiome, presPerc = 0.00,minMRelAb = 0.0 ,minMedRelAb=0.0,
                                      rescaleTaxa=T,verbose=T,
                                      keepDomains=c('Bacteria','Archaea'),
                                      keepLevels=c('S'),
                                      keepUnknown=F) 





#temp_convert <- as.data.frame(t(df_metaphlan_3_species_all))
#colnames(temp_convert) <- updated_newnames
# Identify duplicated column names
#duplicate_cols <- duplicated(names(temp_convert))

# Remove columns with duplicated names
#temp_convert <- temp_convert[, !duplicate_cols]

#df_metaphlan_3_species_all <- as.data.frame(t(temp_convert))

#df_metaphlan_3_species_all[,1:709] = as.data.frame(sapply(df_metaphlan_3_species_all[,1:709], as.numeric))


shortnames <- c()
for (i in colnames(df_dag3_microbiome_species_all)){
  shortnames <- c(shortnames, paste('s__' ,str_split_i(i, '__',8), sep=''))
}
colnames(df_dag3_microbiome_species_all) <- shortnames

#fixed_df_species_with_markers <- df_species_with_markers[df_species_with_markers$all_P.value < 0.000000000000001,]


metaphlan3_dag3_species <- df_dag3_microbiome_species_all[, species_list_from_paper]


metaphlan3_dag3_species[metaphlan3_dag3_species == 0] <- NA

metaphlan3_dag3_species[sapply(metaphlan3_dag3_species, is.numeric)] <- lapply(metaphlan3_dag3_species[sapply(metaphlan3_dag3_species, is.numeric)], function(a) ifelse(is.na(a), min(a, na.rm = TRUE)/2, a))

metaphlan3_dag3_species_plot <- metaphlan3_dag3_species

for (col in colnames(metaphlan3_dag3_species)){
  metaphlan3_dag3_species[,col] <- log10(metaphlan3_dag3_species[,col]) * as.numeric(fixed_df_species_with_markers[fixed_df_species_with_markers$Species == col,]$all_gFC)
}
metaphlan3_dag3_species$IBD_score <- rowSums(metaphlan3_dag3_species)


#metaphlan3_dag3_species$IBD_score
#rownames(metaphlan3_dag3_species)

dag3_output_dataframe <- as.data.frame(metaphlan3_dag3_species[,'IBD_score'])

dag3_output_dataframe$sample_id <- rownames(metaphlan3_dag3_species)
colnames(dag3_output_dataframe) <- c('IBD_score','DAG3_ids')
#write_csv(dag3_output_dataframe, '/Users/iwan/Research/IBD_risk/project_output/dataframes/DAG3_microbiome_scores.csv', )


```


```{r comparison dag3 metaphlan and ibd_lld }

metaphlan3_species_plot
metaphlan3_dag3_species_plot

# Load the necessary libraries
library(ggplot2)
library(tidyr)
library(dplyr)

# Add an identifier column to distinguish between the two datasets
metaphlan3_species_plot$Dataset <- "LLD_IBD"
metaphlan3_dag3_species_plot$Dataset <- "Dag3"

# Combine the two dataframes
combined_df <- bind_rows(metaphlan3_species_plot, metaphlan3_dag3_species_plot)

# Convert the dataframe to long format
long_df <- combined_df %>%
  pivot_longer(cols = -Dataset, names_to = "Species", values_to = "Value")

# Create the boxplot
ggplot(long_df, aes(x = Species, y = Value, fill = Dataset)) +
  geom_boxplot() +
  labs(title = "Comparison of Species Values Between Two Datasets",
       x = "Species",
       y = "Values") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))  # Rotate x-axis labels

```


```{r comparison dag3 metaphlan and ibd_lld }

#Load the necessary libraries
library(ggplot2)
library(tidyr)
library(dplyr)

# Add an identifier column to distinguish between the two datasets
metaphlan3_species$Dataset <- "LLD_IBD"
metaphlan3_dag3_species$Dataset <- "Dag3"

# Combine the two dataframes
combined_df <- bind_rows(metaphlan3_species, metaphlan3_dag3_species)
combined_df$IBD_score <- NULL
combined_df$Diagnosis <- NULL
# Convert the dataframe to long format
long_df <- combined_df %>%
  pivot_longer(cols = -Dataset, names_to = "Species", values_to = "Value")

# Create the boxplot
ggplot(long_df, aes(x = Species, y = Value, fill = Dataset)) +
  geom_boxplot() +
  labs(title = "Comparison of Species Values Between Two Datasets",
       x = "Species",
       y = "Values") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))  # Rotate x-axis labels

```


```{r plot the score difference between cohorts, then try rescaling}
metaphlan3_species$Dataset <- "LLD_IBD"
metaphlan3_dag3_species$Dataset <- "Dag3"

combined_df <- bind_rows(metaphlan3_species, metaphlan3_dag3_species)
#long_df <- combined_df %>%
#  pivot_longer(cols = -Dataset, names_to = "Species", values_to = "Value")
ggplot(combined_df, aes(x = Dataset, y = IBD_score, fill = Dataset)) +
  geom_boxplot() +
  labs(title = "Comparison of Species Values Between Two Datasets",
       x = "Species",
       y = "Values") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))  # Rotate x-axis labels


```
```{r same plot but rescaled}
metaphlan3_species$Dataset <- "LLD_IBD"
metaphlan3_dag3_species$Dataset <- "Dag3"

my_rescale <- function(values){
  corrected_values <- (values - mean(values)) / sd(values)
  return(corrected_values)
}

metaphlan3_species_copy <- metaphlan3_species 
metaphlan3_dag3_species_copy <-metaphlan3_dag3_species  
metaphlan3_species_copy$rescale_score <- rescale(metaphlan3_species_copy$IBD_score, to=c(0,1))
metaphlan3_dag3_species_copy$rescale_score <- rescale(metaphlan3_dag3_species_copy$IBD_score, to=c(0,1))
combined_df <- bind_rows(metaphlan3_species_copy, metaphlan3_dag3_species_copy)
#long_df <- combined_df %>%
#  pivot_longer(cols = -Dataset, names_to = "Species", values_to = "Value")
ggplot(combined_df, aes(x = Dataset, y = rescale_score, fill = Dataset)) +
  geom_boxplot() +
  labs(title = "Rescaled comparison of IBD score Values Between Two Datasets",
       x = "cohorts",
       y = "Values") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))  # Rotate x-axis labels



metaphlan3_species_copy$rescale_score <- my_rescale(metaphlan3_species_copy$IBD_score)
metaphlan3_dag3_species_copy$rescale_score <- my_rescale(metaphlan3_dag3_species_copy$IBD_score)
combined_df <- bind_rows(metaphlan3_species_copy, metaphlan3_dag3_species_copy)
#long_df <- combined_df %>%
#  pivot_longer(cols = -Dataset, names_to = "Species", values_to = "Value")
ggplot(combined_df, aes(x = Dataset, y = rescale_score, fill = Dataset)) +
  geom_boxplot() +
  labs(title = "My rescaled comparison of IBD score Values Between Two Datasets",
       x = "cohorts",
       y = "Values") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))  # Rotate x-axis labels



```

```{r DAG3 Genetics}
prs_plot_df_copy <- prs_plot_df
prs_plot_df_copy$base_IBDconf <- 'LLD_1000IBD'

dag3_genetics_df <- read.csv('/Users/iwan/Research/IBD_risk/local_gsa/UGLI_prs_results.sscore', sep='\t')

dag3_genetics_df_plot <- dag3_genetics_df[, c('IID', 'SCORE1_SUM')]
colnames(dag3_genetics_df_plot) <- c('base_IBDconf', 'PRS_score')
dag3_genetics_df_plot$base_IBDconf <- 'DAG3'


combined_df <- bind_rows(prs_plot_df_copy, dag3_genetics_df_plot)

ggplot(combined_df, aes(x = base_IBDconf, y = PRS_score, fill = base_IBDconf)) +
  geom_boxplot() +
  labs(title = "Comparison of PRS score Values Between Two Datasets",
       x = "cohorts",
       y = "Values") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))  # Rotate x-axis labels

```


```{r DAG3 rescaled Genetics}
prs_plot_df_copy <- prs_plot_df
prs_plot_df_copy$base_IBDconf <- 'LLD_1000IBD'
prs_plot_df_copy$PRS_score <- rescale(prs_plot_df_copy$PRS_score, to=c(0,1))

dag3_genetics_df <- read.csv('/Users/iwan/Research/IBD_risk/local_gsa/UGLI_prs_results.sscore', sep='\t')

dag3_genetics_df_plot <- dag3_genetics_df[, c('IID', 'SCORE1_SUM')]
colnames(dag3_genetics_df_plot) <- c('base_IBDconf', 'PRS_score')
dag3_genetics_df_plot$base_IBDconf <- 'DAG3'
dag3_genetics_df_plot$PRS_score <- rescale(dag3_genetics_df_plot$PRS_score, to=c(0,1))

combined_df <- bind_rows(prs_plot_df_copy, dag3_genetics_df_plot)

ggplot(combined_df, aes(x = base_IBDconf, y = PRS_score, fill = base_IBDconf)) +
  geom_boxplot() +
  labs(title = "Rescaled comparison of PRS score Values Between Two Datasets",
       x = "cohorts",
       y = "Values") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))  # Rotate x-axis labels


```

```{r PRS output}
dag3_genetics_df <- read.csv('/Users/iwan/Research/IBD_risk/local_gsa/UGLI_prs_results.sscore', sep='\t')


dag3_output_df <- dag3_genetics_df[, c('IID','SCORE1_SUM')]
colnames(dag3_output_df) <- c('research_id','PRS_score')
dag3_output_df$research_id <- str_split_i(dag3_output_df$research_id, '_', 1)
write_csv(dag3_output_df, '/Users/iwan/Research/IBD_risk/project_output/dataframes/DAG3_PRS_2015.csv', )
```


# here we rescale the variables and rerun the model
```{r model training and testing}
library(scales)
# split test and train
model_dataframe <- na.omit(updated_full_train_data[, c('microbiome_score', 'PRS_score', 'diet_LLDS', 'envscore_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBDconf')])

# rescale every factor
model_dataframe$microbiome_score <- rescale(model_dataframe$microbiome_score, to=c(0,1))
model_dataframe$PRS_score <- rescale(model_dataframe$PRS_score, to=c(0,1))
model_dataframe$diet_LLDS <- rescale(model_dataframe$diet_LLDS, to=c(0,1))
model_dataframe$envscore_u15_sel_weigh <- rescale(model_dataframe$envscore_u15_sel_weigh, to=c(0,1))
model_dataframe$base_AgeFecalDiet <- rescale(model_dataframe$base_AgeFecalDiet, to=c(0,1))



trainIndex <- sample(1:nrow(model_dataframe), 0.75 * nrow(model_dataframe))

# setup and train
formula <- 'base_IBDconf ~microbiome_score + PRS_score + diet_LLDS + envscore_u15_sel_weigh + base_AgeFecalDiet + base_sex'
clinical_model <- glm(formula, data=model_dataframe[trainIndex,], family='gaussian')

x_test <- model_dataframe[-trainIndex,c('microbiome_score', 'PRS_score', 'diet_LLDS', 'envscore_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex')]
AUC_clincical <- pROC::auc(pROC::roc(as.data.frame(model_dataframe[-trainIndex, 'base_IBDconf'])$base_IBDconf, predict(clinical_model, x_test, quiet=T)))
    #cat('Clinical features auc = ', micro_ratio_model_AUC_clincical)
plot(pROC::roc(as.data.frame(model_dataframe[-trainIndex, 'base_IBDconf'])$base_IBDconf, predict(clinical_model, x_test, quiet=T)))

# Load required libraries
library(pROC)
library(ggplot2)

# Calculate the ROC
roc_obj <- pROC::roc(as.data.frame(model_dataframe[-trainIndex, 'base_IBDconf'])$base_IBDconf, 
                     predict(clinical_model, x_test, quiet = TRUE))

# Extract ROC data for ggplot
roc_data <- data.frame(
  tpr = roc_obj$sensitivities,  # True Positive Rate (Sensitivity)
  fpr = 1 - roc_obj$specificities,  # False Positive Rate (1 - Specificity)
  thresholds = roc_obj$thresholds
)

# Calculate AUC
auc_value <- round(pROC::auc(roc_obj), 3)

# Plot ROC curve using ggplot2
ggplot(roc_data, aes(x = fpr, y = tpr)) +
  geom_line(color = "#2c7bb6", size = 1.2) +  # ROC curve
  geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "gray") +  # Diagonal line
  theme_minimal() +  # Clean theme
  labs(title = "ROC Curve for IBD risk training Model",
       subtitle = paste("AUC =", auc_value),
       x = "False Positive Rate (1 - Specificity)",
       y = "True Positive Rate (Sensitivity)") +
  theme(plot.title = element_text(size = 16, face = "bold"),
        plot.subtitle = element_text(size = 14),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.text = element_text(size = 10)) +
  coord_equal()  # Equal scaling for axes

# test

# plot

summary(clinical_model)


clinical_model_full_dataset <- glm(formula, data=model_dataframe, family='gaussian')
summary(clinical_model_full_dataset)
```

# now we can apply the model to the dag3 data
```{r predicting on dag3 merged data}

Dag3_merged_df <- read.csv('')
dag3_scores <- predict(clinical_model, Dag3_merged_df, quiet=T)
dag3_scores_full_model <- predict(clinical_model_full_dataset, Dag3_merged_df, quiet=T)




```


# proposed output plot using chatgtp
```{r }
# Load necessary libraries
library(ggplot2)
library(dplyr)

# Set seed for reproducibility
set.seed(123)

# Simulate a dataset
n <- 1000  # number of data points
values <- runif(n, min = -1, max = 1)  # simulate values between -1 and 1
diagnosis <- sample(c("Healthy", "Disease"), n, replace = TRUE, prob = c(0.9, 0.1))  # simulate diagnosis variable

# Create a dataframe
data <- data.frame(values, diagnosis)

# Create a smoothed density plot and highlight "Disease" cases
ggplot(data, aes(x = values)) +
  geom_density(aes(y = after_stat(density)), fill = "lightblue", color = "black", alpha = 0.5) +  # Smooth density for all cases
  geom_density(data = filter(data, diagnosis == "Disease"), aes(y = ..density..), 
               fill = "red", color = "black", alpha = 0.7) +  # Smooth density for "Disease" cases
  labs(title = "Smoothed Distribution of Values with Highlighted Disease Cases",
       x = "Values",
       y = "Density") +
  theme_minimal()



```


```{r option 2}
# Load necessary libraries
library(ggplot2)
library(dplyr)

# Set seed for reproducibility
set.seed(123)

# Simulate a dataset
n <- 1000  # number of data points
values <- runif(n, min = -1, max = 1)  # simulate values between -1 and 1
diagnosis <- sample(c("Healthy", "Disease"), n, replace = TRUE, prob = c(0.99, 0.01))  # simulate diagnosis variable

# Create a dataframe
data <- data.frame(values, diagnosis)

# Calculate the overall density for all values
density_data <- density(data$values)

# Create a dataframe for the density values (x and y)
density_df <- data.frame(x = density_data$x, y = density_data$y)

# For "Disease" cases, find the corresponding height on the density curve
disease_cases <- data %>% filter(diagnosis == "Disease")

# Interpolate the y-values (heights) for the "Disease" cases from the density curve
disease_cases$height <- approx(density_df$x, density_df$y, xout = disease_cases$values)$y

# Plot the smooth density for all values and individual lines for "Disease" cases
ggplot() +
  geom_line(data = density_df, aes(x = x, y = y), color = "lightblue", size = 1) +  # Smooth density for all
  geom_segment(data = disease_cases, aes(x = values, xend = values, y = 0, yend = height),
               color = "red", size = 0.8) +  # Vertical lines for "Disease" cases
  labs(title = "Smooth Density with Highlighted Disease Cases",
       x = "Values",
       y = "Density") +
  theme_minimal()


```


## here we run the final model for 2015
```{r # loading the full combined metadatafile}

# selecting samples with full info
updated_full_train_data <-  read_csv('/Users/iwan/Research/IBD_risk/project_output/dataframes/train_metafull_v2.csv')
updated_full_train_data$...1 <- NULL

output_df <- read.csv('/Users/iwan/Research/IBD_risk/project_output/dataframes/Microbiome_scores_v2.csv', )
updated_full_train_data <- merge(updated_full_train_data, output_df, by.x = 'A_ID_LLDUMCGIBD', by.y = 'umcg_id',all.x=T)
#updated_full_train_data$score_microbiome_log <- NULL 
updated_full_train_data$score_microbiome_log <- updated_full_train_data$IBD_score
updated_full_train_data$score_microbiome_shannon <- updated_full_train_data$shannon_training

```







```{r redo plots with updated_metadata}


table(updated_full_train_data$base_IBDconf)


prs_plot_df <- na.omit(updated_full_train_data[,c('score_prs_2015', 'base_IBDconf')])
microbiome_score_plot_df<- updated_full_train_data[,c('score_microbiome_log', 'base_IBDconf', 'A_ID_LLDUMCGIBD', 'base_OKforMetagenomicAnalysis')]
microbiome_score_plot_df$base_OKforMetagenomicAnalysis <- replace_na(microbiome_score_plot_df$base_OKforMetagenomicAnalysis , 'Yes')
microbiome_score_plot_df <- microbiome_score_plot_df[microbiome_score_plot_df$base_OKforMetagenomicAnalysis != 'no',]
microbiome_score_plot_df$base_OKforMetagenomicAnalysis <- NULL
microbiome_score_plot_df <- na.omit(microbiome_score_plot_df)
shannon_plot_df <- na.omit(updated_full_train_data[,c('score_microbiome_shannon', 'base_IBDconf')]) 

shannon_plot_df<- updated_full_train_data[,c('score_microbiome_shannon', 'base_IBDconf', 'base_OKforMetagenomicAnalysis')]
shannon_plot_df$base_OKforMetagenomicAnalysis <- replace_na(shannon_plot_df$base_OKforMetagenomicAnalysis , 'Yes')
shannon_plot_df <- shannon_plot_df[shannon_plot_df$base_OKforMetagenomicAnalysis != 'no',]
shannon_plot_df$base_OKforMetagenomicAnalysis <- NULL
shannon_plot_df <- na.omit(shannon_plot_df)

prs_plot_df_test <- updated_full_train_data[,c('score_prs_2015', 'base_IBDconf','A_ID_LLDUMCGIBD')]

env_plot_df_weighted <- na.omit(updated_full_train_data[,c('score_env_u15_sel_weigh', 'base_IBDconf')])
env_plot_df_unweighted <- na.omit(updated_full_train_data[,c('score_env_u15_sel_unweigh', 'base_IBDconf')])
diet_plot_df <- na.omit(updated_full_train_data[,c('score_diet_LLDS', 'base_IBDconf')])



diet_med_plot_df <- na.omit(updated_full_train_data[,c('score_diet_aMED', 'base_IBDconf')])




# create uc and cd scores then separate in plotting dfs

updated_full_train_data$UCenvscore_sel=log(0.58)*updated_full_train_data$env_currentsmoking+log(0.78)*updated_full_train_data$env_breastfeeding_binomial + log(0.75)*updated_full_train_data$env_petsu15 +log(1.17)*updated_full_train_data$env_urbanicitychild + log(0.39)*updated_full_train_data$env_appendectomy + log(1.28)*updated_full_train_data$env_HCP


updated_full_train_data$CDenvscore_sel=log(1.76)*updated_full_train_data$env_currentsmoking+log(0.71)*updated_full_train_data$env_breastfeeding_binomial + log(0.77)*updated_full_train_data$env_petsu15  + log(0.93)*updated_full_train_data$env_siblings_twoormore+log(1.42)*updated_full_train_data$env_urbanicitychild + log(1.38)*updated_full_train_data$env_Csection + log(1.61)*updated_full_train_data$env_appendectomy + log(1.37)*updated_full_train_data$env_tonsillectomy+ log(1.25)*updated_full_train_data$env_HCP

metaCD=filter(updated_full_train_data, !is.na(updated_full_train_data$base_CDconf))

metaCD=filter(metaCD,!is.na(metaCD$CDenvscore_sel))

median=median(metaCD$env_totalactivityscore, na.rm = T)

metaCD$env_activitycat=ifelse(metaCD$env_totalactivityscore >= median, 1, 0)     # make activity categorical

metaCD$CDenvscore_sel_v2=log(1.76)*metaCD$env_currentsmoking+log(0.63)*metaCD$env_activitycat+log(0.71)*metaCD$env_breastfeeding_binomial +log(0.77)*metaCD$env_petsu15 + log(0.93)*metaCD$env_siblings_twoormore+ log(1.42)*metaCD$env_urbanicitychild +log(1.38)*metaCD$env_Csection + log(1.61)*metaCD$env_appendectomy + log(1.37)*metaCD$env_tonsillectomy+ log(1.25)*metaCD$env_HCP


#updated_full_train_data_v2 <- merge(updated_full_train_data, metaCD_merge, by = 'A_ID_LLDUMCGIBD')
cd_env_plot_df <- na.omit(metaCD[,c('CDenvscore_sel_v2', 'base_CDconf')])
mean(cd_env_plot_df[cd_env_plot_df$base_CDconf == 1, ]$CDenvscore_sel_v2)
uc_env_plot_df <- na.omit(updated_full_train_data[,c('UCenvscore_sel', 'base_IBDconf')])

```




```{r  prs score}

library(ggsignif)
prs_plot_df$base_IBDconf[prs_plot_df$base_IBDconf == '0'] <- 'GeneralPopulation'
prs_plot_df$base_IBDconf[prs_plot_df$base_IBDconf == '1'] <- 'IBD'

table(prs_plot_df$base_IBDconf)

p <- ggplot(prs_plot_df , aes(x=base_IBDconf, y=score_prs_2015, fill=base_IBDconf)) +
  geom_jitter(aes(color=base_IBDconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_IBDconf), alpha=0, width=0.3, linewidth=0.8)+
  geom_violin(aes(color = base_IBDconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=5, face="bold", colour = "black"),
    axis.text.x = element_text(size=5, face="bold", colour = "black"),
    axis.text.y = element_text(size=5, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=5, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  scale_color_manual(values = c("GeneralPopulation" = "#00BFC4", "IBD" = "#f8766D")) + # Define colors for points/lines
  scale_fill_manual(values = c("GeneralPopulation" = "#00BFC4", "IBD" = "#f8766D")) +  # Define colors for the fill (violin/box)

  ylab("PRS sum score")+
  ggtitle("PRS score 2015 in IBD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('general population, n=1252', 'IBD, n=1078'  ))


pval <- wilcox.test(prs_plot_df$score_prs_2015[prs_plot_df$base_IBDconf == 'IBD'],  prs_plot_df$score_prs_2015[prs_plot_df$base_IBDconf == 'GeneralPopulation'])


p <- p +
  geom_signif(comparisons = list(c("IBD", 'GeneralPopulation')), annotations = '2.00e-99',
              map_signif_level = TRUE, step_increase=0.1) 


p_PRS <- p

pdf('/Users/iwan/Research/IBD_risk/project_output/images/PRS2015_boxplot_paper.pdf')
p

dev.off()


```


```{r  microbiome score}

microbiome_score_plot_df$base_IBDconf[microbiome_score_plot_df$base_IBDconf == '0'] <- 'GeneralPopulation'
microbiome_score_plot_df$base_IBDconf[microbiome_score_plot_df$base_IBDconf == '1'] <- 'IBD'

table(microbiome_score_plot_df$base_IBDconf)

p <- ggplot(microbiome_score_plot_df , aes(x=base_IBDconf, y=score_microbiome_log, fill=base_IBDconf)) +
  geom_jitter(aes(color=base_IBDconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_IBDconf), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = base_IBDconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=5, face="bold", colour = "black"),
    axis.text.x = element_text(size=5, face="bold", colour = "black"),
    axis.text.y = element_text(size=5, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=5, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  scale_color_manual(values = c("GeneralPopulation" = "#00BFC4", "IBD" = "#f8766D")) + # Define colors for points/lines
  scale_fill_manual(values = c("GeneralPopulation" = "#00BFC4", "IBD" = "#f8766D")) +  # Define colors for the fill (violin/box)

  ylab("Microbiome score")+
  ggtitle("Microbiome score in IBD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('general population, n=1083', 'IBD, n=432'  ))


pval <- wilcox.test(microbiome_score_plot_df[microbiome_score_plot_df$base_IBDconf == 'IBD',]$score_microbiome_log,  microbiome_score_plot_df[microbiome_score_plot_df$base_IBDconf == 'GeneralPopulation',]$score_microbiome_log)


p <- p +
  geom_signif(comparisons = list(c("IBD", 'GeneralPopulation')), annotations = '3.90e-53',
              map_signif_level = TRUE, step_increase=0.1) 

p_microbiome_score <- p


pdf('/Users/iwan/Research/IBD_risk/project_output/images/Microbiome_Score_boxplot_paper.pdf')
p

dev.off()
```


```{r  shannon score}

shannon_plot_df$base_IBDconf[shannon_plot_df$base_IBDconf == '0'] <- 'GeneralPopulation'
shannon_plot_df$base_IBDconf[shannon_plot_df$base_IBDconf == '1'] <- 'IBD'
shannon_plot_df_plot <- shannon_plot_df #[shannon_plot_df$score_microbiome_shannon > 1,]
table(shannon_plot_df_plot$base_IBDconf)

p <- ggplot(shannon_plot_df_plot , aes(x=base_IBDconf, y=score_microbiome_shannon, fill=base_IBDconf)) +
  geom_jitter(aes(color=base_IBDconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_IBDconf), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = base_IBDconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("Shannon diversity")+
  ggtitle("alpha diversity in IBD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('general population, n=1083', 'IBD, n=432'  ))


pval <- wilcox.test(shannon_plot_df[shannon_plot_df$base_IBDconf == 'IBD',]$score_microbiome_shannon,  shannon_plot_df[shannon_plot_df$base_IBDconf == 'GeneralPopulation',]$score_microbiome_shannon)


p <- p +
  geom_signif(comparisons = list(c("IBD", 'GeneralPopulation')), annotations = '1.35e-11',
              map_signif_level = TRUE, step_increase=0.1) 

p


pdf('/Users/iwan/Research/IBD_risk/project_output/images/Shannon_Score_boxplot_paper.pdf')
p

dev.off()
```

```{r  env weigthed risk plot}

#env_plot_df_weighted <- updated_full_train_data[,c('score_env_u15_sel_weigh', 'base_IBDconf')]


env_plot_df_weighted$base_IBDconf[env_plot_df_weighted$base_IBDconf == '0'] <- 'GeneralPopulation'
env_plot_df_weighted$base_IBDconf[env_plot_df_weighted$base_IBDconf == '1'] <- 'IBD'
table(env_plot_df_weighted$base_IBDconf)

p <- ggplot(env_plot_df_weighted , aes(x=base_IBDconf, y=score_env_u15_sel_weigh, fill=base_IBDconf)) +
  geom_jitter(aes(color=base_IBDconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_IBDconf), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = base_IBDconf), adjust=3, trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=5, face="bold", colour = "black"),
    axis.text.x = element_text(size=5, face="bold", colour = "black"),
    axis.text.y = element_text(size=5, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=5, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  scale_color_manual(values = c("GeneralPopulation" = "#00BFC4", "IBD" = "#f8766D")) + # Define colors for points/lines
  scale_fill_manual(values = c("GeneralPopulation" = "#00BFC4", "IBD" = "#f8766D")) +  # Define colors for the fill (violin/box)
  ylab("Env Risk score")+
  ggtitle("Weighted environmental score in IBD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('general population, n=1219', 'IBD, n=513'  ))


pval <- wilcox.test(env_plot_df_weighted[env_plot_df_weighted$base_IBDconf == 'IBD',]$score_env_u15_sel_weigh,  env_plot_df_weighted[env_plot_df_weighted$base_IBDconf == 'GeneralPopulation',]$score_env_u15_sel_weigh)


p <- p +
  geom_signif(comparisons = list(c("IBD", 'GeneralPopulation')), annotations = '0.13',
              map_signif_level = TRUE, step_increase=0.1) 

p_weighted_env  <- p

p_weighted_env

#pdf('/Users/iwan/Research/IBD_risk/project_output/images/environmental_weighted_Score_boxplot_paper.pdf')
#p

#dev.off()
```



```{r  env unweigthed risk plot}

#env_plot_df_weighted <- updated_full_train_data[,c('score_env_u15_sel_weigh', 'base_IBDconf')]
#env_plot_df_unweighted <- updated_full_train_data[,c('score_env_u15_sel_unweigh', 'base_IBDconf')]
#diet_plot_df <- updated_full_train_data[,c('score_diet_LLDS', 'base_IBDconf')]

env_plot_df_unweighted$base_IBDconf[env_plot_df_unweighted$base_IBDconf == '0'] <- 'GeneralPopulation'
env_plot_df_unweighted$base_IBDconf[env_plot_df_unweighted$base_IBDconf == '1'] <- 'IBD'
table(env_plot_df_unweighted$base_IBDconf)

p <- ggplot(env_plot_df_unweighted , aes(x=base_IBDconf, y=score_env_u15_sel_unweigh, fill=base_IBDconf)) +
  geom_jitter(aes(color=base_IBDconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_IBDconf), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = base_IBDconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=5, face="bold", colour = "black"),
    axis.text.x = element_text(size=5, face="bold", colour = "black"),
    axis.text.y = element_text(size=5, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=5, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("Env Risk score")+
  ggtitle("Unweighted environmental score in IBD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('general population, n=1219', 'IBD, n=513'  ))


pval <- wilcox.test(env_plot_df_unweighted[env_plot_df_unweighted$base_IBDconf == 'IBD',]$score_env_u15_sel_unweigh,  env_plot_df_unweighted[env_plot_df_unweighted$base_IBDconf == 'GeneralPopulation',]$score_env_u15_sel_unweigh)


p <- p +
  geom_signif(comparisons = list(c("IBD", 'GeneralPopulation')), annotations = '0.05',
              map_signif_level = TRUE, step_increase=0.1) 

p

```

```{r  diet plot}
#diet_plot_df <- updated_full_train_data[,c('score_diet_LLDS', 'base_IBDconf')]
#color_scheme <- c("Cases n=68" = "red", "Controls n=5135" = "blue")

diet_plot_df$base_IBDconf[diet_plot_df$base_IBDconf == '0'] <- 'GeneralPopulation'
diet_plot_df$base_IBDconf[diet_plot_df$base_IBDconf == '1'] <- 'IBD'
diet_plot_df <- diet_plot_df[diet_plot_df$score_diet_LLDS > 1,]
table(diet_plot_df$base_IBDconf)


p <- ggplot(diet_plot_df , aes(x=base_IBDconf, y=score_diet_LLDS, fill=base_IBDconf)) +
  geom_jitter(aes(color=base_IBDconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_IBDconf), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = base_IBDconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=5, face="bold", colour = "black"),
    axis.text.x = element_text(size=5, face="bold", colour = "black"),
    axis.text.y = element_text(size=5, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=5, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("Diet score")+
  ggtitle("Lifelines diet score in IBD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('general population, n=1402', 'IBD, n=532'  ))

p <- ggplot(diet_plot_df , aes(x=base_IBDconf, y=score_diet_LLDS, fill=base_IBDconf)) +
  geom_jitter(aes(color=base_IBDconf), alpha=0.4, width=0.1) +
  geom_boxplot(aes(color=base_IBDconf), alpha=0, width=0.3, size=0.8) +
  geom_violin(aes(color = base_IBDconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8) +
  theme_bw() +
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=5, face="bold", colour = "black"),
    axis.text.x = element_text(size=5, face="bold", colour = "black"),
    axis.text.y = element_text(size=5, face="bold", colour = "black"),
    strip.text = element_text(size=5, face="bold", colour = "black")
  ) +
  theme(legend.position = "none") +
  scale_color_manual(values = c("GeneralPopulation" = "#00BFC4", "IBD" = "#f8766D")) + # Define colors for points/lines
  scale_fill_manual(values = c("GeneralPopulation" = "#00BFC4", "IBD" = "#f8766D")) +  # Define colors for the fill (violin/box)
  ylab("Diet score") +
  ggtitle("Lifelines diet score in IBD vs Non-IBD in LLD and 1000IBD") +
  scale_x_discrete(labels=c('General population, n=1402', 'IBD, n=532'))


pval <- wilcox.test(diet_plot_df[diet_plot_df$base_IBDconf == 'IBD',]$score_diet_LLDS,  diet_plot_df[diet_plot_df$base_IBDconf == 'GeneralPopulation',]$score_diet_LLDS)


p <- p +
  geom_signif(comparisons = list(c("IBD", 'GeneralPopulation')), annotations = '4.79e-08',
              map_signif_level = TRUE, step_increase=0.1) 

p_diet <- p
p_diet


pdf('/Users/iwan/Research/IBD_risk/project_output/images/LLdiet_Score_boxplot_paper.pdf')
p

dev.off()
```



```{r Med diet score plot}
diet_med_plot_df$base_IBDconf[diet_med_plot_df$base_IBDconf == '0'] <- 'GeneralPopulation'
diet_med_plot_df$base_IBDconf[diet_med_plot_df$base_IBDconf == '1'] <- 'IBD'
table(diet_med_plot_df$base_IBDconf)

p <- ggplot(diet_med_plot_df , aes(x=base_IBDconf, y=score_diet_aMED, fill=base_IBDconf)) +
  geom_jitter(aes(color=base_IBDconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_IBDconf), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = base_IBDconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("Diet score")+
  ggtitle("Mediterranean diet score in IBD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('general population, n=1402', 'IBD, n=532'  ))


pval <- wilcox.test(diet_med_plot_df[diet_med_plot_df$base_IBDconf == 'IBD',]$score_diet_aMED,  diet_med_plot_df[diet_med_plot_df$base_IBDconf == 'GeneralPopulation',]$score_diet_aMED)


p <- p +
  geom_signif(comparisons = list(c("IBD", 'GeneralPopulation')), annotations = '0.08',
              map_signif_level = TRUE, step_increase=0.1) 

p



pdf('/Users/iwan/Research/IBD_risk/project_output/images/Med_diet_Score_boxplot_paper.pdf')
p

dev.off()
```


## cd uc plots
```{r cd plot}

cd_env_plot_df <- na.omit(metaCD[,c('CDenvscore_sel_v2', 'base_CDconf')])

cd_env_plot_df$base_CDconf[cd_env_plot_df$base_CDconf == '0'] <- 'GeneralPopulation'
cd_env_plot_df$base_CDconf[cd_env_plot_df$base_CDconf == '1'] <- 'CD'
table(cd_env_plot_df$base_CDconf)

p <- ggplot(cd_env_plot_df , aes(x=base_CDconf, y=CDenvscore_sel_v2, fill=base_CDconf)) +
  geom_jitter(aes(color=base_CDconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_CDconf), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = base_CDconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("Environmental risk score")+
  ggtitle("Environmental risk score CD in CD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('CD, n=175', 'GeneralPopulation, n=912'  ))


pval <- wilcox.test(cd_env_plot_df[cd_env_plot_df$base_CDconf == 'CD',]$CDenvscore_sel_v2,  cd_env_plot_df[cd_env_plot_df$base_CDconf == 'GeneralPopulation',]$CDenvscore_sel_v2)


p <- p +
  geom_signif(comparisons = list(c("CD", 'GeneralPopulation')), annotations = '0.17',
              map_signif_level = TRUE, step_increase=0.1) 

p



pdf('/Users/iwan/Research/IBD_risk/project_output/images/cd_env_score_boxplot_paper.pdf')
p

dev.off()
```

```{r combined plots}
p_diet
p_weighted_env
p_microbiome_score
p_PRS

ga<- ggplotGrob(p_diet)
gb<- ggplotGrob(p_weighted_env)
gc<- ggplotGrob(p_microbiome_score)
gd<- ggplotGrob(p_PRS)
grid::grid.newpage()
#grid::grid.draw(rbind(c(ga, gb), c(gc, gd)))
pdf('/Users/iwan/Research/IBD_risk/project_output/images/combined_training_plots_arranged_paper.pdf')

grid.arrange(gb, ga, gc, gd, ncol=2)
dev.off()


pdf('/Users/iwan/Research/IBD_risk/project_output/images/combined_training_plots_paper.pdf')




grid.arrange(p_weighted_env, p_diet, p_microbiome_score, p_PRS,
             layout_matrix = rbind(c(1, 2),
                        c(4, 3)))
dev.off()


```


```{r uc plot}


uc_env_plot_df <- na.omit(updated_full_train_data[,c('UCenvscore_sel', 'base_UCconf')])
uc_env_plot_df$base_UCconf[uc_env_plot_df$base_UCconf == '0'] <- 'GeneralPopulation'
uc_env_plot_df$base_UCconf[uc_env_plot_df$base_UCconf == '1'] <- 'UC'
table(uc_env_plot_df$base_UCconf)

p <- ggplot(uc_env_plot_df , aes(x=base_UCconf, y=UCenvscore_sel, fill=base_UCconf)) +
  geom_jitter(aes(color=base_UCconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_UCconf), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = base_UCconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("Environmental risk score")+
  ggtitle("Environmental risk score UC in UC vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('General Population, n=1217', 'UC, n=227'  ))


pval <- wilcox.test(uc_env_plot_df[uc_env_plot_df$base_UCconf == 'UC',]$UCenvscore_sel,  uc_env_plot_df[uc_env_plot_df$base_UCconf == 'GeneralPopulation',]$UCenvscore_sel)


p <- p +
  geom_signif(comparisons = list(c("UC", 'GeneralPopulation')), annotations = '1.7e-4',
              map_signif_level = TRUE, step_increase=0.1) 

p



pdf('/Users/iwan/Research/IBD_risk/project_output/images/UC_env_Score_boxplot_paper.pdf')
p

dev.off()
```





```{r model training and testing}
# split test and train
set.seed(1234)

model_dataframe <- na.omit(updated_full_train_data[, c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBDconf')])


# here we set rescale parameters for the training data, and then later we choose the same rescaling parameters for the DAG3 cohort
mean_prs <- mean(model_dataframe$score_prs_2015)
sd_prs <- sd(model_dataframe$score_prs_2015)
mean_microbiome <-mean(model_dataframe$score_microbiome_log)
sd_microbiome <- sd(model_dataframe$score_microbiome_log)
mean_diet <- mean(model_dataframe$score_diet_LLDS)
sd_diet <- sd(model_dataframe$score_diet_LLDS)
mean_env <- mean(model_dataframe$score_env_u15_sel_weigh)
sd_env <- sd(model_dataframe$score_env_u15_sel_weigh)
mean_age <- mean(model_dataframe$base_AgeFecalDiet)
sd_age <- sd(model_dataframe$base_AgeFecalDiet)

mean_list <- c(mean_prs, mean_microbiome, mean_diet, mean_env, mean_age)
sd_list <- c(sd_prs, sd_microbiome, sd_diet, sd_env, sd_age)
rownames_list <- c('PRS', 'microbiome', 'diet', 'environment', 'age')
df_scalers <- data.frame(List1 = mean_list, List2 = sd_list)

# Set the row names from the rownames_list
rownames(df_scalers) <- rownames_list
colnames(df_scalers) <- c('Mean', 'SD')
write_csv(df_scalers, '/Users/iwan/Research/IBD_risk/project_output/dataframes/training_set_scalers.csv')

# Apply scalars
model_dataframe$score_prs_2015_scaled <- (model_dataframe$score_prs_2015 - mean_prs) / sd_prs

model_dataframe$score_diet_LLDS_scaled <- (model_dataframe$score_diet_LLDS - mean_diet) / sd_diet

model_dataframe$score_microbiome_log_scaled <- (model_dataframe$score_microbiome_log - mean_microbiome) / sd_microbiome

model_dataframe$score_env_u15_sel_weigh_scaled <- (model_dataframe$score_env_u15_sel_weigh - mean_env) / sd_env

model_dataframe$base_AgeFecalDiet_scaled <- (model_dataframe$base_AgeFecalDiet - mean_age) / sd_age



trainIndex <- sample(1:nrow(model_dataframe), 0.75 * nrow(model_dataframe))

# setup and train
formula <- 'base_IBDconf ~score_microbiome_log_scaled + score_prs_2015_scaled + score_diet_LLDS_scaled + score_env_u15_sel_weigh_scaled + base_AgeFecalDiet_scaled + base_sex'
clinical_model <- glm(formula, data=model_dataframe[trainIndex,], family='gaussian')

x_test <- model_dataframe[-trainIndex,c('score_microbiome_log_scaled', 'score_prs_2015_scaled', 'score_diet_LLDS_scaled', 'score_env_u15_sel_weigh_scaled', 'base_AgeFecalDiet_scaled', 'base_sex')]
AUC_clincical <- pROC::auc(pROC::roc(as.data.frame(model_dataframe[-trainIndex, 'base_IBDconf'])[[1]], predict(clinical_model, x_test, quiet=T)))
    #cat('Clinical features auc = ', micro_ratio_model_AUC_clincical)
plot(pROC::roc(as.data.frame(model_dataframe[-trainIndex, 'base_IBDconf'])[[1]], predict(clinical_model, x_test, quiet=T)))

# Load required libraries
library(pROC)
library(ggplot2)

# Calculate the ROC
roc_obj <- pROC::roc(as.data.frame(model_dataframe[-trainIndex, 'base_IBDconf'])[[1]], 
                     predict(clinical_model, x_test, quiet = TRUE))

auc_full <- round(pROC::auc(roc_obj), 3)
# Extract ROC data for ggplot
roc_data_full <- data.frame(
  tpr = roc_obj$sensitivities,  # True Positive Rate (Sensitivity)
  fpr = 1 - roc_obj$specificities,  # False Positive Rate (1 - Specificity)
  thresholds = roc_obj$thresholds
)


# Calculate the ROC
formula <- 'base_IBDconf ~score_env_u15_sel_weigh_scaled'
env_model <- glm(formula, data=model_dataframe[trainIndex,], family='gaussian')

roc_obj <- pROC::roc(as.data.frame(model_dataframe[-trainIndex, 'base_IBDconf'])[[1]], 
                     predict(env_model, x_test, quiet = TRUE))
auc_env <- round(pROC::auc(roc_obj), 3)
# Extract ROC data for ggplot
roc_data_env <- data.frame(
  tpr = roc_obj$sensitivities,  # True Positive Rate (Sensitivity)
  fpr = 1 - roc_obj$specificities,  # False Positive Rate (1 - Specificity)
  thresholds = roc_obj$thresholds
)


# Calculate the ROC
formula <- 'base_IBDconf ~score_microbiome_log_scaled'
microbiome_model <- glm(formula, data=model_dataframe[trainIndex,], family='gaussian')

roc_obj <- pROC::roc(as.data.frame(model_dataframe[-trainIndex, 'base_IBDconf'])[[1]], 
                     predict(microbiome_model, x_test, quiet = TRUE))
auc_microbiome <- round(pROC::auc(roc_obj), 3)
# Extract ROC data for ggplot
roc_data_microbiome <- data.frame(
  tpr = roc_obj$sensitivities,  # True Positive Rate (Sensitivity)
  fpr = 1 - roc_obj$specificities,  # False Positive Rate (1 - Specificity)
  thresholds = roc_obj$thresholds
)

# Calculate the ROC
formula <- 'base_IBDconf ~score_diet_LLDS_scaled'
diet_model <- glm(formula, data=model_dataframe[trainIndex,], family='gaussian')

roc_obj <- pROC::roc(as.data.frame(model_dataframe[-trainIndex, 'base_IBDconf'])[[1]], 
                     predict(diet_model, x_test, quiet = TRUE))
auc_diet <- round(pROC::auc(roc_obj), 3)
# Extract ROC data for ggplot
roc_data_diet <- data.frame(
  tpr = roc_obj$sensitivities,  # True Positive Rate (Sensitivity)
  fpr = 1 - roc_obj$specificities,  # False Positive Rate (1 - Specificity)
  thresholds = roc_obj$thresholds
)

# Calculate the ROC
formula <- 'base_IBDconf ~score_prs_2015_scaled '
prs_model <- glm(formula, data=model_dataframe[trainIndex,], family='gaussian')

roc_obj <- pROC::roc(as.data.frame(model_dataframe[-trainIndex, 'base_IBDconf'])[[1]], 
                     predict(prs_model, x_test, quiet = TRUE))
auc_prs <- round(pROC::auc(roc_obj), 3)
# Extract ROC data for ggplot
roc_data_prs <- data.frame(
  tpr = roc_obj$sensitivities,  # True Positive Rate (Sensitivity)
  fpr = 1 - roc_obj$specificities,  # False Positive Rate (1 - Specificity)
  thresholds = roc_obj$thresholds
)
# Calculate AUC
auc_value <- round(pROC::auc(roc_obj), 3)


# Add a model column with AUC values in each data frame
roc_data_full$model <- paste("Combined model (AUC =", round(auc_full, 2), ")")
roc_data_env$model <- paste("Environment Score (AUC =", round(auc_env, 2), ")")
roc_data_microbiome$model <- paste("Microbiome Score (AUC =", round(auc_microbiome, 2), ")")
roc_data_diet$model <- paste("Diet Score (AUC =", round(auc_diet, 2), ")")
roc_data_prs$model <- paste("Polygenic Risk Score (AUC =", round(auc_prs, 2), ")")

# Combine the datasets
combined_roc_data <- rbind(roc_data_full, roc_data_env, roc_data_microbiome, roc_data_diet, roc_data_prs)

# Create a vector of AUC values and corresponding model labels
auc_values <- c(auc_full, auc_env, auc_microbiome, auc_diet, auc_prs)
model_labels <- c(
  paste("Combined model (AUC =", round(auc_full, 2), ")"),
  paste("Environment Score (AUC =", round(auc_env, 2), ")"),
  paste("Microbiome Score (AUC =", round(auc_microbiome, 2), ")"),
  paste("Diet Score (AUC =", round(auc_diet, 2), ")"),
  paste("Polygenic Risk Score (AUC =", round(auc_prs, 2), ")")
)

# Sort the labels by AUC in descending order
sorted_model_labels <- model_labels[order(-auc_values)]

# Set the factor levels based on sorted AUC values
combined_roc_data$model <- factor(combined_roc_data$model, levels = sorted_model_labels)


pdf('/Users/iwan/Research/IBD_risk/project_output/images/AUC_training_plot_paper.pdf')

# Plot with the legend
ggplot(combined_roc_data, aes(x = fpr, y = tpr, color = model)) + 
  geom_line(size = 1.2) +  # Plot all ROC curves, color based on model
  geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "gray") +  # Diagonal line
  theme_minimal() +  # Clean theme
  labs(title = "ROC Curve for IBD Risk Training Models",
       subtitle = "Comparison of Multiple Models",
       x = "False Positive Rate (1 - Specificity)",
       y = "True Positive Rate (Sensitivity)",
       color = "Model (AUC)") +  # Add legend title
  theme(plot.title = element_text(size = 16, face = "bold"),
        plot.subtitle = element_text(size = 14),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.text = element_text(size = 10),
        legend.position = c(0.75, 0.25),  # Adjust legend position to bottom-right
        legend.background = element_rect(fill = "white", color = "black"),
        legend.title = element_text(size = 8, face = "bold"),
        legend.text = element_text(size = 6)) +
  coord_equal()  # Equal scaling for axes

dev.off()






# 
# ggplot() +
#   geom_line(data = roc_data_full, aes(x = fpr, y = tpr), color = "#2c7bb6", size = 1.2) +  # ROC curve 1
#   geom_line(data = roc_data_env, aes(x = fpr, y = tpr), color = "#d7191c", size = 1.2) +  # ROC curve 2
#   geom_line(data = roc_data_microbiome, aes(x = fpr, y = tpr), color = "#fdae61", size = 1.2) +  # ROC curve 3
#   geom_line(data = roc_data_diet, aes(x = fpr, y = tpr), color = "#d7191e", size = 1.2) +  # ROC curve 2
#   geom_line(data = roc_data_prs, aes(x = fpr, y = tpr), color = "#fdae61", size = 1.2) +  # ROC curve 3
#   geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "gray") +  # Diagonal line
#   theme_minimal() +  # Clean theme
#   labs(title = "ROC Curve for IBD risk training Model",
#        subtitle = paste("AUC =", auc_value),  # Assuming auc_value is defined somewhere
#        x = "False Positive Rate (1 - Specificity)",
#        y = "True Positive Rate (Sensitivity)") +
#   theme(plot.title = element_text(size = 16, face = "bold"),
#         plot.subtitle = element_text(size = 14),
#         axis.title.x = element_text(size = 12),
#         axis.title.y = element_text(size = 12),
#         axis.text = element_text(size = 10)) +
#   coord_equal()  # Equal scaling for axes
# 
# 
# dev.off()

print(summary(clinical_model))
# test

# plot

formula <- 'base_IBDconf ~score_microbiome_log_scaled + score_prs_2015_scaled + score_diet_LLDS_scaled + score_env_u15_sel_weigh_scaled + base_AgeFecalDiet_scaled + base_sex'
full_model <- glm(formula, data=model_dataframe, family='gaussian')
summary(full_model)


# here we keep testing and adding variables 
formula <- 'base_IBDconf ~score_diet_LLDS_scaled '
full_model <- glm(formula, data=model_dataframe, family='gaussian')
summary(full_model)

# here we keep testing and adding variables 
formula <- 'base_IBDconf ~score_diet_LLDS_scaled +  score_env_u15_sel_weigh_scaled'
full_model <- glm(formula, data=model_dataframe, family='gaussian')
summary(full_model)


# here we keep testing and adding variables 
formula <- 'base_IBDconf ~score_diet_LLDS_scaled +  score_env_u15_sel_weigh_scaled + score_prs_2015_scaled'
full_model <- glm(formula, data=model_dataframe, family='gaussian')
summary(full_model)



# here we keep testing and adding variables 
formula <- 'base_IBDconf ~score_diet_LLDS_scaled +  score_env_u15_sel_weigh_scaled + score_prs_2015_scaled + score_microbiome_log_scaled'
full_model <- glm(formula, data=model_dataframe, family='gaussian')
summary(full_model)



# here we keep testing and adding variables 
formula <- 'base_IBDconf ~score_diet_LLDS_scaled +  score_env_u15_sel_weigh_scaled + score_prs_2015_scaled + score_microbiome_log_scaled+ base_AgeFecalDiet_scaled '
full_model <- glm(formula, data=model_dataframe, family='gaussian')
summary(full_model)
```

### now we load and prepare dag3 data
```{r DAG3 testing}

# load dag3 data
DAG3_dataframe <- as.data.frame(read_csv('/Users/iwan/Research/IBD_risk/project_output/dataframes/DAG3_metafull_v4.csv'))
#dag3_output_dataframe$IBD_score
# micro <- dag3_output_dataframe
# names(micro)[names(micro) == "IBD_score"] <- "score_microbiome_log"
# meta <- DAG3_dataframe
# meta$score_microbiome_log <- NULL
# meta=merge(meta,micro,by.x="A_DAG3_sampleID",by.y="DAG3_ids",all.x=T)
# 
# meta=merge(meta,micro,by.x="A_DAG3_sampleID_additional",by.y="DAG3_ids",all.x=T)
# 
# meta=merge(meta,micro,by.x="A_DAG3_sampleID_duplicate",by.y="DAG3_ids",all.x=T)
# 
# meta$score_microbiome_log=ifelse(is.na(meta$score_microbiome_log.x), meta$score_microbiome_log,meta$score_microbiome_log.x)
# 
# meta$score_microbiome_log=ifelse(is.na(meta$score_microbiome_log), meta$score_microbiome_log.y,meta$score_microbiome_log)
# 
# meta <- select(meta, -c("score_microbiome_log.x","score_microbiome_log.y"))
# 
# 
# DAG3_dataframe <- meta
#write_csv(DAG3_dataframe, '/Users/iwan/Research/IBD_risk/project_output/dataframes/DAG3_metafull_v3.csv') #base_AgeDietEnvironment
#DAG3_dataframe <- as.data.frame(merge(DAG3_dataframe, dag3_output_dataframe, by.x ='A_DAG3_sampleID', by.y='DAG3_ids'))
#DAG3_dataframe$score_microbiome_log <- DAG3_dataframe$IBD_score
DAG3_df_model <- na.omit(DAG3_dataframe[,c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecal', 'base_gender', 'base_IBD1a2a')])# select variables for predicting

colnames(DAG3_df_model) <- c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBD')
# apply scalars
DAG3_df_model$score_prs_2015_scaled <- (DAG3_df_model$score_prs_2015 - mean_prs) / sd_prs

DAG3_df_model$score_diet_LLDS_scaled <- (DAG3_df_model$score_diet_LLDS - mean_diet) / sd_diet

DAG3_df_model$score_microbiome_log_scaled <- (DAG3_df_model$score_microbiome_log - mean_microbiome) / sd_microbiome

DAG3_df_model$score_env_u15_sel_weigh_scaled <- (DAG3_df_model$score_env_u15_sel_weigh - mean_env) / sd_env

DAG3_df_model$base_AgeFecalDiet_scaled <- (DAG3_df_model$base_AgeFecalDiet - mean_age) / sd_age

#c('score_microbiome_log_scaled', 'score_prs_2015_scaled', 'score_diet_LLDS_scaled', 'score_env_u15_sel_weigh_scaled', 'base_AgeFecalDiet_scaled', 'base_sex')

# predict using previous model
DAG3_df_model$predicted_risk_score <- predict(clinical_model, DAG3_df_model, quiet=T)

DAG3_subset_dataframe <- DAG3_dataframe[rownames(DAG3_df_model), ]
DAG3_subset_dataframe$predicted_risk_score <- DAG3_df_model$predicted_risk_score
cases <- DAG3_df_model[DAG3_df_model$base_IBD == 1, 'predicted_risk_score']
controls <- DAG3_df_model[DAG3_df_model$base_IBD == 0, 'predicted_risk_score']

wilcox.test(cases, controls)


high_riskers <- DAG3_subset_dataframe[DAG3_subset_dataframe$predicted_risk_score > (mean(DAG3_subset_dataframe$predicted_risk_score) +  sd(DAG3_subset_dataframe$predicted_risk_score)), ]
low_riskers <- DAG3_subset_dataframe[DAG3_subset_dataframe$predicted_risk_score < (mean(DAG3_subset_dataframe$predicted_risk_score) -  sd(DAG3_subset_dataframe$predicted_risk_score)), ]


high_riskers <- DAG3_subset_dataframe[DAG3_subset_dataframe$predicted_risk_score > quantile(DAG3_subset_dataframe$predicted_risk_score)[4], ]

low_riskers <- DAG3_subset_dataframe[DAG3_subset_dataframe$predicted_risk_score < quantile(DAG3_subset_dataframe$predicted_risk_score)[2], ]
table(high_riskers$base_IBD1a2a)
table(low_riskers$base_IBD1a2a)

# plot code replace data with real output
# Simulate a dataset
#n <- 1000  # number of data points
#values <- runif(n, min = -1, max = 1)  # simulate values between -1 and 1
#diagnosis <- sample(c("Healthy", "Disease"), n, replace = TRUE, prob = c(0.99, 0.01))  # simulate diagnosis variable

# Create a dataframe
#data <- data.frame(values, diagnosis)

# Calculate the overall density for all values
density_data <- density(DAG3_df_model$predicted_risk_score)

# Create a dataframe for the density values (x and y)
density_df <- data.frame(x = density_data$x, y = density_data$y)

# For "Disease" cases, find the corresponding height on the density curve
disease_cases <- DAG3_df_model %>% filter(base_IBD == 1)

# Interpolate the y-values (heights) for the "Disease" cases from the density curve
disease_cases$height <- approx(density_df$x, density_df$y, xout = disease_cases$predicted_risk_score)$y

# Plot the smooth density for all values and individual lines for "Disease" cases
ggplot() +
  geom_line(data = density_df, aes(x = x, y = y), color = "lightblue", size = 1) +  # Smooth density for all
  geom_segment(data = disease_cases, aes(x = predicted_risk_score, xend = predicted_risk_score, y = 0, yend = height),
               color = "red", size = 0.4) +  # Vertical lines for "Disease" cases
  labs(title = "Smooth Density with Highlighted Disease Cases",
       x = "Values",
       y = "Density") +
  theme_minimal()


# now plotted as loose distributions
Dag3_cases <- DAG3_df_model[DAG3_df_model$base_IBD == 1, 'predicted_risk_score']
Dag3_controls <- DAG3_df_model[DAG3_df_model$base_IBD == 0, 'predicted_risk_score']

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep("Cases n=68", length(Dag3_cases)), rep("Controls n=5135", length(Dag3_controls))))
)

#pdf('/Users/iwan/Research/IBD_risk/project_output/images/IBD_risk_casecontrol_paper.pdf')

# # Plot density distributions for both lists
# ggplot(data, aes(x = Value, fill = Group)) +
#   geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
#   labs(title = "IBD risk score over IBD cases and non-IBD controls in DAG3",
#        x = "IBD risk score",
#        y = "Density") +
#   theme_minimal()




median_cases <- median(Dag3_cases, na.rm = TRUE)
median_controls <- median(Dag3_controls, na.rm = TRUE)

# Step 2: Add the distance between the median of Dag3_new and the other two groups

# Step 3: Define consistent color scheme
color_scheme <- c("Cases n=68" = "red", "Controls n=5135" = "blue")


pdf('/Users/iwan/Research/IBD_risk/project_output/images/IBD_risk_casecontrol_v2_paper.pdf')
# Step 4: Plot density distributions with consistent colors for both density and median lines
ggplot(data, aes(x = Value, fill = Group, color = Group)) + 
  geom_density(alpha = 0.5) +  # Overlay the density plots with transparency
  geom_vline(aes(xintercept = median_cases, color = "Cases n=68"), linetype = "dashed", size = 1) +  # Median for Cases
  geom_vline(aes(xintercept = median_controls, color = "Controls n=5135"), linetype = "dashed", size = 1) +  # Median for Controls
  labs(title = "IBD risk score over IBD cases and non-IBD controls in DAG3",
       x = "IBD risk score",
       y = "Density") +
  theme_minimal() +
  
  # Step 5: Use the same color scheme for density plots and median lines
  scale_fill_manual(name = "Group", values = color_scheme) +
  scale_color_manual(name = "Group", values = color_scheme) +
  
  # Step 6: Annotate the distances between the medians
  theme(legend.position = "top")

dev.off()

dag3_p_value <- wilcox.test(Dag3_cases, Dag3_controls)

DAG3_dataframe$base_IBD_baseline <- replace_na(DAG3_dataframe$base_IBD_baseline, 0)

table(DAG3_dataframe$base_IBD_followup - DAG3_dataframe$base_IBD_baseline)

DAG3_dataframe$new_ibd <- DAG3_dataframe$base_IBD_followup - DAG3_dataframe$base_IBD_baselinesafe
  
#DAG3_dataframe[DAG3_dataframe$new_ibd== 1, c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeDietEnvironment', 'base_gender', 'base_IBD')]

new_onset_dag3 <- na.omit(DAG3_dataframe[, c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecal', 'base_gender', 'new_ibd', 'A_project_pseudo_id')])
colnames(new_onset_dag3) <- c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBD', 'A_project_pseudo_id')
# apply scalars
new_onset_dag3$score_prs_2015_scaled <- (new_onset_dag3$score_prs_2015 - mean_prs) / sd_prs

new_onset_dag3$score_diet_LLDS_scaled <- (new_onset_dag3$score_diet_LLDS - mean_diet) / sd_diet

new_onset_dag3$score_microbiome_log_scaled <- (new_onset_dag3$score_microbiome_log - mean_microbiome) / sd_microbiome

new_onset_dag3$score_env_u15_sel_weigh_scaled <- (new_onset_dag3$score_env_u15_sel_weigh - mean_env) / sd_env

new_onset_dag3$base_AgeFecalDiet_scaled <- (new_onset_dag3$base_AgeFecalDiet - mean_age) / sd_age

#c('score_microbiome_log_scaled', 'score_prs_2015_scaled', 'score_diet_LLDS_scaled', 'score_env_u15_sel_weigh_scaled', 'base_AgeFecalDiet_scaled', 'base_sex')

# predict using previous model
new_onset_dag3_complete <- new_onset_dag3[new_onset_dag3$base_IBD == 1, ]


new_onset_dag3_complete$predicted_risk_score <- predict(clinical_model, new_onset_dag3_complete, quiet=T)

#new_onset_dag3_complete$predicted_risk_score



# now plotted as loose distributions
Dag3_cases <- DAG3_df_model[DAG3_df_model$base_IBD == 1, 'predicted_risk_score']
Dag3_controls <- DAG3_df_model[DAG3_df_model$base_IBD == 0, 'predicted_risk_score']
Dag3_new <- new_onset_dag3_complete$predicted_risk_score

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls, Dag3_new),
  Group = factor(c(rep("Cases n=68", length(Dag3_cases)), rep("Controls n=5135", length(Dag3_controls)), rep("New n=9", length(Dag3_new))))
)

median_cases <- median(Dag3_cases, na.rm = TRUE)
median_controls <- median(Dag3_controls, na.rm = TRUE)
median_new <- median(Dag3_new, na.rm = TRUE)

# Step 2: Add the distance between the median of Dag3_new and the other two groups
distance_cases_new <- abs(median_cases - median_new)
distance_controls_new <- abs(median_controls - median_new)
# Step 3: Define consistent color scheme
color_scheme <- c("Cases n=68" = "red", "Controls n=5135" = "blue", "New n=9" = "green")


pdf('/Users/iwan/Research/IBD_risk/project_output/images/IBD_risk_casecontrolnew_v2_paper.pdf')
# Step 4: Plot density distributions with consistent colors for both density and median lines
ggplot(data, aes(x = Value, fill = Group, color = Group)) + 
  geom_density(alpha = 0.5) +  # Overlay the density plots with transparency
  geom_vline(aes(xintercept = median_cases, color = "Cases n=68"), linetype = "dashed", size = 1) +  # Median for Cases
  geom_vline(aes(xintercept = median_controls, color = "Controls n=5135"), linetype = "dashed", size = 1) +  # Median for Controls
  geom_vline(aes(xintercept = median_new, color = "New n=9"), linetype = "dashed", size = 1) +  # Median for New group
  labs(title = "IBD risk score over IBD cases, pre-onset IBD and non-IBD controls in DAG3",
       x = "IBD risk score",
       y = "Density") +
  theme_minimal() +
  
  # Step 5: Use the same color scheme for density plots and median lines
  scale_fill_manual(name = "Group", values = color_scheme) +
  scale_color_manual(name = "Group", values = color_scheme) +
  
  # Step 6: Annotate the distances between the medians
  theme(legend.position = "top")
dev.off()






#pdf('/Users/iwan/Research/IBD_risk/project_output/images/IBD_risk_case_preonset_control_paper.pdf')

# Plot density distributions for both lists
#ggplot(data, aes(x = Value, fill = Group)) +
#  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
#  labs(title = "IBD risk score over IBD cases, pre-onset IBD and non-IBD controls in DAG3",
#       x = "IBD risk score",
#       y = "Density") +
#  theme_minimal()
#dev.off()

# 
# # new onset DAG3 only genetic scores
# Dag3_cases <- DAG3_df_model[DAG3_df_model$base_IBD == 1, 'score_prs_2015']
# Dag3_controls <- DAG3_df_model[DAG3_df_model$base_IBD == 0, 'score_prs_2015']
# Dag3_new <- new_onset_dag3_complete$score_prs_2015
# 
# data <- data.frame(
#   Value = c(Dag3_cases, Dag3_controls, Dag3_new),
#   Group = factor(c(rep("Cases n=68", length(Dag3_cases)), rep("Controls n=5135", length(Dag3_controls)), rep("New n=9", length(Dag3_new))))
# )
# 
# 
# 
# median_cases <- median(Dag3_cases, na.rm = TRUE)
# median_controls <- median(Dag3_controls, na.rm = TRUE)
# median_new <- median(Dag3_new, na.rm = TRUE)
# 
# # Step 2: Add the distance between the median of Dag3_new and the other two groups
# distance_cases_new <- abs(median_cases - median_new)
# distance_controls_new <- abs(median_controls - median_new)
# # Step 3: Define consistent color scheme
# color_scheme <- c("Cases n=68" = "red", "Controls n=5135" = "blue", "New n=9" = "green")
# 
# # Step 4: Plot density distributions with consistent colors for both density and median lines
# ggplot(data, aes(x = Value, fill = Group, color = Group)) + 
#   geom_density(alpha = 0.5) +  # Overlay the density plots with transparency
#   geom_vline(aes(xintercept = median_cases, color = "Cases n=68"), linetype = "dashed", size = 1) +  # Median for Cases
#   geom_vline(aes(xintercept = median_controls, color = "Controls n=5135"), linetype = "dashed", size = 1) +  # Median for Controls
#   geom_vline(aes(xintercept = median_new, color = "New n=9"), linetype = "dashed", size = 1) +  # Median for New group
#   labs(title = "PRS  over IBD cases, pre-onset IBD and non-IBD controls in DAG3",
#        x = "PRS",
#        y = "Density") +
#   theme_minimal() +
#   
#   # Step 5: Use the same color scheme for density plots and median lines
#   scale_fill_manual(name = "Group", values = color_scheme) +
#   scale_color_manual(name = "Group", values = color_scheme) +
#   
#   # Step 6: Annotate the distances between the medians
#   annotate("text", x = median_cases, y = 0.03, label = paste0("Distance median cases to median new: ", round(distance_cases_new, 2)), color = "black", vjust = -1) +
#   annotate("text", x = median_controls, y = 0.04, label = paste0("Distance median controls to median new: ", round(distance_controls_new, 2)), color = "black", vjust = 1) +
#   theme(legend.position = "top")


#pdf('/Users/iwan/Research/IBD_risk/project_output/images/IBD_risk_case_preonset_control_paper.pdf')

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = "PRS over IBD cases, pre-onset IBD and non-IBD controls in DAG3",
       x = "IBD risk score",
       y = "Density") +
  theme_minimal()
#dev.off()

Dag3_cases <- DAG3_df_model[DAG3_df_model$base_IBD == 1, 'score_microbiome_log']
Dag3_controls <- DAG3_df_model[DAG3_df_model$base_IBD == 0, 'score_microbiome_log']
Dag3_new <- new_onset_dag3_complete$score_microbiome_log

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls, Dag3_new),
  Group = factor(c(rep("Cases n=68", length(Dag3_cases)), rep("Controls n=5135", length(Dag3_controls)), rep("New n=9", length(Dag3_new))))
)

#pdf('/Users/iwan/Research/IBD_risk/project_output/images/IBD_risk_case_preonset_control_paper.pdf')

median_cases <- median(Dag3_cases, na.rm = TRUE)
median_controls <- median(Dag3_controls, na.rm = TRUE)
median_new <- median(Dag3_new, na.rm = TRUE)

# Step 2: Add the distance between the median of Dag3_new and the other two groups
distance_cases_new <- abs(median_cases - median_new)
distance_controls_new <- abs(median_controls - median_new)
# Step 3: Define consistent color scheme
color_scheme <- c("Cases n=68" = "red", "Controls n=5135" = "blue", "New n=9" = "green")

pdf('/Users/iwan/Research/IBD_risk/project_output/images/preonset_microbiome_medians.pdf')
# Step 4: Plot density distributions with consistent colors for both density and median lines
ggplot(data, aes(x = Value, fill = Group, color = Group)) + 
  geom_density(alpha = 0.5) +  # Overlay the density plots with transparency
  geom_vline(aes(xintercept = median_cases, color = "Cases n=68"), linetype = "dashed", size = 1) +  # Median for Cases
  geom_vline(aes(xintercept = median_controls, color = "Controls n=5135"), linetype = "dashed", size = 1) +  # Median for Controls
  geom_vline(aes(xintercept = median_new, color = "New n=9"), linetype = "dashed", size = 1) +  # Median for New group
  labs(title = "Microbiome score over IBD cases, pre-onset IBD and non-IBD controls in DAG3",
       x = "Microbiome risk score",
       y = "Density") +
  theme_minimal() +
  
  # Step 5: Use the same color scheme for density plots and median lines
  scale_fill_manual(name = "Group", values = color_scheme) +
  scale_color_manual(name = "Group", values = color_scheme) 
dev.off()
  # Step 6: Annotate the distances between the medians
  #annotate("text", x = median_cases, y = 0.03, label = paste0("Distance median cases to median new: ", round(distance_cases_new, 2)), color = "black", vjust = -1) +
  #annotate("text", x = median_controls, y = 0.04, label = paste0("Distance median controls to median new: ", round(distance_controls_new, 2)), color = "black", vjust = 1) +
  #theme(legend.position = "top")


# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = "Microbiome score over IBD cases, pre-onset IBD and non-IBD controls in DAG3",
       x = "IBD risk score",
       y = "Density") +
  theme_minimal()

Dag3_cases <- DAG3_df_model[DAG3_df_model$base_IBD == 1, 'score_diet_LLDS']
Dag3_controls <- DAG3_df_model[DAG3_df_model$base_IBD == 0, 'score_diet_LLDS']
Dag3_new <- new_onset_dag3_complete$score_diet_LLDS

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls, Dag3_new),
  Group = factor(c(rep("Cases n=68", length(Dag3_cases)), rep("Controls n=5135", length(Dag3_controls)), rep("New n=9", length(Dag3_new))))
)

median_cases <- median(Dag3_cases, na.rm = TRUE)
median_controls <- median(Dag3_controls, na.rm = TRUE)
median_new <- median(Dag3_new, na.rm = TRUE)

# Step 2: Add the distance between the median of Dag3_new and the other two groups
distance_cases_new <- abs(median_cases - median_new)
distance_controls_new <- abs(median_controls - median_new)
# Step 3: Define consistent color scheme
color_scheme <- c("Cases n=68" = "red", "Controls n=5135" = "blue", "New n=9" = "green")

# Step 4: Plot density distributions with consistent colors for both density and median lines
ggplot(data, aes(x = Value, fill = Group, color = Group)) + 
  geom_density(alpha = 0.5) +  # Overlay the density plots with transparency
  geom_vline(aes(xintercept = median_cases, color = "Cases n=68"), linetype = "dashed", size = 1) +  # Median for Cases
  geom_vline(aes(xintercept = median_controls, color = "Controls n=5135"), linetype = "dashed", size = 1) +  # Median for Controls
  geom_vline(aes(xintercept = median_new, color = "New n=9"), linetype = "dashed", size = 1) +  # Median for New group
  labs(title = "env score over IBD cases, pre-onset IBD and non-IBD controls in DAG3",
       x = "IBD risk score",
       y = "Density") +
  theme_minimal() +
  
  # Step 5: Use the same color scheme for density plots and median lines
  scale_fill_manual(name = "Group", values = color_scheme) +
  scale_color_manual(name = "Group", values = color_scheme) +
  
  # Step 6: Annotate the distances between the medians
  annotate("text", x = median_cases, y = 0.03, label = paste0("Distance to New: ", round(distance_cases_new, 2)), color = "red", vjust = -1) +
  annotate("text", x = median_controls, y = 0.03, label = paste0("Distance to New: ", round(distance_controls_new, 2)), color = "blue", vjust = -1) +
  
  theme(legend.position = "top")
#pdf('/Users/iwan/Research/IBD_risk/project_output/images/IBD_risk_case_preonset_control_paper.pdf')

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = "Diet score over IBD cases, pre-onset IBD and non-IBD controls in DAG3",
       x = "IBD risk score",
       y = "Density") +
  theme_minimal()



Dag3_cases <- DAG3_df_model[DAG3_df_model$base_IBD == 1, 'score_env_u15_sel_weigh']
Dag3_controls <- DAG3_df_model[DAG3_df_model$base_IBD == 0, 'score_env_u15_sel_weigh']
Dag3_new <- new_onset_dag3_complete$score_env_u15_sel_weigh

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls, Dag3_new),
  Group = factor(c(rep("Cases n=68", length(Dag3_cases)), rep("Controls n=5135", length(Dag3_controls)), rep("New n=9", length(Dag3_new))))
)

median_cases <- median(Dag3_cases, na.rm = TRUE)
median_controls <- median(Dag3_controls, na.rm = TRUE)
median_new <- median(Dag3_new, na.rm = TRUE)

# Step 2: Add the distance between the median of Dag3_new and the other two groups
distance_cases_new <- abs(median_cases - median_new)
distance_controls_new <- abs(median_controls - median_new)

# Step 3: Plot density distributions with vertical median lines
ggplot(data, aes(x = Value, fill = Group)) + 
  geom_density(alpha = 0.5) +  # Overlay the three density plots with some transparency
  geom_vline(aes(xintercept = median_cases, color = "Cases"), linetype = "dashed", size = 1) +  # Median for Cases
  geom_vline(aes(xintercept = median_controls, color = "Controls"), linetype = "dashed", size = 1) +  # Median for Controls
  geom_vline(aes(xintercept = median_new, color = "New"), linetype = "dashed", size = 1) +  # Median for New group
  labs(title = "env score over IBD cases, pre-onset IBD and non-IBD controls in DAG3",
       x = "IBD risk score",
       y = "Density") +
  theme_minimal() +
  scale_color_manual(name = "Medians", values = c("Cases" = "red", "Controls" = "blue", "New" = "green")) +  # Color for median lines
  theme(legend.position = "top") +
  
  # Step 4: Annotate the distances between the medians
  annotate("text", x = median_cases, y = 0.03, label = paste0("Cases to New: ", round(distance_cases_new, 2)), color = "black", vjust = -1) +
  annotate("text", x = median_controls, y = 0.03, label = paste0("Controls to New: ", round(distance_controls_new, 2)), color = "black", vjust = -1)





wilcox.test(Dag3_cases, Dag3_controls)
wilcox.test(Dag3_cases,Dag3_new)
wilcox.test(Dag3_new, Dag3_controls)


density_data <- density(DAG3_df_model$predicted_risk_score)

# Create a dataframe for the density values (x and y)
density_df <- data.frame(x = density_data$x, y = density_data$y)

# For "Disease" cases, find the corresponding height on the density curve
disease_cases <- DAG3_df_model %>% filter(base_IBD == 1)

new_cases <- new_onset_dag3_complete
new_cases$height <- approx(density_df$x, density_df$y, xout = new_cases$predicted_risk_score)$y
# Interpolate the y-values (heights) for the "Disease" cases from the density curve
disease_cases$height <- approx(density_df$x, density_df$y, xout = disease_cases$predicted_risk_score)$y

# Plot the smooth density for all values and individual lines for "Disease" cases
ggplot() +
  geom_line(data = density_df, aes(x = x, y = y), color = "lightblue", size = 1) +  # Smooth density for all
  geom_segment(data = disease_cases, aes(x = predicted_risk_score, xend = predicted_risk_score, y = 0, yend = height),
               color = "red", size = 0.4) +  # Vertical lines for "Disease" cases
  geom_segment(data = new_cases, aes(x = predicted_risk_score, xend = predicted_risk_score, y = 0, yend = height),
               color = "blue", size = 0.4) +  # Vertical lines for "Disease" cases
  labs(title = "Smooth Density with Highlighted Disease Cases",
       x = "Values",
       y = "Density") +
  theme_minimal()
```


```{r DAG3 other ibd definitions only distribution}

DAG3_df_model <- na.omit(DAG3_dataframe[,c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecal', 'base_gender', 'base_IBD1a2aconf')])# select variables for predicting


colnames(DAG3_df_model) <- c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBD')
# apply scalars
DAG3_df_model$score_prs_2015_scaled <- (DAG3_df_model$score_prs_2015 - mean_prs) / sd_prs

DAG3_df_model$score_diet_LLDS_scaled <- (DAG3_df_model$score_diet_LLDS - mean_diet) / sd_diet

DAG3_df_model$score_microbiome_log_scaled <- (DAG3_df_model$score_microbiome_log - mean_microbiome) / sd_microbiome

DAG3_df_model$score_env_u15_sel_weigh_scaled <- (DAG3_df_model$score_env_u15_sel_weigh - mean_env) / sd_env

DAG3_df_model$base_AgeFecalDiet_scaled <- (DAG3_df_model$base_AgeFecalDiet - mean_age) / sd_age

#c('score_microbiome_log_scaled', 'score_prs_2015_scaled', 'score_diet_LLDS_scaled', 'score_env_u15_sel_weigh_scaled', 'base_AgeFecalDiet_scaled', 'base_sex')

# predict using previous model
DAG3_df_model$predicted_risk_score <- predict(clinical_model, DAG3_df_model, quiet=T)
Dag3_cases <- DAG3_df_model[DAG3_df_model$base_IBD == 1, 'predicted_risk_score']
Dag3_controls <- DAG3_df_model[DAG3_df_model$base_IBD == 0, 'predicted_risk_score']



# plot code replace data with real output
# Simulate a dataset
#n <- 1000  # number of data points
#values <- runif(n, min = -1, max = 1)  # simulate values between -1 and 1
#diagnosis <- sample(c("Healthy", "Disease"), n, replace = TRUE, prob = c(0.99, 0.01))  # simulate diagnosis variable

# Create a dataframe
#data <- data.frame(values, diagnosis)

# Calculate the overall density for all values
density_data <- density(DAG3_df_model$predicted_risk_score)

# Create a dataframe for the density values (x and y)
density_df <- data.frame(x = density_data$x, y = density_data$y)

# For "Disease" cases, find the corresponding height on the density curve
disease_cases <- DAG3_df_model %>% filter(base_IBD == 1)

# Interpolate the y-values (heights) for the "Disease" cases from the density curve
disease_cases$height <- approx(density_df$x, density_df$y, xout = disease_cases$predicted_risk_score)$y

# Plot the smooth density for all values and individual lines for "Disease" cases
ggplot() +
  geom_line(data = density_df, aes(x = x, y = y), color = "lightblue", size = 1) +  # Smooth density for all
  geom_segment(data = disease_cases, aes(x = predicted_risk_score, xend = predicted_risk_score, y = 0, yend = height),
               color = "red", size = 0.4) +  # Vertical lines for "Disease" cases
  labs(title = "Smooth Density with Highlighted Disease Cases",
       x = "Values",
       y = "Density") +
  theme_minimal()


# now plotted as loose distributions
Dag3_cases <- DAG3_df_model[DAG3_df_model$base_IBD == 1, 'predicted_risk_score']
Dag3_controls <- DAG3_df_model[DAG3_df_model$base_IBD == 0, 'predicted_risk_score']

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep("Cases, n=51", length(Dag3_cases)), rep("Controls, n=5135", length(Dag3_controls))))
)

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = "Density Plot of Two Lists",
       x = "Value",
       y = "Density") +
  theme_minimal()



wilcox.test(Dag3_cases, Dag3_controls)

```




#### 2017 prs data
```{r 2017 prs data check }
# 2017 batch

ibd_PRS <- read_table('/Users/iwan/Research/IBD_risk/local_gsa/IBD_2_prs_results_2017.sscore')
LLD_PRS <- read_table('/Users/iwan/Research/IBD_risk/local_gsa/LLD_2_prs_results_2017.sscore')
linkage <- read_table('/Users/iwan/Research/IBD_risk/local_gsa/prs_cs/1000IBD_gsa_linkage.tsv')
gsa_linkage <- read_csv('/Users/iwan/Research/IBD_risk/local_gsa/prs_cs/cytosnp_linkage_file_v2_OV23_00813.csv')
gsa_linkage_2 <- read_csv('/Users/iwan/Research/IBD_risk/local_gsa/prs_cs/deep_linkage_file_v2_OV23_00813.csv')

# match to ids
ibd_gens_ids <- str_split_i(ibd_PRS$IID,'_', 2)  
ibd_PRS$ibd_gens <- ibd_gens_ids

IBD_genetics_prs <- merge(ibd_PRS, linkage, by.x ='ibd_gens', by.y = '#IID')

lld_merged_1 <- merge(LLD_PRS, gsa_linkage, by.x='IID', by.y='cytosnp_ID')

lld_merged_2 <- merge(lld_merged_1, gsa_linkage_2, by = 'project_pseudo_id')



LLD_merged_3 <- merge(lld_merged_2, df_metadata_lld_IBD, by.x='LLDEEP_ID', by.y='UMCGIBDResearchIDorLLDeepID')

IBD_genetics_prs$Diagnosis <- 'IBD'

lld_prs_df <- LLD_merged_3[,c('SCORE1_SUM', 'Diagnosis', 'LLDEEP_ID')]
ibd_prs_df <- IBD_genetics_prs[,c('SCORE1_SUM', 'Diagnosis', 'Research')]
colnames(ibd_prs_df) <- c('PRS_score', 'Diagnosis', 'research_id')
colnames(lld_prs_df) <- c('PRS_score', 'Diagnosis', 'research_id')


combined_prs_dataframe <- rbind(ibd_prs_df, lld_prs_df)



```



```{r plotting prs}

combined_prs_dataframe$Diagnosis <- factor(combined_prs_dataframe$Diagnosis, levels = c("IBD", "generalpopulation"))

p <- ggplot(combined_prs_dataframe , aes(x=Diagnosis, y=PRS_score, fill=Diagnosis)) +
  geom_jitter(aes(color=Diagnosis), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=Diagnosis), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = Diagnosis), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("PRS sum score")+
  ggtitle("PRS score in IBD vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('IBD, n=1071', 'general population, n=1304'  ))


pval <- wilcox.test(combined_prs_dataframe[combined_prs_dataframe$Diagnosis == 'IBD',]$PRS_score,  combined_prs_dataframe[combined_prs_dataframe$Diagnosis == 'generalpopulation',]$PRS_score)


p <- p +
  geom_signif(comparisons = list(c("IBD", 'generalpopulation')), annotations = '0.26',
              map_signif_level = TRUE, step_increase=0.1) 

p



```


### stoma check
```{r stoma check plot}

library(tidyr)
microbiome_stoma_score_plot_df<- updated_full_train_data[,c('score_microbiome_log', 'base_IBDconf', 'A_ID_LLDUMCGIBD', 'base_OKforMetagenomicAnalysis', 'base_CurrentStomaOrPouch', 'base_CurrentStomaOrPouchType')]
microbiome_stoma_score_plot_df$base_OKforMetagenomicAnalysis <- replace_na(microbiome_stoma_score_plot_df$base_OKforMetagenomicAnalysis , 'Yes')
microbiome_stoma_score_plot_df$base_CurrentStomaOrPouch <- replace_na(microbiome_stoma_score_plot_df$base_CurrentStomaOrPouch ,'No')

microbiome_stoma_score_plot_df <- microbiome_stoma_score_plot_df[microbiome_stoma_score_plot_df$base_OKforMetagenomicAnalysis != 'no',]
microbiome_stoma_score_plot_df$base_OKforMetagenomicAnalysis <- NULL


microbiome_stoma_score_plot_df$base_CurrentStomaOrPouchType <- replace_na(microbiome_stoma_score_plot_df$base_CurrentStomaOrPouchType , 'none')

microbiome_stoma_score_plot_df <- na.omit(microbiome_stoma_score_plot_df)
microbiome_stoma_score_plot_df$base_IBDconf[microbiome_stoma_score_plot_df$base_IBDconf == '0'] <- 'GeneralPopulation'
microbiome_stoma_score_plot_df$base_IBDconf[microbiome_stoma_score_plot_df$base_IBDconf == '1'] <- 'IBD'

table(microbiome_stoma_score_plot_df$base_CurrentStomaOrPouch)



p <- ggplot(microbiome_stoma_score_plot_df , aes(x=base_CurrentStomaOrPouchType, y=score_microbiome_log, fill=base_CurrentStomaOrPouchType)) +
  geom_jitter(aes(color=base_CurrentStomaOrPouchType), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_CurrentStomaOrPouchType), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = base_CurrentStomaOrPouchType), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none",
        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("Microbiome score")+
  ggtitle("Microbiome score in stomas vs no stomas in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('colostoma', 'ileostoma', 'no stoma', 'pouch'  ))

p


#pdf('/Users/iwan/Research/IBD_risk/project_output/images/Stoma_boxplot_paper.pdf')
p

#dev.off()

# get a pvalue for no stomas
no_stomas_microbiome <- microbiome_stoma_score_plot_df[microbiome_stoma_score_plot_df$base_CurrentStomaOrPouchType == 'none', ]
ibd_scores_no_stoma <- as.data.frame(na.omit(no_stomas_microbiome[no_stomas_microbiome$base_IBDconf == 1,'score_microbiome_log'] ))
control_scores_no_stoma <- as.data.frame(na.omit(no_stomas_microbiome[no_stomas_microbiome$base_IBDconf == 0,'score_microbiome_log'] ))
stoma_check <- wilcox.test(ibd_scores_no_stoma$`na.omit(no_stomas_microbiome[no_stomas_microbiome$base_IBDconf == 1, "score_microbiome_log"])`, control_scores_no_stoma$`na.omit(no_stomas_microbiome[no_stomas_microbiome$base_IBDconf == 0, "score_microbiome_log"])`)

no_stomas_microbiome_plot <- as.data.frame(na.omit(no_stomas_microbiome[,c('base_IBDconf', 'score_microbiome_log')]))

no_stomas_microbiome_plot$base_IBDconf <- as.factor(no_stomas_microbiome_plot$base_IBDconf)
p <- ggplot(no_stomas_microbiome_plot , aes(x=base_IBDconf, y=score_microbiome_log, fill=base_IBDconf)) +
  geom_jitter(aes(color=base_IBDconf), alpha=0.4, width=0.1)+
  geom_boxplot(aes(color=base_IBDconf), alpha=0, width=0.3, size=0.8)+
  geom_violin(aes(color = base_IBDconf), trim = FALSE, position = position_dodge(0.9), alpha=.10, size=0.8)+
  theme_bw()+
  theme(
    axis.title.x = element_blank(),    
    axis.title.y = element_text(size=14, face="bold", colour = "black"),
    axis.text.x = element_text(size=14, face="bold", colour = "black"),
    axis.text.y = element_text(size=14, face="bold", colour = "black"),
    #plot.title=element_blank(),
    strip.text = element_text(size=14, face="bold", colour = "black"))+
  
  theme(legend.position = "none")+
  ##scale_color_manual(values = c("W2"="#FF0000","M1"="#28B8D5", "M2"="#28B8D5","M3_baby"="#1B7CA5","M6"="#155E8D","M9"="#0F3F74", "M12"="#FF0000", "P12"="#FF0000", "P28"="#FF0000","M3_mom"="#FF0000", "B"="#FF0000" ))+
  #facet_grid(scales = "free") +
  #scale_y_continuous(breaks=c(2.0, 2.5, 3.0, 3.5, 4.0, 4.5, 5.0)) +
  #scale_y_continuous(labels = scales::number_format(accuracy = 0.1, decimal.mark = '.'), breaks=c(0,1,2,3,4))+
  ylab("Microbiome score")+
  ggtitle("Microbiome score in IBD without stomas vs Non-IBD in LLD and 1000IBD")+ 
  scale_x_discrete(labels=c('general population, n=1083', 'IBD, n=385'  ))


p <- p +
  geom_signif(comparisons = list(c("1", '0')), annotations = '3.26e-40',
              map_signif_level = TRUE, step_increase=0.1) 
pdf('/Users/iwan/Research/IBD_risk/project_output/images/Microbiome_Score_no_stomas.pdf')
p

dev.off()
```
### AUC curve of DAG3
```{r auc curve dag3}

#DAG3_df_model$predicted_risk_score <- predict(clinical_model, DAG3_df_model, quiet=T)

DAG3_df_model <- na.omit(DAG3_dataframe[,c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecal', 'base_gender', 'base_IBD1a2a')])# select variables for predicting


colnames(DAG3_df_model) <- c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBD')
# apply scalars
DAG3_df_model$score_prs_2015_scaled <- (DAG3_df_model$score_prs_2015 - mean_prs) / sd_prs

DAG3_df_model$score_diet_LLDS_scaled <- (DAG3_df_model$score_diet_LLDS - mean_diet) / sd_diet

DAG3_df_model$score_microbiome_log_scaled <- (DAG3_df_model$score_microbiome_log - mean_microbiome) / sd_microbiome

DAG3_df_model$score_env_u15_sel_weigh_scaled <- (DAG3_df_model$score_env_u15_sel_weigh - mean_env) / sd_env

DAG3_df_model$base_AgeFecalDiet_scaled <- (DAG3_df_model$base_AgeFecalDiet - mean_age) / sd_age

# Calculate the ROC
roc_obj <- pROC::roc(DAG3_df_model$base_IBD, 
                     predict(clinical_model, DAG3_df_model, quiet = TRUE))

# Extract ROC data for ggplot
roc_data <- data.frame(
  tpr = roc_obj$sensitivities,  # True Positive Rate (Sensitivity)
  fpr = 1 - roc_obj$specificities,  # False Positive Rate (1 - Specificity)
  thresholds = roc_obj$thresholds
)

# Calculate AUC
auc_value <- round(pROC::auc(roc_obj), 3)

# Plot ROC curve using ggplot2
ggplot(roc_data, aes(x = fpr, y = tpr)) +
  geom_line(color = "#2c7bb6", size = 1.2) +  # ROC curve
  geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "gray") +  # Diagonal line
  theme_minimal() +  # Clean theme
  labs(title = "ROC Curve for IBD risk Validation Model",
       subtitle = paste("AUC =", auc_value),
       x = "False Positive Rate (1 - Specificity)",
       y = "True Positive Rate (Sensitivity)") +
  theme(plot.title = element_text(size = 16, face = "bold"),
        plot.subtitle = element_text(size = 14),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.text = element_text(size = 10)) +
  coord_equal()  # Equal scaling for axes

print(summary(clinical_model))




# 
# # Calculate the ROC
x_test <- DAG3_df_model[-trainIndex,c('score_microbiome_log_scaled', 'score_prs_2015_scaled', 'score_diet_LLDS_scaled', 'score_env_u15_sel_weigh_scaled', 'base_AgeFecalDiet_scaled', 'base_sex')]

roc_obj <- pROC::roc(as.data.frame(DAG3_df_model[-trainIndex, 'base_IBD'])[[1]],
                     predict(clinical_model, x_test, quiet = TRUE))

auc_full <- round(pROC::auc(roc_obj), 3)
# Extract ROC data for ggplot
roc_data_full <- data.frame(
  tpr = roc_obj$sensitivities,  # True Positive Rate (Sensitivity)
  fpr = 1 - roc_obj$specificities,  # False Positive Rate (1 - Specificity)
  thresholds = roc_obj$thresholds
)


# Calculate the ROC
formula <- 'base_IBD ~score_env_u15_sel_weigh_scaled'
env_model <- glm(formula, data=DAG3_df_model[trainIndex,], family='gaussian')

roc_obj <- pROC::roc(as.data.frame(DAG3_df_model[-trainIndex, 'base_IBD'])[[1]], 
                     predict(env_model, x_test, quiet = TRUE))
auc_env <- round(pROC::auc(roc_obj), 3)
# Extract ROC data for ggplot
roc_data_env <- data.frame(
  tpr = roc_obj$sensitivities,  # True Positive Rate (Sensitivity)
  fpr = 1 - roc_obj$specificities,  # False Positive Rate (1 - Specificity)
  thresholds = roc_obj$thresholds
)


# Calculate the ROC
formula <- 'base_IBD ~score_microbiome_log_scaled'
microbiome_model <- glm(formula, data=DAG3_df_model[trainIndex,], family='gaussian')

roc_obj <- pROC::roc(as.data.frame(DAG3_df_model[-trainIndex, 'base_IBD'])[[1]], 
                     predict(microbiome_model, x_test, quiet = TRUE))
auc_microbiome <- round(pROC::auc(roc_obj), 3)
# Extract ROC data for ggplot
roc_data_microbiome <- data.frame(
  tpr = roc_obj$sensitivities,  # True Positive Rate (Sensitivity)
  fpr = 1 - roc_obj$specificities,  # False Positive Rate (1 - Specificity)
  thresholds = roc_obj$thresholds
)

# Calculate the ROC
formula <- 'base_IBD ~score_diet_LLDS_scaled'
diet_model <- glm(formula, data=DAG3_df_model[trainIndex,], family='gaussian')

roc_obj <- pROC::roc(as.data.frame(DAG3_df_model[-trainIndex, 'base_IBD'])[[1]], 
                     predict(diet_model, x_test, quiet = TRUE))
auc_diet <- round(pROC::auc(roc_obj), 3)
# Extract ROC data for ggplot
roc_data_diet <- data.frame(
  tpr = roc_obj$sensitivities,  # True Positive Rate (Sensitivity)
  fpr = 1 - roc_obj$specificities,  # False Positive Rate (1 - Specificity)
  thresholds = roc_obj$thresholds
)

# Calculate the ROC
formula <- 'base_IBD ~score_prs_2015_scaled '
prs_model <- glm(formula, data=DAG3_df_model[trainIndex,], family='gaussian')

roc_obj <- pROC::roc(as.data.frame(DAG3_df_model[-trainIndex, 'base_IBD'])[[1]], 
                     predict(prs_model, x_test, quiet = TRUE))
auc_prs <- round(pROC::auc(roc_obj), 3)
# Extract ROC data for ggplot
roc_data_prs <- data.frame(
  tpr = roc_obj$sensitivities,  # True Positive Rate (Sensitivity)
  fpr = 1 - roc_obj$specificities,  # False Positive Rate (1 - Specificity)
  thresholds = roc_obj$thresholds
)
# Calculate AUC
auc_value <- round(pROC::auc(roc_obj), 3)


# Add a model column with AUC values in each data frame
roc_data_full$model <- paste("Combined model (AUC =", round(auc_full, 2), ")")
roc_data_env$model <- paste("Environment Score (AUC =", round(auc_env, 2), ")")
roc_data_microbiome$model <- paste("Microbiome Score (AUC =", round(auc_microbiome, 2), ")")
roc_data_diet$model <- paste("Diet Score (AUC =", round(auc_diet, 2), ")")
roc_data_prs$model <- paste("Polygenic Risk Score (AUC =", round(auc_prs, 2), ")")

# Combine the datasets
combined_roc_data <- rbind(roc_data_full, roc_data_env, roc_data_microbiome, roc_data_diet, roc_data_prs)

# Create a vector of AUC values and corresponding model labels
auc_values <- c(auc_full, auc_env, auc_microbiome, auc_diet, auc_prs)
model_labels <- c(
  paste("Combined model (AUC =", round(auc_full, 2), ")"),
  paste("Environment Score (AUC =", round(auc_env, 2), ")"),
  paste("Microbiome Score (AUC =", round(auc_microbiome, 2), ")"),
  paste("Diet Score (AUC =", round(auc_diet, 2), ")"),
  paste("Polygenic Risk Score (AUC =", round(auc_prs, 2), ")")
)

# Sort the labels by AUC in descending order
sorted_model_labels <- model_labels[order(-auc_values)]

# Set the factor levels based on sorted AUC values
combined_roc_data$model <- factor(combined_roc_data$model, levels = sorted_model_labels)


pdf('/Users/iwan/Research/IBD_risk/project_output/images/AUC_validation_ibd1a2a_plot_paper.pdf')

# Plot with the legend
ggplot(combined_roc_data, aes(x = fpr, y = tpr, color = model)) + 
  geom_line(size = 1.2) +  # Plot all ROC curves, color based on model
  geom_abline(slope = 1, intercept = 0, linetype = "dashed", color = "gray") +  # Diagonal line
  theme_minimal() +  # Clean theme
  labs(title = "ROC Curve for IBD Risk Validation Models",
       subtitle = "Comparison of Multiple Models",
       x = "False Positive Rate (1 - Specificity)",
       y = "True Positive Rate (Sensitivity)",
       color = "Model (AUC)") +  # Add legend title
  theme(plot.title = element_text(size = 16, face = "bold"),
        plot.subtitle = element_text(size = 14),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12),
        axis.text = element_text(size = 10),
        legend.position = c(0.75, 0.25),  # Adjust legend position to bottom-right
        legend.background = element_rect(fill = "white", color = "black"),
        legend.title = element_text(size = 8, face = "bold"),
        legend.text = element_text(size = 6)) +
  coord_equal()  # Equal scaling for axes

dev.off()

```
### individual score curves for the dag3 data
```{r individual score curves}

DAG3_df_model <- na.omit(DAG3_dataframe[,c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecal', 'base_gender', 'base_IBD1a2a')])# select variables for predicting

# diet
# now plotted as loose distributions
Dag3_cases <- DAG3_df_model[DAG3_df_model$base_IBD1a2a == 1, 'score_diet_LLDS']
Dag3_controls <- DAG3_df_model[DAG3_df_model$base_IBD1a2a == 0, 'score_diet_LLDS']

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep("Cases, n=68", length(Dag3_cases)), rep("Controls, n=5135", length(Dag3_controls))))
)
pdf('/Users/iwan/Research/IBD_risk/project_output/images/Diet_Risk_denstiy_DAG3_paper.pdf')

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = "Diet risk score over IBD cases and non-IBD controls in DAG3",
       x = "Diet risk score",
       y = "Density") +
  theme_minimal()

dev.off()
# prs
# now plotted as loose distributions
Dag3_cases <- DAG3_df_model[DAG3_df_model$base_IBD1a2a == 1, 'score_prs_2015']
Dag3_controls <- DAG3_df_model[DAG3_df_model$base_IBD1a2a == 0, 'score_prs_2015']

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep("Cases, n=68", length(Dag3_cases)), rep("Controls, n=5135", length(Dag3_controls))))
)
pdf('/Users/iwan/Research/IBD_risk/project_output/images/PRS_Risk_denstiy_DAG3_paper.pdf')

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = "Polygenic risk score over IBD cases and non-IBD controls in DAG3",
       x = "Polygenic risk score",
       y = "Density") +
  theme_minimal()

dev.off()

# Microbiome
# now plotted as loose distributions
Dag3_cases <- DAG3_df_model[DAG3_df_model$base_IBD1a2a == 1, 'score_microbiome_log']
Dag3_controls <- DAG3_df_model[DAG3_df_model$base_IBD1a2a == 0, 'score_microbiome_log']

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep("Cases, n=68", length(Dag3_cases)), rep("Controls, n=5135", length(Dag3_controls))))
)
pdf('/Users/iwan/Research/IBD_risk/project_output/images/Microbiome_Risk_denstiy_DAG3_paper.pdf')

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = "Microbiome risk score over IBD cases and non-IBD controls in DAG3",
       x = "Microbiome risk score",
       y = "Density") +
  theme_minimal()
dev.off()


# environment
# now plotted as loose distributions
Dag3_cases <- DAG3_df_model[DAG3_df_model$base_IBD1a2a == 1, 'score_env_u15_sel_weigh']
Dag3_controls <- DAG3_df_model[DAG3_df_model$base_IBD1a2a == 0, 'score_env_u15_sel_weigh']

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep("Cases, n=68", length(Dag3_cases)), rep("Controls, n=5135", length(Dag3_controls))))
)
pdf('/Users/iwan/Research/IBD_risk/project_output/images/Environment_Risk_denstiy_DAG3_paper.pdf')

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = "Environment risk score over IBD cases and non-IBD controls in DAG3",
       x = "Environment risk score",
       y = "Density") +
  theme_minimal()
dev.off()

```





### here we use the DAG3 phenotypes to check for disease related to ibd risk score
```{r compare ibd risk score with phenotypes}
# load in phenotypes
dag3_phenos <- read_csv('/Users/iwan/Research/IBD_risk/phenotypes/DAG3_metadata_merged_ready_v27.csv')
dag3_phenos$Rows <- NULL


dag3_scores_phenos <- merge(DAG3_dataframe, dag3_phenos, by.x = 'A_DAG3_sampleID', by.y ='DAG3_sampleID')


#dag3_scores_phenos


DAG3_df_model <- na.omit(dag3_scores_phenos[,c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecal', 'base_gender', 'base_IBD1a2a', 'A_DAG3_sampleID')])# select variables for predicting


colnames(DAG3_df_model) <- c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBD', 'A_DAG3_sampleID')
# apply scalars
DAG3_df_model$score_prs_2015_scaled <- (DAG3_df_model$score_prs_2015 - mean_prs) / sd_prs

DAG3_df_model$score_diet_LLDS_scaled <- (DAG3_df_model$score_diet_LLDS - mean_diet) / sd_diet

DAG3_df_model$score_microbiome_log_scaled <- (DAG3_df_model$score_microbiome_log - mean_microbiome) / sd_microbiome

DAG3_df_model$score_env_u15_sel_weigh_scaled <- (DAG3_df_model$score_env_u15_sel_weigh - mean_env) / sd_env

DAG3_df_model$base_AgeFecalDiet_scaled <- (DAG3_df_model$base_AgeFecalDiet - mean_age) / sd_age

#c('score_microbiome_log_scaled', 'score_prs_2015_scaled', 'score_diet_LLDS_scaled', 'score_env_u15_sel_weigh_scaled', 'base_AgeFecalDiet_scaled', 'base_sex')

# predict using previous model
DAG3_df_model$predicted_risk_score <- predict(clinical_model, DAG3_df_model, quiet=T)

rownames(dag3_scores_phenos) <- dag3_scores_phenos$A_DAG3_sampleID

dag3_scores_phenos_model <- dag3_scores_phenos[DAG3_df_model$A_DAG3_sampleID, ] 
dag3_scores_phenos_model$predicted_risk_score   <- DAG3_df_model$predicted_risk_score
lookup_df <- dag3_scores_phenos_model[, c('predicted_risk_score', 'MED.DISEASES.Gastrointestinal.Autoimmune.IBD.UC', 'MED.DISEASES.Gastrointestinal.Autoimmune.IBD.CD', 'base_IBD1a2a')]


# logistic regression model phenotype ~ risk score
pheno_list<-c()
coef_list<-c()
p_val_list<-c()
for (phenotype in colnames(dag3_scores_phenos_model)[grepl('MED.DISEASES',colnames(dag3_scores_phenos_model))]){
  print(phenotype)
  dag3_scores_phenos_model$outcome <- as.numeric(dag3_scores_phenos_model[,phenotype] == "Y")
  print(table(dag3_scores_phenos_model$outcome))
  formula <- 'outcome ~ predicted_risk_score'
  phenotype_model <- glm(formula, data=dag3_scores_phenos_model, family='binomial')
  model_summary <- summary(phenotype_model)
  print(phenotype)
  pheno_list <- c(pheno_list, phenotype)
  print(model_summary$coefficients[2,1])
  coef_list <- c(coef_list, model_summary$coefficients[2,1])
  print(model_summary$coefficients[2,4])
  p_val_list <- c(p_val_list, model_summary$coefficients[2,4])
}

clinical_phenotypes_df <- data.frame(pheno_list, coef_list, p_val_list)
clinical_phenotypes_df$adjusted_p_value <- p.adjust(clinical_phenotypes_df$p_val_list, method = "BH")

pheno_list<-c()
p_val_list<-c()
for (phenotype in colnames(dag3_scores_phenos_model)[grepl('MED.DISEASES',colnames(dag3_scores_phenos_model))]){
  dag3_scores_phenos_model$outcome <- as.numeric(dag3_scores_phenos_model[,phenotype] == "Y")
  #print(table(dag3_scores_phenos_model$outcome))
  if (length(na.omit(dag3_scores_phenos_model[dag3_scores_phenos_model$outcome==1 ,]$predicted_risk_score )) > 1){
    wilcox_p <- wilcox.test(dag3_scores_phenos_model[dag3_scores_phenos_model$outcome==1 ,]$predicted_risk_score, dag3_scores_phenos_model[dag3_scores_phenos_model$outcome==0 ,]$predicted_risk_score )
    pheno_list <- c(pheno_list, phenotype)
  p_val_list <- c(p_val_list, wilcox_p$p.value)
  }
}

clinical_mann_phenotypes_df <- data.frame(pheno_list, p_val_list)
clinical_mann_phenotypes_df$adjusted_p_value <- p.adjust(clinical_mann_phenotypes_df$p_val_list, method = "BH")



```





### Super healthy people from dag3 
```{r compare ibd risk score with super healthy people}
# load in phenotypes
table(dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases)


# just plot score vs healthy people

Dag3_cases <- dag3_scores_phenos_model[dag3_scores_phenos_model$base_IBD1a2a == 1, 'predicted_risk_score']
Dag3_controls_step1 <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'N',]
Dag3_controls <- Dag3_controls_step1[Dag3_controls_step1$base_IBD1a2a == 0, 'predicted_risk_score']
Dag3_healthy <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'Y', 'predicted_risk_score']
#Dag3_new <- new_onset_dag3_complete$predicted_risk_score

data <- data.frame(
  Value = c(Dag3_cases, Dag3_healthy),
  Group = factor(c(rep("Cases, n=68", length(Dag3_cases)), rep("Healthy, n=865", length(Dag3_healthy))))
)

# # Plot density distributions for both lists
# ggplot(data, aes(x = Value, fill = Group)) +
#   geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
#   labs(title = "Density Plot of Two Lists",
#        x = "Value",
#        y = "Density") +
#   theme_minimal()


median_cases <- median(Dag3_cases, na.rm = TRUE)
median_controls <- median(Dag3_healthy, na.rm = TRUE)

# Step 2: Add the distance between the median of Dag3_new and the other two groups
# Step 3: Define consistent color scheme
color_scheme <- c("Cases, n=68" = "red", "Healthy, n=865" = "#0099CC")


pdf('/Users/iwan/Research/IBD_risk/project_output/images/IBD_risk_casehealthy_v3_paper.pdf')
# Step 4: Plot density distributions with consistent colors for both density and median lines
ggplot(data, aes(x = Value, fill = Group, color = Group)) + 
  geom_density(alpha = 0.5) +  # Overlay the density plots with transparency
  geom_vline(aes(xintercept = median_cases, color = "Cases, n=68"), linetype = "dashed", size = 1) +  # Median for Cases
  geom_vline(aes(xintercept = median_controls, color = "Healthy, n=865"), linetype = "dashed", size = 1) +  # Median for Controls
  labs(title = "IBD risk score over IBD cases and Healthy controls in DAG3",
       x = "IBD risk score",
       y = "Density") +
  theme_minimal() +
  
  # Step 5: Use the same color scheme for density plots and median lines
  scale_fill_manual(name = "Group", values = color_scheme) +
  scale_color_manual(name = "Group", values = color_scheme) +
  scale_x_continuous(limits=c(-1.03,0.54)) +
  # Step 6: Annotate the distances between the medians
  theme(legend.position = "top")
dev.off()
wilcox.test(Dag3_cases, Dag3_healthy)


Dag3_cases_risk_calc <- dag3_scores_phenos_model[dag3_scores_phenos_model$base_IBD1a2a == 1, ]
Dag3_healthy_risk_calc <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'Y', ]
dag3_healthy_comparison <- rbind(Dag3_cases_risk_calc, Dag3_healthy_risk_calc)

#high_riskers <- dag3_healthy_comparison[dag3_healthy_comparison$predicted_risk_score > (mean(dag3_healthy_comparison$predicted_risk_score) +  sd(dag3_healthy_comparison$predicted_risk_score)), ]
#low_riskers <- dag3_healthy_comparison[dag3_healthy_comparison$predicted_risk_score < (mean(dag3_healthy_comparison$predicted_risk_score) -  sd(dag3_healthy_comparison$predicted_risk_score)), ]


high_riskers <- dag3_healthy_comparison[dag3_healthy_comparison$predicted_risk_score > quantile(dag3_healthy_comparison$predicted_risk_score)[4], ]

low_riskers <- dag3_healthy_comparison[dag3_healthy_comparison$predicted_risk_score < quantile(dag3_healthy_comparison$predicted_risk_score)[2], ]
table(high_riskers$MED.DISEASES.None.No.Diseases)
table(low_riskers$MED.DISEASES.None.No.Diseases)




density_data <- density(dag3_scores_phenos_model$predicted_risk_score)

# Create a dataframe for the density values (x and y)
density_df <- data.frame(x = density_data$x, y = density_data$y)

# For "Disease" cases, find the corresponding height on the density curve
disease_cases <- dag3_scores_phenos_model %>% filter(base_IBD1a2a == 1)

healthy_cases <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'Y', ]
healthy_cases$height <- approx(density_df$x, density_df$y, xout = healthy_cases$predicted_risk_score)$y
# Interpolate the y-values (heights) for the "Disease" cases from the density curve
disease_cases$height <- approx(density_df$x, density_df$y, xout = disease_cases$predicted_risk_score)$y

# Plot the smooth density for all values and individual lines for "Disease" cases
ggplot() +
  geom_line(data = density_df, aes(x = x, y = y), color = "lightblue", size = 1) +  # Smooth density for all
  geom_segment(data = healthy_cases, aes(x = predicted_risk_score, xend = predicted_risk_score, y = 0, yend = height),
               color = "lightblue", size = 0.1) +  # Vertical lines for "Disease" cases
  geom_segment(data = disease_cases, aes(x = predicted_risk_score, xend = predicted_risk_score, y = 0, yend = height),
               color = "red", size = 0.4) +  # Vertical lines for "Disease" cases
  
  labs(title = "Smooth Density with Highlighted Disease cases and healthy",
       x = "Values",
       y = "Density") +
  theme_minimal()
```



## here we plot the distribution of the other clinical phenotypes
```{r plotting the output of the glm model}

phenotype <- 'MED.DISEASES.Pulmonary.COPD'
Dag3_cases <- as.numeric(na.omit(dag3_scores_phenos_model[dag3_scores_phenos_model[,phenotype] == 'Y', 'predicted_risk_score']))
#Dag3_controls_step1 <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'N',]
Dag3_controls <- as.numeric(na.omit(Dag3_controls_step1[Dag3_controls_step1[, phenotype] == 'N', 'predicted_risk_score']))
#Dag3_healthy <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'Y', 'predicted_risk_score']
#Dag3_new <- new_onset_dag3_complete$predicted_risk_score

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep(paste("Cases, n=", length(Dag3_cases)), length(Dag3_cases)), rep(paste("Controls, n=", length(Dag3_controls)), length(Dag3_controls))))
)

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = paste("Density Plot of ", phenotype),
       x = "Value",
       y = "Density") +
  theme_minimal() + 
  annotate("text", 
           x = Inf, y = Inf, 
           label = paste('Linear model pval=', round(clinical_phenotypes_df[clinical_phenotypes_df$pheno_list == phenotype, 'adjusted_p_value'],10)), 
           hjust = 1.1, vjust = 2, 
           size = 5, color = "black")



phenotype <- 'MED.DISEASES.Endocrine.DiabetesT2'
Dag3_cases <- as.numeric(na.omit(dag3_scores_phenos_model[dag3_scores_phenos_model[,phenotype] == 'Y', 'predicted_risk_score']))
#Dag3_controls_step1 <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'N',]
Dag3_controls <- as.numeric(na.omit(Dag3_controls_step1[Dag3_controls_step1[, phenotype] == 'N', 'predicted_risk_score']))
#Dag3_healthy <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'Y', 'predicted_risk_score']
#Dag3_new <- new_onset_dag3_complete$predicted_risk_score

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep(paste("Cases, n=", length(Dag3_cases)), length(Dag3_cases)), rep(paste("Controls, n=", length(Dag3_controls)), length(Dag3_controls))))
)

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = paste("Density Plot of ", phenotype),
       x = "Value",
       y = "Density") +
  theme_minimal() + 
  annotate("text", 
           x = Inf, y = Inf, 
           label = paste('Linear model pval=', round(clinical_phenotypes_df[clinical_phenotypes_df$pheno_list == phenotype, 'adjusted_p_value'],5)), 
           hjust = 1.1, vjust = 2, 
           size = 5, color = "black")



phenotype <- 'MED.DISEASES.Gastrointestinal.Rome3_IBS.TypeD'
Dag3_cases <- as.numeric(na.omit(dag3_scores_phenos_model[dag3_scores_phenos_model[,phenotype] == 'Y', 'predicted_risk_score']))
#Dag3_controls_step1 <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'N',]
Dag3_controls <- as.numeric(na.omit(Dag3_controls_step1[Dag3_controls_step1[, phenotype] == 'N', 'predicted_risk_score']))
#Dag3_healthy <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'Y', 'predicted_risk_score']
#Dag3_new <- new_onset_dag3_complete$predicted_risk_score

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep(paste("Cases, n=", length(Dag3_cases)), length(Dag3_cases)), rep(paste("Controls, n=", length(Dag3_controls)), length(Dag3_controls))))
)

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = paste("Density Plot of ", phenotype),
       x = "Value",
       y = "Density") +
  theme_minimal() + 
  annotate("text", 
           x = Inf, y = Inf, 
           label = paste('Linear model pval=', round(clinical_phenotypes_df[clinical_phenotypes_df$pheno_list == phenotype, 'adjusted_p_value'],5)), 
           hjust = 1.1, vjust = 2, 
           size = 5, color = "black")






phenotype <- 'MED.DISEASES.Neurological.Dizziness.Falling'
Dag3_cases <- as.numeric(na.omit(dag3_scores_phenos_model[dag3_scores_phenos_model[,phenotype] == 'Y', 'predicted_risk_score']))
#Dag3_controls_step1 <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'N',]
Dag3_controls <- as.numeric(na.omit(Dag3_controls_step1[Dag3_controls_step1[, phenotype] == 'N', 'predicted_risk_score']))
#Dag3_healthy <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'Y', 'predicted_risk_score']
#Dag3_new <- new_onset_dag3_complete$predicted_risk_score

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep(paste("Cases, n=", length(Dag3_cases)), length(Dag3_cases)), rep(paste("Controls, n=", length(Dag3_controls)), length(Dag3_controls))))
)

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = paste("Density Plot of ", phenotype),
       x = "Value",
       y = "Density") +
  theme_minimal() + 
  annotate("text", 
           x = Inf, y = Inf, 
           label = paste('Linear model pval=', round(clinical_phenotypes_df[clinical_phenotypes_df$pheno_list == phenotype, 'adjusted_p_value'],5)), 
           hjust = 1.1, vjust = 2, 
           size = 5, color = "black")





phenotype <- 'MED.DISEASES.Other.Kidney.Stones'
Dag3_cases <- as.numeric(na.omit(dag3_scores_phenos_model[dag3_scores_phenos_model[,phenotype] == 'Y', 'predicted_risk_score']))
#Dag3_controls_step1 <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'N',]
Dag3_controls <- as.numeric(na.omit(Dag3_controls_step1[Dag3_controls_step1[, phenotype] == 'N', 'predicted_risk_score']))
#Dag3_healthy <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'Y', 'predicted_risk_score']
#Dag3_new <- new_onset_dag3_complete$predicted_risk_score

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep(paste("Cases, n=", length(Dag3_cases)), length(Dag3_cases)), rep(paste("Controls, n=", length(Dag3_controls)), length(Dag3_controls))))
)

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = paste("Density Plot of ", phenotype),
       x = "Value",
       y = "Density") +
  theme_minimal() + 
  annotate("text", 
           x = Inf, y = Inf, 
           label = paste('Linear model pval=', round(clinical_phenotypes_df[clinical_phenotypes_df$pheno_list == phenotype, 'adjusted_p_value'],5)), 
           hjust = 1.1, vjust = 2, 
           size = 5, color = "black")






phenotype <- 'MED.DISEASES.Gastrointestinal.FGID.Bloating'
Dag3_cases <- as.numeric(na.omit(dag3_scores_phenos_model[dag3_scores_phenos_model[,phenotype] == 'Y', 'predicted_risk_score']))
#Dag3_controls_step1 <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'N',]
Dag3_controls <- as.numeric(na.omit(Dag3_controls_step1[Dag3_controls_step1[, phenotype] == 'N', 'predicted_risk_score']))
#Dag3_healthy <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'Y', 'predicted_risk_score']
#Dag3_new <- new_onset_dag3_complete$predicted_risk_score

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep(paste("Cases, n=", length(Dag3_cases)), length(Dag3_cases)), rep(paste("Controls, n=", length(Dag3_controls)), length(Dag3_controls))))
)

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = paste("Density Plot of ", phenotype),
       x = "Value",
       y = "Density") +
  theme_minimal() + 
  annotate("text", 
           x = Inf, y = Inf, 
           label = paste('Linear model pval=', round(clinical_phenotypes_df[clinical_phenotypes_df$pheno_list == phenotype, 'adjusted_p_value'],5)), 
           hjust = 1.1, vjust = 2, 
           size = 5, color = "black")





```








### rerun the model without IBD cases
```{r compare ibd risk score with phenotypes}
# load in phenotypes
dag3_phenos <- read_csv('/Users/iwan/Research/IBD_risk/phenotypes/DAG3_metadata_merged_ready_v27.csv')
dag3_phenos$Rows <- NULL


dag3_scores_phenos <- merge(DAG3_dataframe, dag3_phenos, by.x = 'A_DAG3_sampleID', by.y ='DAG3_sampleID')


dag3_scores_phenos


DAG3_df_model <- na.omit(dag3_scores_phenos[,c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecal', 'base_gender', 'base_IBD1a2a', 'A_DAG3_sampleID')])# select variables for predicting


colnames(DAG3_df_model) <- c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBD', 'A_DAG3_sampleID')
# apply scalars
DAG3_df_model$score_prs_2015_scaled <- (DAG3_df_model$score_prs_2015 - mean_prs) / sd_prs

DAG3_df_model$score_diet_LLDS_scaled <- (DAG3_df_model$score_diet_LLDS - mean_diet) / sd_diet

DAG3_df_model$score_microbiome_log_scaled <- (DAG3_df_model$score_microbiome_log - mean_microbiome) / sd_microbiome

DAG3_df_model$score_env_u15_sel_weigh_scaled <- (DAG3_df_model$score_env_u15_sel_weigh - mean_env) / sd_env

DAG3_df_model$base_AgeFecalDiet_scaled <- (DAG3_df_model$base_AgeFecalDiet - mean_age) / sd_age

#c('score_microbiome_log_scaled', 'score_prs_2015_scaled', 'score_diet_LLDS_scaled', 'score_env_u15_sel_weigh_scaled', 'base_AgeFecalDiet_scaled', 'base_sex')

# predict using previous model
DAG3_df_model$predicted_risk_score <- predict(clinical_model, DAG3_df_model, quiet=T)

rownames(dag3_scores_phenos) <- dag3_scores_phenos$A_DAG3_sampleID

dag3_scores_phenos_model <- dag3_scores_phenos[DAG3_df_model$A_DAG3_sampleID, ] 
dag3_scores_phenos_model$predicted_risk_score   <- DAG3_df_model$predicted_risk_score

dag3_scores_phenos_model_noibd <- dag3_scores_phenos_model[dag3_scores_phenos_model$base_IBD1a2a ==0,]


# logistic regression model phenotype ~ risk score
pheno_list<-c()
coef_list<-c()
p_val_list<-c()
for (phenotype in colnames(dag3_scores_phenos_model_noibd)[grepl('MED.DISEASES',colnames(dag3_scores_phenos_model_noibd))]){
  print(phenotype)
  dag3_scores_phenos_model_noibd$outcome <- as.numeric(dag3_scores_phenos_model_noibd[,phenotype] == "Y")
  print(table(dag3_scores_phenos_model_noibd$outcome))
  formula <- 'outcome ~ predicted_risk_score'
  phenotype_model <- glm(formula, data=dag3_scores_phenos_model_noibd, family='binomial')
  model_summary <- summary(phenotype_model)
  print(phenotype)
  pheno_list <- c(pheno_list, phenotype)
  print(model_summary$coefficients[2,1])
  coef_list <- c(coef_list, model_summary$coefficients[2,1])
  print(model_summary$coefficients[2,4])
  p_val_list <- c(p_val_list, model_summary$coefficients[2,4])
}

clinical_phenotypes_df_noibd <- data.frame(pheno_list, coef_list, p_val_list)
clinical_phenotypes_df_noibd$adjusted_p_value <- p.adjust(clinical_phenotypes_df_noibd$p_val_list, method = "BH")



pheno_list<-c()
p_val_list<-c()
for (phenotype in colnames(dag3_scores_phenos_model_noibd)[grepl('MED.DISEASES',colnames(dag3_scores_phenos_model_noibd))]){
  dag3_scores_phenos_model_noibd$outcome <- as.numeric(dag3_scores_phenos_model_noibd[,phenotype] == "Y")
  #print(table(dag3_scores_phenos_model$outcome))
  if (length(na.omit(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==1 ,]$predicted_risk_score )) > 1){
    wilcox_p <- wilcox.test(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==1 ,]$predicted_risk_score, dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==0 ,]$predicted_risk_score )
    pheno_list <- c(pheno_list, phenotype)
  p_val_list <- c(p_val_list, wilcox_p$p.value)
  }
}

clinical_mann_phenotypes_df_noibd <- data.frame(pheno_list, p_val_list)
clinical_mann_phenotypes_df_noibd$adjusted_p_value <- p.adjust(clinical_mann_phenotypes_df_noibd$p_val_list, method = "BH")
write_csv(clinical_mann_phenotypes_df_noibd, '/Users/iwan/Research/IBD_risk/project_output/dataframes/mann_whitney_clinical_model_assocations.csv')

dag3_scores_phenos_model_stats <- dag3_scores_phenos_model
dag3_scores_phenos_model_stats$MED.DISEASES.Gastrointestinal.Autoimmune.IBD.UC <- NULL
dag3_scores_phenos_model_stats$MED.DISEASES.Gastrointestinal.Autoimmune.IBD.CD <- NULL

dag3_scores_phenos_model_stats$MED.DISEASES.Gastrointestinal.Autoimmune.IBD <- dag3_scores_phenos_model_stats$base_IBD1a2a

dag3_scores_phenos_model_stats[dag3_scores_phenos_model_stats$MED.DISEASES.Gastrointestinal.Autoimmune.IBD == 1,'MED.DISEASES.Gastrointestinal.Autoimmune.IBD'] <- "Y"
dag3_scores_phenos_model_stats[dag3_scores_phenos_model_stats$MED.DISEASES.Gastrointestinal.Autoimmune.IBD == 0,'MED.DISEASES.Gastrointestinal.Autoimmune.IBD'] <- "N"
dag3_scores_phenos_model_stats$score_microbiome_log_scaled <- (dag3_scores_phenos_model_stats$score_microbiome_log - mean_microbiome) / sd_microbiome



pheno_list<-c()
coef_list<-c()
p_val_list<-c()
#auc_list <- c()
for (phenotype in colnames(dag3_scores_phenos_model_stats)[grepl('MED.DISEASES',colnames(dag3_scores_phenos_model_stats))]){
  print(phenotype)
  dag3_scores_phenos_model_stats$outcome <- as.numeric(dag3_scores_phenos_model_stats[,phenotype] == "Y")
  formula <- 'outcome ~ predicted_risk_score'
  phenotype_model <- glm(formula, data=dag3_scores_phenos_model_stats, family='binomial')
  model_summary <- summary(phenotype_model)
  pheno_list <- c(pheno_list, phenotype)
  coef_list <- c(coef_list, model_summary$coefficients[2,1])
  p_val_list <- c(p_val_list, model_summary$coefficients[2,4])
}

clinical_linear_phenotypes_df <- data.frame(pheno_list, coef_list, p_val_list)
clinical_linear_phenotypes_df$adjusted_p_value <- p.adjust(clinical_linear_phenotypes_df$p_val_list, method = "BH")
clinical_linear_phenotypes_df$OR <-  exp(clinical_linear_phenotypes_df$coef_list)



write_csv(clinical_linear_phenotypes_df_noibd, '/Users/iwan/Research/IBD_risk/project_output/dataframes/linear_clinical_model_assocations.csv')

```
```{r BH calculation}
ranks<-rank(clinical_mann_phenotypes_df_noibd$p_val_list, ties.method = "last")
p_m_over_k<-clinical_mann_phenotypes_df_noibd$p_val_list*length(clinical_mann_phenotypes_df_noibd$p_val_list)/ranks
for (r in length(clinical_mann_phenotypes_df_noibd$p_val_list):1) {
  print(p_m_over_k[ranks>=r])
}
pvalues_adj<-c()
for (i in 1:length(clinical_mann_phenotypes_df_noibd$p_val_list)) {
  
  # find the rank
  tmp_rank<-ranks[i]
  
  # get all the p_m_over_k that are greater or equal to this rank
  # and get the min value
  pvalues_adj<-c(pvalues_adj, min(1,min(p_m_over_k[ranks>=tmp_rank])))
}
print(pvalues_adj)

clinical_mann_phenotypes_df_noibd$p_second_adjust <- pvalues_adj
```


## here we plot the distribution of the other clinical phenotypes
```{r plotting the output of the glm model}

phenotype <- 'MED.DISEASES.Pulmonary.COPD'
phenotype_density <- function(phenotype){
  

  Dag3_cases <- as.numeric(na.omit(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd[,phenotype] == 'Y', 'predicted_risk_score']))
  #Dag3_controls_step1 <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'N',]
  Dag3_controls <- as.numeric(na.omit(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd[, phenotype] == 'N', 'predicted_risk_score']))
  #Dag3_healthy <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'Y', 'predicted_risk_score']
  #Dag3_new <- new_onset_dag3_complete$predicted_risk_score
  
  data <- data.frame(
    Value = c(Dag3_cases, Dag3_controls),
    Group = factor(c(rep(paste("Cases, n=", length(Dag3_cases)), length(Dag3_cases)), rep(paste("Controls, n=", length(Dag3_controls)), length(Dag3_controls))))
  )
  
  # Plot density distributions for both lists
  ggplot(data, aes(x = Value, fill = Group)) +
    geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
    labs(title = paste("Density Plot of", phenotype, 'in DAG3'),
         x = "IBD risk score",
         y = "Density") +
    theme_minimal() + 
    annotate("text", 
             x = Inf, y = Inf, 
             label = paste('Mann-Whitney FDR:', round(clinical_mann_phenotypes_df_noibd[clinical_mann_phenotypes_df_noibd$pheno_list == phenotype, 'adjusted_p_value'],5)), 
             hjust = 1.1, vjust = 2, 
             size = 5, color = "black")
}
pdf('/Users/iwan/Research/IBD_risk/project_output/images/Med_diseases_association_DAG3_paper.pdf')
phenotype_density('MED.DISEASES.Pulmonary.COPD')
phenotype_density('MED.DISEASES.None.No.Diseases')
phenotype_density('MED.DISEASES.Endocrine.DiabetesT2')
phenotype_density('MED.DISEASES.Neurological.Dizziness.Falling')
phenotype_density('MED.DISEASES.Other.Autoimmune.Rheumatoid.Artritis')
phenotype_density('MED.DISEASES.Other.Osteoarthritis')
phenotype_density('MED.DISEASES.Pulmonary.Autoimmune.Asthma')
phenotype_density('MED.DISEASES.Cardiovascular.Colesterol.high')
phenotype_density('MED.DISEASES.Gastrointestinal.FGID.Bloating')
dev.off()

```





### check overlapping phenotypes
```{r overlap between phenotype}

#clinical_phenotypes_df_noibd[clinical_phenotypes_df_noibd$adjusted_p_value < 0.05,'pheno_list']

#dag3_scores_phenos_model_noibd[,clinical_mann_phenotypes_df_noibd[clinical_mann_phenotypes_df_noibd$adjusted_p_value < 0.05,'pheno_list']]

matrix_df <- dag3_scores_phenos_model_noibd[,clinical_mann_phenotypes_df_noibd[clinical_mann_phenotypes_df_noibd$adjusted_p_value < 0.05,'pheno_list']]
matrix_df[is.na(matrix_df)] <- 'N'
# Step 1: Convert Y/N to 1/0
df_binary <- as.data.frame(lapply(matrix_df, function(x) as.numeric(x == "Y")))

# Step 2: Compute the co-occurrence matrix using matrix multiplication
cooccurrence_matrix <- t(as.matrix(df_binary)) %*% as.matrix(df_binary)
#cooccurrence_matrix$MED.DISEASES.None.No.Diseases <- NULL
# Display the co-occurrence matrix
print(cooccurrence_matrix)
# Step 3: Convert the co-occurrence matrix into long format for ggplot
library(reshape)
cooccurrence_long <- melt(cooccurrence_matrix)
cooccurrence_long <- cooccurrence_long[cooccurrence_long$X1 != 'MED.DISEASES.None.No.Diseases',]
cooccurrence_long <- cooccurrence_long[cooccurrence_long$X2 != 'MED.DISEASES.None.No.Diseases',]


cooccurrence_long[cooccurrence_long$X1 == cooccurrence_long$X2,]$value <- 0 
# Step 4: Create the heatmap using ggplot
ggplot(cooccurrence_long, aes(X1, X2, fill = value)) + 
  geom_tile(color = "white") +
  scale_fill_gradient(low = "white", high = "steelblue") + 
  labs(title = "Disease Co-occurrence Heatmap", x = "Disease", y = "Disease", fill = "Co-occurrence") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

```




## checking all phenotypes in the same model see whats left
```{r summary check }
formula <- clinical_phenotypes_df_noibd[clinical_phenotypes_df_noibd$adjusted_p_value < 0.05,'pheno_list']
formula <- ''
formula <- 'predicted_risk_score ~ MED.DISEASES.Endocrine.DiabetesT2 + MED.DISEASES.Other.Kidney.Stones + MED.DISEASES.Gastrointestinal.FGID.Bloating + MED.DISEASES.Neurological.Dizziness.Falling + MED.DISEASES.Pulmonary.COPD + MED.DISEASES.None.No.Diseases + MED.DISEASES.Gastrointestinal.Rome3_IBS.TypeD'
phenotype_model <- glm(formula, data=dag3_scores_phenos_model_noibd, family='gaussian')
  
summary(phenotype_model)
```





```{r check if copd is driven by microbiome or genetics}



phenotype <- 'MED.DISEASES.Pulmonary.COPD'
Dag3_cases <- as.numeric(na.omit(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd[,phenotype] == 'Y', 'score_microbiome_log']))
#Dag3_controls_step1 <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'N',]
Dag3_controls <- as.numeric(na.omit(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd[, phenotype] == 'N', 'score_microbiome_log']))
#Dag3_healthy <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'Y', 'predicted_risk_score']
#Dag3_new <- new_onset_dag3_complete$predicted_risk_score

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep(paste("Cases, n=", length(Dag3_cases)), length(Dag3_cases)), rep(paste("Controls, n=", length(Dag3_controls)), length(Dag3_controls))))
)

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = paste("Density Plot of ", phenotype),
       x = "Value",
       y = "Density") +
  theme_minimal() 



phenotype <- 'MED.DISEASES.Pulmonary.COPD'
Dag3_cases <- as.numeric(na.omit(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd[,phenotype] == 'Y', 'score_prs_2015']))
#Dag3_controls_step1 <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'N',]
Dag3_controls <- as.numeric(na.omit(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd[, phenotype] == 'N', 'score_prs_2015']))
#Dag3_healthy <- dag3_scores_phenos_model[dag3_scores_phenos_model$MED.DISEASES.None.No.Diseases == 'Y', 'predicted_risk_score']
#Dag3_new <- new_onset_dag3_complete$predicted_risk_score

data <- data.frame(
  Value = c(Dag3_cases, Dag3_controls),
  Group = factor(c(rep(paste("Cases, n=", length(Dag3_cases)), length(Dag3_cases)), rep(paste("Controls, n=", length(Dag3_controls)), length(Dag3_controls))))
)

# Plot density distributions for both lists
ggplot(data, aes(x = Value, fill = Group)) +
  geom_density(alpha = 0.5) +  # Overlay the two density plots with some transparency
  labs(title = paste("Density Plot of ", phenotype),
       x = "Value",
       y = "Density") +
  theme_minimal() 


```






```{r train the model on data without microbiome score then check association}

model_dataframe <- na.omit(updated_full_train_data[, c('score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBDconf')])
# score_microbiome_log 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh'

# here we set rescale parameters for the training data, and then later we choose the same rescaling parameters for the DAG3 cohort
mean_prs <- mean(model_dataframe$score_prs_2015)
sd_prs <- sd(model_dataframe$score_prs_2015)
mean_diet <- mean(model_dataframe$score_diet_LLDS)
sd_diet <- sd(model_dataframe$score_diet_LLDS)
mean_env <- mean(model_dataframe$score_env_u15_sel_weigh)
sd_env <- sd(model_dataframe$score_env_u15_sel_weigh)
mean_age <- mean(model_dataframe$base_AgeFecalDiet)
sd_age <- sd(model_dataframe$base_AgeFecalDiet)


# Apply scalars
model_dataframe$score_prs_2015_scaled <- (model_dataframe$score_prs_2015 - mean_prs) / sd_prs

model_dataframe$score_diet_LLDS_scaled <- (model_dataframe$score_diet_LLDS - mean_diet) / sd_diet

model_dataframe$score_env_u15_sel_weigh_scaled <- (model_dataframe$score_env_u15_sel_weigh - mean_env) / sd_env

model_dataframe$base_AgeFecalDiet_scaled <- (model_dataframe$base_AgeFecalDiet - mean_age) / sd_age



trainIndex <- sample(1:nrow(model_dataframe), 0.75 * nrow(model_dataframe))

# setup and train
formula <- 'base_IBDconf ~ score_prs_2015_scaled + score_diet_LLDS_scaled + score_env_u15_sel_weigh_scaled + base_AgeFecalDiet_scaled + base_sex'
no_microbiome_clinical_model <- glm(formula, data=model_dataframe[trainIndex,], family='gaussian')



summary(no_microbiome_clinical_model)



DAG3_df_model <- na.omit(dag3_scores_phenos[,c('score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecal', 'base_gender', 'base_IBD1a2a', 'A_DAG3_sampleID')])# select variables for predicting


colnames(DAG3_df_model) <- c( 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBD', 'A_DAG3_sampleID')
# apply scalars
DAG3_df_model$score_prs_2015_scaled <- (DAG3_df_model$score_prs_2015 - mean_prs) / sd_prs

DAG3_df_model$score_diet_LLDS_scaled <- (DAG3_df_model$score_diet_LLDS - mean_diet) / sd_diet

DAG3_df_model$score_env_u15_sel_weigh_scaled <- (DAG3_df_model$score_env_u15_sel_weigh - mean_env) / sd_env

DAG3_df_model$base_AgeFecalDiet_scaled <- (DAG3_df_model$base_AgeFecalDiet - mean_age) / sd_age

#c('score_microbiome_log_scaled', 'score_prs_2015_scaled', 'score_diet_LLDS_scaled', 'score_env_u15_sel_weigh_scaled', 'base_AgeFecalDiet_scaled', 'base_sex')

# predict using previous model
DAG3_df_model$predicted_risk_score <- predict(no_microbiome_clinical_model, DAG3_df_model, quiet=T)


rownames(dag3_scores_phenos) <- dag3_scores_phenos$A_DAG3_sampleID

dag3_scores_phenos_model <- dag3_scores_phenos[DAG3_df_model$A_DAG3_sampleID, ] 
dag3_scores_phenos_model$predicted_risk_score   <- DAG3_df_model$predicted_risk_score

dag3_scores_phenos_model_noibd <- dag3_scores_phenos_model[dag3_scores_phenos_model$base_IBD1a2a ==0,]



pheno_list<-c()
p_val_list<-c()
for (phenotype in colnames(dag3_scores_phenos_model_noibd)[grepl('MED.DISEASES',colnames(dag3_scores_phenos_model_noibd))]){
  dag3_scores_phenos_model_noibd$outcome <- as.numeric(dag3_scores_phenos_model_noibd[,phenotype] == "Y")
  #print(table(dag3_scores_phenos_model$outcome))
  if (length(na.omit(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==1 ,]$predicted_risk_score )) > 1){
    wilcox_p <- wilcox.test(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==1 ,]$predicted_risk_score, dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==0 ,]$predicted_risk_score )
    pheno_list <- c(pheno_list, phenotype)
  p_val_list <- c(p_val_list, wilcox_p$p.value)
  }
}

clinical_mann_phenotypes_df_noibd_nomicrobiome <- data.frame(pheno_list, p_val_list)
clinical_mann_phenotypes_df_noibd_nomicrobiome$adjusted_p_value <- p.adjust(clinical_mann_phenotypes_df_noibd_nomicrobiome$p_val_list, method = "BH")

print(clinical_mann_phenotypes_df_noibd_nomicrobiome[clinical_mann_phenotypes_df_noibd_nomicrobiome$adjusted_p_value < 0.05,] )

```



```{r train the model on data without genetics score then check association}

model_dataframe <- na.omit(updated_full_train_data[, c('score_microbiome_log', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBDconf')])
# score_microbiome_log 'score_prs_2015', 'score_diet_LLDS', 'score_env_u15_sel_weigh'

# here we set rescale parameters for the training data, and then later we choose the same rescaling parameters for the DAG3 cohort
mean_prs <- mean(model_dataframe$score_prs_2015)
sd_prs <- sd(model_dataframe$score_prs_2015)
mean_diet <- mean(model_dataframe$score_diet_LLDS)
sd_diet <- sd(model_dataframe$score_diet_LLDS)
mean_env <- mean(model_dataframe$score_env_u15_sel_weigh)
sd_env <- sd(model_dataframe$score_env_u15_sel_weigh)
mean_age <- mean(model_dataframe$base_AgeFecalDiet)
sd_age <- sd(model_dataframe$base_AgeFecalDiet)


# Apply scalars
model_dataframe$score_microbiome_log_scaled <- (model_dataframe$score_microbiome_log - mean_microbiome) / sd_microbiome

model_dataframe$score_diet_LLDS_scaled <- (model_dataframe$score_diet_LLDS - mean_diet) / sd_diet

model_dataframe$score_env_u15_sel_weigh_scaled <- (model_dataframe$score_env_u15_sel_weigh - mean_env) / sd_env

model_dataframe$base_AgeFecalDiet_scaled <- (model_dataframe$base_AgeFecalDiet - mean_age) / sd_age



trainIndex <- sample(1:nrow(model_dataframe), 0.75 * nrow(model_dataframe))

# setup and train
formula <- 'base_IBDconf ~ score_microbiome_log_scaled + score_diet_LLDS_scaled + score_env_u15_sel_weigh_scaled + base_AgeFecalDiet_scaled + base_sex'
no_microbiome_clinical_model <- glm(formula, data=model_dataframe[trainIndex,], family='gaussian')



summary(no_microbiome_clinical_model)



DAG3_df_model <- na.omit(dag3_scores_phenos[,c('score_microbiome_log', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecal', 'base_gender', 'base_IBD1a2a', 'A_DAG3_sampleID')])# select variables for predicting


colnames(DAG3_df_model) <- c( 'score_microbiome_log', 'score_diet_LLDS', 'score_env_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBD', 'A_DAG3_sampleID')
# apply scalars
DAG3_df_model$score_microbiome_log_scaled <- (DAG3_df_model$score_microbiome_log - mean_microbiome) / sd_microbiome

DAG3_df_model$score_diet_LLDS_scaled <- (DAG3_df_model$score_diet_LLDS - mean_diet) / sd_diet

DAG3_df_model$score_env_u15_sel_weigh_scaled <- (DAG3_df_model$score_env_u15_sel_weigh - mean_env) / sd_env

DAG3_df_model$base_AgeFecalDiet_scaled <- (DAG3_df_model$base_AgeFecalDiet - mean_age) / sd_age

#c('score_microbiome_log_scaled', 'score_prs_2015_scaled', 'score_diet_LLDS_scaled', 'score_env_u15_sel_weigh_scaled', 'base_AgeFecalDiet_scaled', 'base_sex')

# predict using previous model
DAG3_df_model$predicted_risk_score <- predict(no_microbiome_clinical_model, DAG3_df_model, quiet=T)


rownames(dag3_scores_phenos) <- dag3_scores_phenos$A_DAG3_sampleID

dag3_scores_phenos_model <- dag3_scores_phenos[DAG3_df_model$A_DAG3_sampleID, ] 
dag3_scores_phenos_model$predicted_risk_score   <- DAG3_df_model$predicted_risk_score

dag3_scores_phenos_model_noibd <- dag3_scores_phenos_model[dag3_scores_phenos_model$base_IBD1a2a ==0,]



pheno_list<-c()
p_val_list<-c()
for (phenotype in colnames(dag3_scores_phenos_model_noibd)[grepl('MED.DISEASES',colnames(dag3_scores_phenos_model_noibd))]){
  dag3_scores_phenos_model_noibd$outcome <- as.numeric(dag3_scores_phenos_model_noibd[,phenotype] == "Y")
  #print(table(dag3_scores_phenos_model$outcome))
  if (length(na.omit(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==1 ,]$predicted_risk_score )) > 1){
    wilcox_p <- wilcox.test(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==1 ,]$predicted_risk_score, dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==0 ,]$predicted_risk_score )
    pheno_list <- c(pheno_list, phenotype)
  p_val_list <- c(p_val_list, wilcox_p$p.value)
  }
}

clinical_mann_phenotypes_df_noibd_nomicrobiome <- data.frame(pheno_list, p_val_list)
clinical_mann_phenotypes_df_noibd_nomicrobiome$adjusted_p_value <- p.adjust(clinical_mann_phenotypes_df_noibd_nomicrobiome$p_val_list, method = "BH")

print(clinical_mann_phenotypes_df_noibd_nomicrobiome[clinical_mann_phenotypes_df_noibd_nomicrobiome$adjusted_p_value < 0.05,] )

```
```{r train the model on data without genetics score then check association}

model_dataframe <- na.omit(updated_full_train_data[, c('score_microbiome_log', 'score_prs_2015', 'score_env_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBDconf')])
# score_microbiome_log '', 'score_diet_LLDS', 'score_env_u15_sel_weigh'

# here we set rescale parameters for the training data, and then later we choose the same rescaling parameters for the DAG3 cohort
mean_prs <- mean(model_dataframe$score_prs_2015)
sd_prs <- sd(model_dataframe$score_prs_2015)
mean_diet <- mean(model_dataframe$score_diet_LLDS)
sd_diet <- sd(model_dataframe$score_diet_LLDS)
mean_env <- mean(model_dataframe$score_env_u15_sel_weigh)
sd_env <- sd(model_dataframe$score_env_u15_sel_weigh)
mean_age <- mean(model_dataframe$base_AgeFecalDiet)
sd_age <- sd(model_dataframe$base_AgeFecalDiet)


# Apply scalars
model_dataframe$score_prs_2015_scaled <- (model_dataframe$score_prs_2015 - mean_prs) / sd_prs

model_dataframe$score_microbiome_log_scaled <- (model_dataframe$score_microbiome_log - mean_microbiome) / sd_microbiome

#model_dataframe$score_diet_LLDS_scaled <- (model_dataframe$score_diet_LLDS - mean_diet) / sd_diet

model_dataframe$score_env_u15_sel_weigh_scaled <- (model_dataframe$score_env_u15_sel_weigh - mean_env) / sd_env

model_dataframe$base_AgeFecalDiet_scaled <- (model_dataframe$base_AgeFecalDiet - mean_age) / sd_age



trainIndex <- sample(1:nrow(model_dataframe), 0.75 * nrow(model_dataframe))

# setup and train
formula <- 'base_IBDconf ~ score_microbiome_log_scaled + score_prs_2015_scaled + score_env_u15_sel_weigh_scaled + base_AgeFecalDiet_scaled + base_sex'
no_microbiome_clinical_model <- glm(formula, data=model_dataframe[trainIndex,], family='gaussian')



summary(no_microbiome_clinical_model)



DAG3_df_model <- na.omit(dag3_scores_phenos[,c('score_microbiome_log', 'score_prs_2015', 'score_env_u15_sel_weigh', 'base_AgeFecal', 'base_gender', 'base_IBD1a2a', 'A_DAG3_sampleID')])# select variables for predicting


colnames(DAG3_df_model) <- c( 'score_microbiome_log', 'score_prs_2015', 'score_env_u15_sel_weigh', 'base_AgeFecalDiet', 'base_sex', 'base_IBD', 'A_DAG3_sampleID')
# apply scalars
DAG3_df_model$score_prs_2015_scaled <- (DAG3_df_model$score_prs_2015 - mean_prs) / sd_prs

DAG3_df_model$score_microbiome_log_scaled <- (DAG3_df_model$score_microbiome_log - mean_microbiome) / sd_microbiome

#DAG3_df_model$score_diet_LLDS_scaled <- (DAG3_df_model$score_diet_LLDS - mean_diet) / sd_diet

DAG3_df_model$score_env_u15_sel_weigh_scaled <- (DAG3_df_model$score_env_u15_sel_weigh - mean_env) / sd_env

DAG3_df_model$base_AgeFecalDiet_scaled <- (DAG3_df_model$base_AgeFecalDiet - mean_age) / sd_age

#c('score_microbiome_log_scaled', 'score_prs_2015_scaled', 'score_diet_LLDS_scaled', 'score_env_u15_sel_weigh_scaled', 'base_AgeFecalDiet_scaled', 'base_sex')

# predict using previous model
DAG3_df_model$predicted_risk_score <- predict(no_microbiome_clinical_model, DAG3_df_model, quiet=T)


rownames(dag3_scores_phenos) <- dag3_scores_phenos$A_DAG3_sampleID

dag3_scores_phenos_model <- dag3_scores_phenos[DAG3_df_model$A_DAG3_sampleID, ] 
dag3_scores_phenos_model$predicted_risk_score   <- DAG3_df_model$predicted_risk_score

dag3_scores_phenos_model_noibd <- dag3_scores_phenos_model[dag3_scores_phenos_model$base_IBD1a2a ==0,]



pheno_list<-c()
p_val_list<-c()
for (phenotype in colnames(dag3_scores_phenos_model_noibd)[grepl('MED.DISEASES',colnames(dag3_scores_phenos_model_noibd))]){
  dag3_scores_phenos_model_noibd$outcome <- as.numeric(dag3_scores_phenos_model_noibd[,phenotype] == "Y")
  #print(table(dag3_scores_phenos_model$outcome))
  if (length(na.omit(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==1 ,]$predicted_risk_score )) > 1){
    wilcox_p <- wilcox.test(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==1 ,]$predicted_risk_score, dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==0 ,]$predicted_risk_score )
    pheno_list <- c(pheno_list, phenotype)
  p_val_list <- c(p_val_list, wilcox_p$p.value)
  }
}

clinical_mann_phenotypes_df_noibd_nomicrobiome <- data.frame(pheno_list, p_val_list)
clinical_mann_phenotypes_df_noibd_nomicrobiome$adjusted_p_value <- p.adjust(clinical_mann_phenotypes_df_noibd_nomicrobiome$p_val_list, method = "BH")

print(clinical_mann_phenotypes_df_noibd_nomicrobiome[clinical_mann_phenotypes_df_noibd_nomicrobiome$adjusted_p_value < 0.05,] )

```

```{r train the model on data without genetics score then check association}

model_dataframe <- na.omit(updated_full_train_data[, c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'base_AgeFecalDiet', 'base_sex', 'base_IBDconf')])
# score_microbiome_log '', '', 'score_env_u15_sel_weigh'

# here we set rescale parameters for the training data, and then later we choose the same rescaling parameters for the DAG3 cohort
mean_prs <- mean(model_dataframe$score_prs_2015)
sd_prs <- sd(model_dataframe$score_prs_2015)
mean_diet <- mean(model_dataframe$score_diet_LLDS)
sd_diet <- sd(model_dataframe$score_diet_LLDS)
mean_env <- mean(model_dataframe$score_env_u15_sel_weigh)
sd_env <- sd(model_dataframe$score_env_u15_sel_weigh)
mean_age <- mean(model_dataframe$base_AgeFecalDiet)
sd_age <- sd(model_dataframe$base_AgeFecalDiet)


# Apply scalars
model_dataframe$score_prs_2015_scaled <- (model_dataframe$score_prs_2015 - mean_prs) / sd_prs

model_dataframe$score_microbiome_log_scaled <- (model_dataframe$score_microbiome_log - mean_microbiome) / sd_microbiome

model_dataframe$score_diet_LLDS_scaled <- (model_dataframe$score_diet_LLDS - mean_diet) / sd_diet

#model_dataframe$score_env_u15_sel_weigh_scaled <- (model_dataframe$score_env_u15_sel_weigh - mean_env) / sd_env

model_dataframe$base_AgeFecalDiet_scaled <- (model_dataframe$base_AgeFecalDiet - mean_age) / sd_age



trainIndex <- sample(1:nrow(model_dataframe), 0.75 * nrow(model_dataframe))

# setup and train
formula <- 'base_IBDconf ~ score_microbiome_log_scaled + score_prs_2015_scaled + score_diet_LLDS_scaled + base_AgeFecalDiet_scaled + base_sex'
no_microbiome_clinical_model <- glm(formula, data=model_dataframe[trainIndex,], family='gaussian')



summary(no_microbiome_clinical_model)



DAG3_df_model <- na.omit(dag3_scores_phenos[,c('score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'base_AgeFecal', 'base_gender', 'base_IBD1a2a', 'A_DAG3_sampleID')])# select variables for predicting


colnames(DAG3_df_model) <- c( 'score_microbiome_log', 'score_prs_2015', 'score_diet_LLDS', 'base_AgeFecalDiet', 'base_sex', 'base_IBD', 'A_DAG3_sampleID')
# apply scalars
DAG3_df_model$score_prs_2015_scaled <- (DAG3_df_model$score_prs_2015 - mean_prs) / sd_prs

DAG3_df_model$score_microbiome_log_scaled <- (DAG3_df_model$score_microbiome_log - mean_microbiome) / sd_microbiome

DAG3_df_model$score_diet_LLDS_scaled <- (DAG3_df_model$score_diet_LLDS - mean_diet) / sd_diet

#DAG3_df_model$score_env_u15_sel_weigh_scaled <- (DAG3_df_model$score_env_u15_sel_weigh - mean_env) / sd_env

DAG3_df_model$base_AgeFecalDiet_scaled <- (DAG3_df_model$base_AgeFecalDiet - mean_age) / sd_age

#c('score_microbiome_log_scaled', 'score_prs_2015_scaled', 'score_diet_LLDS_scaled', 'score_env_u15_sel_weigh_scaled', 'base_AgeFecalDiet_scaled', 'base_sex')

# predict using previous model
DAG3_df_model$predicted_risk_score <- predict(no_microbiome_clinical_model, DAG3_df_model, quiet=T)


rownames(dag3_scores_phenos) <- dag3_scores_phenos$A_DAG3_sampleID

dag3_scores_phenos_model <- dag3_scores_phenos[DAG3_df_model$A_DAG3_sampleID, ] 
dag3_scores_phenos_model$predicted_risk_score   <- DAG3_df_model$predicted_risk_score

dag3_scores_phenos_model_noibd <- dag3_scores_phenos_model[dag3_scores_phenos_model$base_IBD1a2a ==0,]



pheno_list<-c()
p_val_list<-c()
for (phenotype in colnames(dag3_scores_phenos_model_noibd)[grepl('MED.DISEASES',colnames(dag3_scores_phenos_model_noibd))]){
  dag3_scores_phenos_model_noibd$outcome <- as.numeric(dag3_scores_phenos_model_noibd[,phenotype] == "Y")
  #print(table(dag3_scores_phenos_model$outcome))
  if (length(na.omit(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==1 ,]$predicted_risk_score )) > 1){
    wilcox_p <- wilcox.test(dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==1 ,]$predicted_risk_score, dag3_scores_phenos_model_noibd[dag3_scores_phenos_model_noibd$outcome==0 ,]$predicted_risk_score )
    pheno_list <- c(pheno_list, phenotype)
  p_val_list <- c(p_val_list, wilcox_p$p.value)
  }
}

clinical_mann_phenotypes_df_noibd_nomicrobiome <- data.frame(pheno_list, p_val_list)
clinical_mann_phenotypes_df_noibd_nomicrobiome$adjusted_p_value <- p.adjust(clinical_mann_phenotypes_df_noibd_nomicrobiome$p_val_list, method = "BH")

print(clinical_mann_phenotypes_df_noibd_nomicrobiome[clinical_mann_phenotypes_df_noibd_nomicrobiome$adjusted_p_value < 0.05,] )

```


```{r bristol stool scale vs microbiome score}
dag3_scores_phenos_model_noibd$score_microbiome_log

dag3_scores_phenos_model_noibd$META.POOP.BristolMean

formula <- 'base_IBD1a2a ~score_microbiome_log +  score_diet_LLDS + score_env_u15_sel_weigh + META.POOP.BristolMean'
phenotype_model <- glm(formula, data=dag3_scores_phenos_model, family='gaussian')
summary(phenotype_model)

df_plot

#plot(dag3_scores_phenos_model_noibd$score_microbiome_log, dag3_scores_phenos_model_noibd$META.POOP.BristolMean)
df_plot_bristol <-  na.omit(dag3_scores_phenos_model[,c('score_microbiome_log', 'META.POOP.BristolMean')])
correlation <- cor(df_plot_bristol$score_microbiome_log, df_plot_bristol$META.POOP.BristolMean)

# Step 2: Create the scatter plot with a trendline and correlation
ggplot(df_plot_bristol, aes(x = score_microbiome_log, y = META.POOP.BristolMean)) + 
  geom_point() +  # Scatter plot
  geom_smooth(method = "lm", col = "blue", se = FALSE) +  # Add linear trendline
  labs(title = "Scatter Plot with Trendline",
       subtitle = paste("Correlation: ", round(correlation, 2)),
       x = "X Axis Label",
       y = "Y Axis Label") +
  theme_minimal() +
  ggtitle('Bristol stool scale versus microbiome score')



df_plot_bristol <-  na.omit(dag3_scores_phenos_model[,c('predicted_risk_score', 'META.POOP.BristolMean')])
correlation <- cor(df_plot_bristol$predicted_risk_score, df_plot_bristol$META.POOP.BristolMean)

# Step 2: Create the scatter plot with a trendline and correlation
ggplot(df_plot_bristol, aes(x = predicted_risk_score, y = META.POOP.BristolMean)) + 
  geom_point() +  # Scatter plot
  geom_smooth(method = "lm", col = "blue", se = FALSE) +  # Add linear trendline
  labs(title = "Scatter Plot with Trendline",
       subtitle = paste("Correlation: ", round(correlation, 2)),
       x = "X Axis Label",
       y = "Y Axis Label") +
  theme_minimal() +
  ggtitle('Bristol stool scale versus prediction score')


df_plot_bristol <-  na.omit(dag3_scores_phenos_model[,c('predicted_risk_score', 'score_microbiome_log')])
correlation <- cor(df_plot_bristol$predicted_risk_score, df_plot_bristol$score_microbiome_log)

# Step 2: Create the scatter plot with a trendline and correlation
ggplot(df_plot_bristol, aes(x = predicted_risk_score, y = score_microbiome_log)) + 
  geom_point() +  # Scatter plot
  geom_smooth(method = "lm", col = "blue", se = FALSE) +  # Add linear trendline
  labs(title = "Scatter Plot with Trendline",
       subtitle = paste("Correlation: ", round(correlation, 2)),
       x = "X Axis Label",
       y = "Y Axis Label") +
  theme_minimal() +
  ggtitle('microbiome score versus prediction score')





```


```{r all disease boxplot}

disease_names <- clinical_mann_phenotypes_df_noibd[clinical_mann_phenotypes_df_noibd$adjusted_p_value < 0.05,]$pheno_list

diseases_to_plot <-  c("MED.DISEASES.Other.Autoimmune.Rheumatoid.Artritis", "MED.DISEASES.Pulmonary.Autoimmune.Asthma", "MED.DISEASES.Gastrointestinal.FGID.Bloating" , "MED.DISEASES.Pulmonary.COPD"  , "MED.DISEASES.Other.Osteoarthritis", "MED.DISEASES.Endocrine.DiabetesT2", 'base_IBD1a2a'   )
#dag3_scores_phenos_model_noibd_plot <- as.numeric(dag3_scores_phenos_model_noibd[,disease_names] == "Y")

dag3_diseases_plot <- dag3_scores_phenos_model[,c(diseases_to_plot, 'score_microbiome_log')]



# Load necessary libraries
library(dplyr)
library(tidyr)
library(ggplot2)

# Assuming 'dag3_diseases_plot' is your dataframe and has multiple disease columns plus 'predicted_risk_score'
# Step 1: Standardize disease columns (convert to 0 and 1 for presence and absence)

# First, replace 'Y', 'N', and NA values with 1 (presence) and 0 (absence)
dag3_diseases_plot_clean <- dag3_diseases_plot %>%
  mutate(across(where(is.character), ~ recode(., "Y" = 1, "N" = 0))) %>%
  mutate(across(where(is.numeric), ~ replace_na(., 0))) # NA is treated as no disease (0)

# Step 2: Create a new column that identifies people with no diseases
# This assumes all disease columns are binary (0 or 1)
dag3_diseases_plot_clean <- dag3_diseases_plot_clean %>%
  mutate(no_disease = ifelse(rowSums(select(., -score_microbiome_log)) == 0, 1, 0)) 

dag3_diseases_plot_clean <- dag3_diseases_plot_clean[dag3_diseases_plot_clean$score_microbiome_log > 10,]
# Step 3: Reshape the dataframe to a long format for ggplot
dag3_diseases_long <- dag3_diseases_plot_clean %>%
  pivot_longer(cols = -score_microbiome_log, 
               names_to = "disease", 
               values_to = "presence") %>%
  filter(presence == 1 | disease == "no_disease") # Keep only those with the disease or no disease group

# Step 4: Plot the violin plot using ggplot
ggplot(dag3_diseases_long, aes(x = disease, y = score_microbiome_log, fill = disease)) +
  geom_violin(trim = FALSE) +
  geom_boxplot() +
  geom_jitter(width = 0.2, alpha = 0.3) + # Optional: add jitter to show individual data points
  labs(title = "Violin Plot of Predicted Risk Score by Disease Group",
       x = "Disease Group",
       y = "Predicted Risk Score") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))




# Step 1: Standardize disease columns (convert to 0 and 1 for presence and absence)
dag3_diseases_plot_clean <- dag3_diseases_plot %>%
  mutate(across(where(is.character), ~ recode(., "Y" = 1, "N" = 0))) %>%
  mutate(across(where(is.numeric), ~ replace_na(., 0))) # NA is treated as no disease (0)
dag3_diseases_plot_clean <- dag3_diseases_plot_clean[dag3_diseases_plot_clean$score_microbiome_log > 10,]
# Step 2: Create a new column that identifies people with no diseases
dag3_diseases_plot_clean <- dag3_diseases_plot_clean %>%
  mutate(no_disease = ifelse(rowSums(select(., -score_microbiome_log)) == 0, 1, 0))

# Step 3: Reshape the dataframe to long format for ggplot
dag3_diseases_long <- dag3_diseases_plot_clean %>%
  pivot_longer(cols = -score_microbiome_log, 
               names_to = "disease", 
               values_to = "presence") %>%
  filter(presence == 1 | disease == "no_disease")

# Step 4: Calculate median risk scores for each disease group
medians <- dag3_diseases_long %>%
  group_by(disease) %>%
  summarise(median_risk = median(score_microbiome_log, na.rm = TRUE))

# Step 5: Reorder the disease groups by median predicted risk score
dag3_diseases_long$disease <- factor(dag3_diseases_long$disease, 
                                     levels = medians$disease[order(medians$median_risk)])

# Step 6: Plot the violin plot with ordered groups
ggplot(dag3_diseases_long, aes(x = disease, y = score_microbiome_log, fill = disease)) +
  geom_violin(trim = FALSE) +
  geom_boxplot() +
  geom_jitter(width = 0.2, alpha = 0.3) + # Optional: add jitter to show individual data points
  labs(title = "Violin Plot of Predicted Risk Score by Disease Group (Ordered by Median)",
       x = "Disease Group",
       y = "Predicted Risk Score") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))



```





```{r distribution with lines and % of cases found given any score}

# Calculate the overall density for all values
density_data <- density(DAG3_df_model$predicted_risk_score)

# Create a dataframe for the density values (x and y)
density_df <- data.frame(x = density_data$x, y = density_data$y)

# For "Disease" cases, find the corresponding height on the density curve
disease_cases <- DAG3_df_model %>% filter(base_IBD == 1)

# Interpolate the y-values (heights) for the "Disease" cases from the density curve
disease_cases$height <- approx(density_df$x, density_df$y, xout = disease_cases$predicted_risk_score)$y

# Plot the smooth density for all values and individual lines for "Disease" cases
ggplot() +
  geom_line(data = density_df, aes(x = x, y = y), color = "lightblue", size = 1) +  # Smooth density for all
  geom_segment(data = disease_cases, aes(x = predicted_risk_score, xend = predicted_risk_score, y = 0, yend = height),
               color = "red", size = 0.4) +  # Vertical lines for "Disease" cases
  labs(title = "Smooth Density with Highlighted Disease Cases",
       x = "Values",
       y = "Density") +
  theme_minimal()

line_plot_df <- DAG3_df_model

# Step 1: Sort dataframe by score
line_plot_df <- line_plot_df %>% arrange(predicted_risk_score)

# Step 2: Calculate the cumulative number of IBD diagnoses for each score cutoff
df_cutoff <- line_plot_df %>%
  mutate(cases_detected = cumsum(base_IBD)) # Cumulative sum of diagnosed cases

# Step 3: Create a line plot
ggplot(df_cutoff, aes(x = predicted_risk_score, y = cases_detected)) +
  geom_line(color = "blue", size = 1) +
  labs(
    title = "Number of IBD Cases Detected at Different Risk Score Cutoffs",
    x = "Risk Score Cutoff",
    y = "Number of Cases Detected"
  ) +
  theme_minimal()







line_plot_df <- line_plot_df %>% arrange(desc(predicted_risk_score))

# Step 2: Calculate cumulative number of IBD diagnoses and number of people tested
df_cutoff <- line_plot_df %>%
  mutate(cumulative_cases = cumsum(base_IBD),   # Cumulative sum of diagnosed cases
         cumulative_people = row_number())       # Cumulative number of people tested

# Step 3: Determine the total number of cases
total_cases <- sum(line_plot_df$base_IBD)

# Step 4: Calculate thresholds for 40%, 60%, 80% of total cases
threshold_40 <- 0.40 * total_cases
threshold_60 <- 0.60 * total_cases
threshold_80 <- 0.80 * total_cases

# Step 5: Find the cutoff scores for each threshold
cutoff_40 <- df_cutoff %>% filter(cumulative_cases >= threshold_40) %>% slice(1) %>% pull(predicted_risk_score)
cutoff_60 <- df_cutoff %>% filter(cumulative_cases >= threshold_60) %>% slice(1) %>% pull(predicted_risk_score)
cutoff_80 <- df_cutoff %>% filter(cumulative_cases >= threshold_80) %>% slice(1) %>% pull(predicted_risk_score)

# Step 6: Create a plot
ggplot(df_cutoff, aes(x = predicted_risk_score, y = cumulative_people)) +
  geom_line(color = "blue", size = 1) +                             # Plot the cumulative number of people tested
  geom_vline(xintercept = cutoff_40, linetype = "dashed", color = "red") +  # Add vertical line for 40%
  geom_vline(xintercept = cutoff_60, linetype = "dashed", color = "green") +# Add vertical line for 60%
  geom_vline(xintercept = cutoff_80, linetype = "dashed", color = "purple") +# Add vertical line for 80%
  annotate("text", x = cutoff_40, y = max(df_cutoff$cumulative_people), label = "40%", vjust = -1, color = "red") +
  annotate("text", x = cutoff_60, y = max(df_cutoff$cumulative_people), label = "60%", vjust = -1, color = "green") +
  annotate("text", x = cutoff_80, y = max(df_cutoff$cumulative_people), label = "80%", vjust = -1, color = "purple") +
  labs(
    title = "Cumulative Number of People Tested vs. Risk Score",
    x = "Risk Score",
    y = "Cumulative Number of People Tested"
  ) +
  theme_minimal()




cutoff_40 <- df_cutoff %>% filter(cumulative_cases >= threshold_40) %>% slice(1)
cutoff_60 <- df_cutoff %>% filter(cumulative_cases >= threshold_60) %>% slice(1)
cutoff_80 <- df_cutoff %>% filter(cumulative_cases >= threshold_80) %>% slice(1)


ggplot(df_cutoff, aes(x = predicted_risk_score, y = cumulative_people)) +
  geom_line(color = "blue", size = 1) +                             # Plot the cumulative number of people tested
  geom_vline(xintercept = cutoff_40$predicted_risk_score, linetype = "dashed", color = "red") +  # Add vertical line for 40%
  geom_vline(xintercept = cutoff_60$predicted_risk_score, linetype = "dashed", color = "green") +# Add vertical line for 60%
  geom_vline(xintercept = cutoff_80$predicted_risk_score, linetype = "dashed", color = "purple") +# Add vertical line for 80%
  
  # Annotate the number of people tested at each cutoff
  annotate("text", x = cutoff_40$predicted_risk_score, y = cutoff_40$cumulative_people, label = paste("40%:", cutoff_40$cumulative_people),hjust=1.1, vjust = -1, color = "red") +
  annotate("text", x = cutoff_60$predicted_risk_score, y = cutoff_60$cumulative_people, label = paste("60%:", cutoff_60$cumulative_people),hjust=1.5, vjust = -1, color = "green") +
  annotate("text", x = cutoff_80$predicted_risk_score, y = cutoff_80$cumulative_people, label = paste("80%:", cutoff_80$cumulative_people),hjust=2.0, vjust = -1, color = "purple") +

  # Reverse the x-axis so it goes from high to low score
  scale_x_reverse() +  

  # Labels and theme
  labs(
    title = "Cumulative Number of People Tested vs. Risk Score (High to Low)",
    x = "Risk Score",
    y = "Cumulative Number of People Tested"
  ) +
  theme_minimal()









threshold_25 <- 0.25 * total_cases
threshold_40 <- 0.40 * total_cases
threshold_50 <- 0.50 * total_cases
threshold_60 <- 0.60 * total_cases
threshold_75 <- 0.75 * total_cases
threshold_80 <- 0.80 * total_cases

# Step 5: Find the cutoff scores and number of tests for each threshold
cutoff_25 <- df_cutoff %>% filter(cumulative_cases >= threshold_25) %>% slice(1)
cutoff_40 <- df_cutoff %>% filter(cumulative_cases >= threshold_40) %>% slice(1)
cutoff_50 <- df_cutoff %>% filter(cumulative_cases >= threshold_50) %>% slice(1)
cutoff_60 <- df_cutoff %>% filter(cumulative_cases >= threshold_60) %>% slice(1)
cutoff_75 <- df_cutoff %>% filter(cumulative_cases >= threshold_75) %>% slice(1)
cutoff_80 <- df_cutoff %>% filter(cumulative_cases >= threshold_80) %>% slice(1)

# Step 6: Create a plot
ggplot(df_cutoff, aes(x = predicted_risk_score, y = cumulative_people)) +
  geom_line(color = "blue", size = 1) +  # Plot the cumulative number of people tested

  # Add vertical lines for each percentage cutoff
  geom_vline(xintercept = cutoff_25$predicted_risk_score, linetype = "dashed", color = "orange") + # 25%
  geom_vline(xintercept = cutoff_40$predicted_risk_score, linetype = "dashed", color = "red") +    # 40%
  geom_vline(xintercept = cutoff_50$predicted_risk_score, linetype = "dashed", color = "lightblue") + # 50%
  geom_vline(xintercept = cutoff_60$predicted_risk_score, linetype = "dashed", color = "green") +  # 60%
  geom_vline(xintercept = cutoff_75$predicted_risk_score, linetype = "dashed", color = "blue") +   # 75%
  geom_vline(xintercept = cutoff_80$predicted_risk_score, linetype = "dashed", color = "purple") + # 80%

  # Annotate the number of people tested at each cutoff
  annotate("text", x = cutoff_25$predicted_risk_score, y = cutoff_25$cumulative_people, label = paste("25%:", cutoff_25$cumulative_people),hjust=-1.7, vjust = -1, color = "orange") +
  annotate("text", x = cutoff_40$predicted_risk_score, y = cutoff_40$cumulative_people, label = paste("40%:", cutoff_40$cumulative_people),hjust=1.5, vjust = -1, color = "red") +
  annotate("text", x = cutoff_50$predicted_risk_score, y = cutoff_50$cumulative_people, label = paste("50%:", cutoff_50$cumulative_people),hjust=-0.9, vjust = -1, color = "lightblue") +
  annotate("text", x = cutoff_60$predicted_risk_score, y = cutoff_60$cumulative_people, label = paste("60%:", cutoff_60$cumulative_people),hjust=1.9, vjust = -1, color = "green") +
  annotate("text", x = cutoff_75$predicted_risk_score, y = cutoff_75$cumulative_people, label = paste("75%:", cutoff_75$cumulative_people),hjust=-0.2, vjust = -1, color = "blue") +
  annotate("text", x = cutoff_80$predicted_risk_score, y = cutoff_80$cumulative_people, label = paste("80%:", cutoff_80$cumulative_people),hjust=2.4, vjust = -1, color = "purple") +

  # Reverse the x-axis so it goes from high to low score
  scale_x_reverse() +  

  # Labels and theme
  labs(
    title = "Cumulative Number of People Tested vs. Risk Score (High to Low)",
    x = "Risk Score",
    y = "Cumulative Number of People Tested"
  ) +
  theme_minimal()
library(ggplot2)
library(dplyr)

# Define the scaling factor
scaling_factor <- max(df_cutoff$cumulative_people) / max(density(df_cutoff %>% filter(base_IBD == 1) %>% pull(predicted_risk_score))$y)
pdf('/Users/iwan/Research/IBD_risk/project_output/images/IBDcases_vs_tests.pdf')

ggplot(df_cutoff) + 
  # Plot the cumulative number of people tested with `geom_line()`, add label to show in legend
  geom_line(aes(x = predicted_risk_score, y = cumulative_people, color = "Cumulative People Tested"), size = 1) +
  
  # Density plot for IBD cases, scaled to fit on the same plot with a secondary axis
  geom_density(
    data = df_cutoff %>% filter(base_IBD == 1), 
    aes(x = predicted_risk_score, y = ..density.. * scaling_factor, fill = "Density of IBD Cases", color = "Density of IBD Cases"), 
    alpha = 0.2
  ) +
  
  # Add vertical lines for each percentage cutoff with color labels for easier identification
  geom_vline(xintercept = cutoff_25$predicted_risk_score, linetype = "dashed", color = "orange") + # 25%
  geom_vline(xintercept = cutoff_40$predicted_risk_score, linetype = "dashed", color = "red") +    # 40%
  geom_vline(xintercept = cutoff_50$predicted_risk_score, linetype = "dashed", color = "lightblue") + # 50%
  geom_vline(xintercept = cutoff_60$predicted_risk_score, linetype = "dashed", color = "green") +  # 60%
  geom_vline(xintercept = cutoff_75$predicted_risk_score, linetype = "dashed", color = "blue") +   # 75%
  geom_vline(xintercept = cutoff_80$predicted_risk_score, linetype = "dashed", color = "purple") + # 80%
  
  # Annotate the number of people tested at each cutoff with clearer alignment
  annotate("text", x = cutoff_25$predicted_risk_score, y = cutoff_25$cumulative_people, label = paste("25%:", cutoff_25$cumulative_people), hjust=-1.98, vjust = -3, color = "orange") +
  annotate("text", x = cutoff_40$predicted_risk_score, y = cutoff_40$cumulative_people, label = paste("40%:", cutoff_40$cumulative_people), hjust=-2.15, vjust = -3, color = "red") +
  annotate("text", x = cutoff_50$predicted_risk_score, y = cutoff_50$cumulative_people, label = paste("50%:", cutoff_50$cumulative_people), hjust=-2.32, vjust = -3.1, color = "lightblue") +
  annotate("text", x = cutoff_60$predicted_risk_score, y = cutoff_60$cumulative_people, label = paste("60%:", cutoff_60$cumulative_people), hjust=-2.55, vjust = -3, color = "green") +
  annotate("text", x = cutoff_75$predicted_risk_score, y = cutoff_75$cumulative_people, label = paste("75%:", cutoff_75$cumulative_people), hjust=-2.98, vjust = -0.5, color = "blue") +
  annotate("text", x = cutoff_80$predicted_risk_score, y = cutoff_80$cumulative_people, label = paste("80%:", cutoff_80$cumulative_people), hjust=-3.05, vjust = -1, color = "purple") +
  
  
  # Add secondary y-axis for density
  scale_y_continuous(
    name = "Cumulative Number of People Tested",
    sec.axis = sec_axis(~ . / scaling_factor, name = "Density of IBD Cases")
  ) +

  # Labels and legend
  labs(
    title = "Cumulative Number of People Tested vs. Risk Score (High to Low)",
    x = "Risk Score",
    y = "Cumulative Number of People Tested",
    color = "", fill = ""
  ) +
  
  # Customize colors in legend
  scale_color_manual(values = c("Cumulative People Tested" = "blue")) +
  scale_fill_manual(values = c("Density of IBD Cases" = "red")) +

  theme_minimal() +
  theme(legend.position = "top")  # Place legend at the top for better readability


dev.off()
```

# calculate quantiles and plot
```{r }


high_riskers <- DAG3_subset_dataframe[DAG3_subset_dataframe$predicted_risk_score > (mean(DAG3_subset_dataframe$predicted_risk_score) +  sd(DAG3_subset_dataframe$predicted_risk_score)), ]
low_riskers <- DAG3_subset_dataframe[DAG3_subset_dataframe$predicted_risk_score < (mean(DAG3_subset_dataframe$predicted_risk_score) -  sd(DAG3_subset_dataframe$predicted_risk_score)), ]


#high_riskers <- DAG3_subset_dataframe[DAG3_subset_dataframe$predicted_risk_score > quantile(DAG3_subset_dataframe$predicted_risk_score)[4], ]

#low_riskers <- DAG3_subset_dataframe[DAG3_subset_dataframe$predicted_risk_score < quantile(DAG3_subset_dataframe$predicted_risk_score)[2], ]
#table(high_riskers$base_IBD1a2a)
#table(low_riskers$base_IBD1a2a)



q1 <- DAG3_subset_dataframe[DAG3_subset_dataframe$predicted_risk_score < quantile(DAG3_subset_dataframe$predicted_risk_score)[2], ]
q2 <- DAG3_subset_dataframe[DAG3_subset_dataframe$predicted_risk_score > quantile(DAG3_subset_dataframe$predicted_risk_score)[3], ]
q3 <- DAG3_subset_dataframe[DAG3_subset_dataframe$predicted_risk_score > quantile(DAG3_subset_dataframe$predicted_risk_score)[4], ]
q4 <- DAG3_subset_dataframe[DAG3_subset_dataframe$predicted_risk_score > quantile(DAG3_subset_dataframe$predicted_risk_score)[4], ]


# Load necessary libraries
library(dplyr)
library(ggplot2)

# Create quantiles for predicted_risk_score
DAG3_subset_dataframe$quantile <- cut(
  DAG3_subset_dataframe$predicted_risk_score, 
  breaks = quantile(DAG3_subset_dataframe$predicted_risk_score, probs = seq(0, 1, 0.25), na.rm = TRUE),
  include.lowest = TRUE,
  labels = c("Q1", "Q2", "Q3", "Q4")
)

# Calculate proportions of cases within each quantile
quantile_summary <- DAG3_subset_dataframe %>%
  group_by(quantile) %>%
  summarise(
    total_cases = sum(base_IBD1a2a, na.rm = TRUE), # Sum the number of cases in each quantile
    total_individuals = n(),                       # Total number of individuals in each quantile
    proportion = total_cases / total_individuals   # Proportion of cases in each quantile
  )

pdf('/Users/iwan/Research/IBD_risk/project_output/images/riskscore_quantiles_over_cases.pdf')

# Plot the data
ggplot(quantile_summary, aes(x = quantile, y = proportion)) +
  geom_bar(stat = "identity", fill = "red", alpha = 0.8) +
  scale_y_continuous(labels = scales::percent_format()) + # Convert y-axis to percentage
  labs(
    title = "Proportion of Cases by Predicted Risk Score Quantiles",
    x = "Predicted Risk Score Quantile",
    y = "Proportion of Cases"
  ) +
  theme_classic()
dev.off()


```